{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define major functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Import packages and define functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Packages imported and functions defined\n"
     ]
    }
   ],
   "source": [
    "# Author: Connor Ludwig\n",
    "# Organization: Bintu Lab, Stanford University\n",
    "# Date: 10/8/2021\n",
    "\n",
    "import sys\n",
    "import os\n",
    "import numpy as np\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.ticker as ticker\n",
    "import matplotlib.gridspec as gridspec\n",
    "from matplotlib.lines import Line2D\n",
    "from scipy.special import factorial\n",
    "import seaborn as sns\n",
    "import pandas as pd\n",
    "import re\n",
    "import statistics\n",
    "from scipy import stats\n",
    "import math\n",
    "\n",
    "# !{sys.executable} -m pip install mpl-axes-aligner\n",
    "import mpl_axes_aligner\n",
    "\n",
    "##############################\n",
    "\n",
    "# define function charge() that calculates the net charge of an amino acid sequence\n",
    "def charge(AAseq):\n",
    "    netCharge = 0\n",
    "    for aa in AAseq:\n",
    "        if (aa == 'K') or (aa == 'R'):\n",
    "            netCharge += 1\n",
    "        elif (aa == 'D') or (aa == 'E'):\n",
    "            netCharge -= 1\n",
    "    return netCharge\n",
    "\n",
    "##############################\n",
    "\n",
    "# annotate viral family, Baltimore classification, and virus genome based on new information\n",
    "famDict = {'adenovirus':'Adenoviridae', 'herpesvirus':'Herpesviridae', 'htlv':'Retroviridae', 'papillomavirus':'Papillomaviridae',\n",
    "           'hepatitis b':'Hepadnaviridae', 'syncytial':'Paramyxoviridae', 'hcv':'Flaviviridae', 'coronavirus':'Coronaviridae',\n",
    "           'aav':'Parvoviridae', 'polyomavirus':'Polyomaviridae', 'duvenhage':'Rhabdoviridae', 'mokola':'Rhabdoviridae',\n",
    "           'lyssavirus':'Rhabdoviridae', 'influenza':'Orthomyxoviridae', 'parvovirus':'Parvoviridae', 't-lymphotropic':'Retroviridae',\n",
    "           'pox':'Poxviridae', 'hiv':'Retroviridae', 'rabies virus':'Rhabdoviridae', 'orf virus':'Poxviridae', 'mumps':'Paramyxoviridae',\n",
    "           'bunya':'Bunyaviridae', 'pneumovirus':'Pneumoviridae', 'borna':'Bornaviridae', 'lcmv':'Arenaviridae', 'simian virus':'Polyomaviridae',\n",
    "           'yaba':'Poxviridae', 'retrovirus':'Retroviridae', 'vaccinia':'Poxviridae', 'measles':'Paramyxoviridae', 'rift valley fever':'Bunyaviridae',\n",
    "           'molluscum':'Poxviridae', 'bovine papular stomatitis':'Poxviridae', 'encephalitis':'Bunyaviridae', 'simian foamy':'Retroviridae',\n",
    "           'vesicular stomatitis':'Rhabdoviridae', 'hdv':'Deltaviridae'}\n",
    "\n",
    "BaltDict = {'Adenoviridae':'1', 'Herpesviridae':'1', 'Papillomaviridae':'1', 'Polyomaviridae':'1', 'Poxviridae':'1', 'Anelloviridae':'2',\n",
    "            'Circoviridae':'2', 'Parvoviridae':'2', 'Picobirnaviridae':'3', 'Reoviridae':'3', 'Astroviridae':'4', 'Caliciviridae':'4',\n",
    "            'Coronaviridae':'4', 'Flaviviridae':'4', 'Hepeviridae':'4', 'Picornaviridae':'4', 'Togaviridae':'4', 'Arenaviridae':'5',\n",
    "            'Bornaviridae':'5', 'Bunyaviridae':'5', 'Filoviridae':'5', 'Orthomyxoviridae':'5', 'Paramyxoviridae':'5', 'Pneumoviridae':'5',\n",
    "            'Rhabdoviridae':'5', 'Retroviridae':'6', 'Hepadnaviridae':'7', 'Deltaviridae':'0'}\n",
    "\n",
    "genDict = {'1':'DNA', '2':'DNA', '3':'RNA', '4':'RNA', '5':'RNA', '6':'RNA', '7':'DNA', '0':'RNA'}\n",
    "\n",
    "def prepDF(Rhos, tileInfo, screentype):\n",
    "    # merge dataframes, sort, and invert scores to be log2(ON:OFF)\n",
    "    # so, positive = activation and negative = repression\n",
    "    Rhos = Rhos.rename(columns={'label':'Oligo ID'})\n",
    "    df = pd.merge(Rhos, tileInfo, how='inner', on='Oligo ID')\n",
    "    df = df.sort_values(by='Avg')\n",
    "\n",
    "    if screentype == 'Activation':\n",
    "        df['R1'] = -1 * df['R1']\n",
    "        df['R2'] = -1 * df['R2']\n",
    "        df['Avg'] = -1 * df['Avg']\n",
    "    elif screentype != 'Repression':\n",
    "        print('Error: must specify Activation or Repression for screen type')\n",
    "        return None\n",
    "\n",
    "    # add tile net charge\n",
    "    chargeList = [charge(seq) for seq in list(df['Tile Sequence'])]\n",
    "    df['Net Charge'] = chargeList\n",
    "    \n",
    "    # if no virus name, fill with '-' (mostly randomers and fiducials)\n",
    "    df['Virus'] = df['Virus'].fillna('-')\n",
    "\n",
    "    # initialize lists for virus family name, Baltimore classification, and genome type\n",
    "    virFam = []\n",
    "    virBalt = []\n",
    "    virGen = []\n",
    "    # iterate over virus name in dataframe\n",
    "    for i in list(df['Virus']):\n",
    "        # default assumption is non-virus\n",
    "        famName = 'control'\n",
    "        BaltClass = '-'\n",
    "        genType = '-'\n",
    "        # if the 'virus' is not '-', proceed:\n",
    "        if i != '-':\n",
    "            # initially set a variable skip to False for each virus name\n",
    "            skip = False\n",
    "            # see if famDict key is a substring of the virus name\n",
    "            for key in famDict:\n",
    "                # if skip is still false, store the family, class, and genome informtion\n",
    "                # if skip is true, do not store this information\n",
    "                if skip == False:\n",
    "                    if key in i.lower():\n",
    "                        famName = famDict[key]\n",
    "                        BaltClass = BaltDict[famName]\n",
    "                        genType = genDict[BaltClass]\n",
    "                        # set skip to true once the info has been found/stored\n",
    "                        skip = True\n",
    "        \n",
    "        # add family, class, and genome info to appropriate lists\n",
    "        # default will be added if these are controls (random, fiducial, non-virus proteins)\n",
    "        virFam.append(famName)\n",
    "        virBalt.append(BaltClass)\n",
    "        virGen.append(genType)\n",
    "    \n",
    "    # add the above info to the dataframe\n",
    "    df['Virus Family'] = virFam\n",
    "    df['Baltimore Class'] = virBalt\n",
    "    df['Genome Type'] = virGen\n",
    "\n",
    "    print('Family, class, and genome type info added')\n",
    "    return df\n",
    "    \n",
    "##############################\n",
    "\n",
    "def mergeDFs(df, prefixlist, tiledfList):\n",
    "    df_cols = df.columns\n",
    "    tiledfs = [pd.read_csv(file) for file in tiledfList]\n",
    "    # add Oligo ID column\n",
    "    for i in range(len(tiledfs)):\n",
    "        tiledfs[i]['Oligo ID'] = prefixList[i] + '_' + tiledfs[i]['Tile ID']\n",
    "        tiledfs[i] = tiledfs[i].drop(columns=['DNA Sequence', 'Library', 'Type'])\n",
    "\n",
    "    # identify which columns are shared and which are unique\n",
    "    tiledf_cols = tiledfs[0].columns\n",
    "\n",
    "    unique_cols = [col for col in df_cols if col not in set(tiledf_cols)]\n",
    "    shared_cols = [col for col in df_cols if col in set(tiledf_cols)]\n",
    "\n",
    "    # merge each dataframe to make a master dataframe\n",
    "    df['In Original'] = 1\n",
    "\n",
    "    fulldf = df.copy()\n",
    "    for i in range(len(tiledfs)):\n",
    "        fulldf = pd.merge(fulldf, tiledfs[i], how='outer', on=shared_cols)\n",
    "    print('Dataframes merged')\n",
    "    fulldf['In Original'] = fulldf['In Original'].fillna(0)\n",
    "\n",
    "    df['Duplicate'] = 0\n",
    "    fulldf['Duplicate'] = 0\n",
    "    \n",
    "    # remove duplicated entries, which can happen if an immune control is in the sublibrary itself\n",
    "    df = df.drop_duplicates(subset='Oligo ID', keep='first')\n",
    "    fulldf = fulldf.drop_duplicates(subset='Oligo ID', keep='first').set_index('Oligo ID')\n",
    "    print('Preparing to re-add dropouts and non-unique tiles')\n",
    "    # identify those tiles that don't have an average value (either dropouts or non-unique)\n",
    "    absent = fulldf[fulldf['Avg'].isnull()]\n",
    "\n",
    "    for ID in list(absent.index):\n",
    "        seq = absent.loc[ID, 'Tile Sequence']\n",
    "        # check if amino acid sequence appeared in screen - if not, these are dropouts\n",
    "        if seq not in list(df['Tile Sequence']):\n",
    "            continue\n",
    "        # fill in missing information for non-unique tiles based on the amino acid sequence\n",
    "        for col in unique_cols:\n",
    "            absent.loc[ID, col] = df.loc[df['Tile Sequence'] == seq, col].values[0]\n",
    "            absent.loc[ID, 'Duplicate'] = 1\n",
    "    \n",
    "    absent = absent.reset_index()\n",
    "    fulldf = fulldf.reset_index()\n",
    "    \n",
    "    # concatenate dataframes, sort, and save\n",
    "    fulldf = pd.concat([df, absent], ignore_index=True)\n",
    "    fulldf = fulldf.sort_values(by='Avg', ascending=False)\n",
    "\n",
    "    # filter out things that weren't sequenced (any protein smaller than 60 residues)\n",
    "    fulldf['Tile End'] = fulldf['Tile End'].fillna(80)\n",
    "    fulldf = fulldf[fulldf['Tile End'] >= 60]\n",
    "    print('Information added for all tiles, including non-unique')\n",
    "    return fulldf\n",
    "\n",
    "##############################\n",
    "\n",
    "\n",
    "def filterDF(in_df, library, screentype, sigma):\n",
    "    df = in_df.copy()\n",
    "    print('Filtering dataframe for %s %s screen' % (library, screentype))\n",
    "    # add ON and OFF counts for each entry (i.e. reads across both fractions)\n",
    "    df['countsSUM_R1'] = df['countsON_R1'] + df['countsOFF_R1']\n",
    "    df['countsSUM_R2'] = df['countsON_R2'] + df['countsOFF_R2']\n",
    "    # create dataframe without duplicated entries for calculation of total reads\n",
    "    df_nondup = df[df['Duplicate']==0]\n",
    "    df_nondup = df_nondup.fillna(0)\n",
    "    \n",
    "    # calculate total reads per replicate\n",
    "    sumR1 = sum(list(df_nondup['countsSUM_R1']))\n",
    "    sumR2 = sum(list(df_nondup['countsSUM_R2'])) \n",
    "    \n",
    "    # remove entries that lack a score in at least one replicate in the original dataframe\n",
    "    df = df.dropna(subset=['R1', 'R2'])\n",
    "    df2 = df.copy()\n",
    "    \n",
    "    # filter out entries with fewer than 50 reads between both fractions, regardless of sequencing depth\n",
    "    df = df[(df['countsSUM_R1']>50) & (df['countsSUM_R2']>50)]\n",
    "    # subset for library of interest\n",
    "    df = df[df['Oligo ID'].str.contains(library)]\n",
    "    # compute random control mean and standard deviation for calling hits\n",
    "    df['Hit'] = 0\n",
    "    randoms = list(df[(df['Oligo ID'].str.contains('random')) & (df['Oligo ID'].str.contains(library))]['Avg'])\n",
    "    \n",
    "    cutoff = statistics.mean(randoms) + sigma*statistics.stdev(randoms)\n",
    "    df.loc[df.Avg >= cutoff, 'Hit'] = 1\n",
    "    \n",
    "    print('Cutoff for hits with %s sigma is %s' % (str(sigma), str(cutoff)))\n",
    "    df['Screen'] = screentype\n",
    "    \n",
    "    # do same for unfiltered dataframe\n",
    "    df2 = df2[df2['Oligo ID'].str.contains(library)]\n",
    "    df2['Hit'] = 0\n",
    "    df2.loc[df2.Avg >= cutoff, 'Hit'] = 1\n",
    "    df2['Screen'] = screentype\n",
    "    \n",
    "    # return filtered and sorted dataframe\n",
    "    print('Dataframe filtered')\n",
    "    return df.sort_values(by='Avg', ascending=False), df2, cutoff\n",
    "\n",
    "##############################\n",
    "\n",
    "print('Packages imported and functions defined')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Prepare dataframes with more detailed information and save with and without duplicate tiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CSV files compiled\n"
     ]
    }
   ],
   "source": [
    "# sys.argv[1] = input csv file with codon-optimized oligos; must have 'Virus' column with various types/strains\n",
    "df = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/uniref_reviewed_HHV+SuHV_all-tiles.csv')\n",
    "\n",
    "# define virus and subfamily classification scheme for mapping of tile to virus to subfamily\n",
    "herpesvirusList = ['HHV-1', 'HHV-2', 'HHV-3', 'HHV-4', 'HHV-5', 'HHV-6', 'HHV-7', 'HHV-8', 'CeHV', 'SuHV']\n",
    "herpesvirus2subfamily = {'HHV-1':'alpha_herpesvirus', 'HHV-2':'alpha_herpesvirus', 'HHV-3':'alpha_herpesvirus',\n",
    "                         'CeHV':'alpha_herpesvirus', 'SuHV':'alpha_herpesvirus', 'HHV-4':'gamma_herpesvirus',\n",
    "                         'HHV-5':'beta_herpesvirus', 'HHV-6':'beta_herpesvirus', 'HHV-7':'beta_herpesvirus',\n",
    "                         'HHV-8':'gamma_herpesvirus', 'other':'alpha_herpesvirus'}\n",
    "\n",
    "# initialize arrays to store mapping info\n",
    "herpesvirus = []\n",
    "subfamily = []\n",
    "\n",
    "# for each row, count the number of times a herpesvirus type appears in the 'Virus' column\n",
    "# store the most common as the associated virus type and store the corresponding subfamily info\n",
    "for row in df['Virus']:\n",
    "    v_count = 0\n",
    "    v_id = 'other'\n",
    "    for v in herpesvirusList:\n",
    "        if row.count(v) > v_count:\n",
    "            v_count = row.count(v)\n",
    "            v_id = v\n",
    "    herpesvirus.append(v_id)\n",
    "    subfamily.append(herpesvirus2subfamily[v_id])\n",
    "\n",
    "# add the virus type and subfamily (sublibrary) to the dataframe\n",
    "df['Type'] = herpesvirus\n",
    "df['Library'] = subfamily\n",
    "\n",
    "# read in files for compilation\n",
    "randset2 = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/random_controls_set02.csv')\n",
    "randset3 = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/random_controls_set03.csv')\n",
    "randset4 = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/random_controls_set04.csv')\n",
    "immunectrls = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/uniprot_reviewed_immune_controls_manual-trim_unique-tiles_codon-opt-oligos.csv')\n",
    "fiducialctrls = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/fiducial_controls_alpha_herpesvirus.csv')\n",
    "\n",
    "# compile and save dataframes\n",
    "df_aHV = df[df['Library']=='alpha_herpesvirus']\n",
    "df_aHV = pd.concat([df_aHV, randset2, immunectrls, fiducialctrls])\n",
    "df_aHV.to_csv('../Supplemental Tables - CSV/HHV-tiling/aHV_all-tiles_random-02_immune_fiducial_compiled.csv', index=False)\n",
    "\n",
    "df_bHV = df[df['Library']=='beta_herpesvirus']\n",
    "df_bHV = pd.concat([df_bHV, randset3, immunectrls, fiducialctrls])\n",
    "df_bHV.to_csv('../Supplemental Tables - CSV/HHV-tiling/bHV_all-tiles_random-03_immune_fiducial_compiled.csv', index=False)\n",
    "\n",
    "df_gHV = df[df['Library']=='gamma_herpesvirus']\n",
    "df_gHV = pd.concat([df_gHV, randset4, immunectrls, fiducialctrls])\n",
    "df_gHV.to_csv('../Supplemental Tables - CSV/HHV-tiling/gHV_all-tiles_random-04_immune_fiducial_compiled.csv', index=False)\n",
    "print('CSV files compiled')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Prepare master dataframe for primary HHV activation and repression screens"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Family, class, and genome type info added\n",
      "Dataframes merged\n",
      "Preparing to re-add dropouts and non-unique tiles\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\cludw\\anaconda3\\lib\\site-packages\\pandas\\core\\indexing.py:1765: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Information added for all tiles, including non-unique\n",
      "aHV-bHV-gHV activation screen data and metadata have been compiled\n",
      "Family, class, and genome type info added\n",
      "Dataframes merged\n",
      "Preparing to re-add dropouts and non-unique tiles\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\cludw\\anaconda3\\lib\\site-packages\\pandas\\core\\indexing.py:1765: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Information added for all tiles, including non-unique\n",
      "aHV-bHV-gHV repression screen data and metadata have been compiled\n"
     ]
    }
   ],
   "source": [
    "# Author: Connor Ludwig\n",
    "# Organization: Bintu Lab, Stanford University\n",
    "# Date: 10/8/2021\n",
    "\n",
    "# read in rhos file, extra tile info, and mapping file\n",
    "Rhos_A = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_ActivD2_combo.csv')\n",
    "Rhos_R = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_ReprD5_combo.csv')\n",
    "tileInfo = pd.read_csv('../Supplemental Tables - CSV/20201012_CL_Twist_libraries_compiled_FINAL+filtered.csv')\n",
    "tileInfo['Protein ID'] = tileInfo['Oligo ID'].str.split('_').str[:-1].str.join('_').str.split('_').str[1:].str.join('_')\n",
    "mapdf = pd.read_csv('../Supplemental Tables - CSV/uID_protein_virus_mapping_HHV_expanded.csv')\n",
    "mapdf = mapdf.rename(columns={'Virus':'Virus Abbreviation', 'UniProt ID':'Protein ID'})\n",
    "mapdf = mapdf[['Protein ID', 'Virus Abbreviation', 'UniProt Protein Family', 'Mocarski Protein Family', 'Protein', 'Gene', 'Original', 'Kinetic Class']]\n",
    "tileInfo = pd.merge(tileInfo, mapdf, how='outer', on='Protein ID')\n",
    "\n",
    "# specify sub-libraries in screen and input files with all tiles\n",
    "prefixList = ['aHV', 'bHV', 'gHV']\n",
    "tiledfList = ['../Supplemental Tables - CSV/HHV-tiling/aHV_all-tiles_random-02_immune_fiducial_compiled.csv',\n",
    "              '../Supplemental Tables - CSV/HHV-tiling/bHV_all-tiles_random-03_immune_fiducial_compiled.csv',\n",
    "              '../Supplemental Tables - CSV/HHV-tiling/gHV_all-tiles_random-04_immune_fiducial_compiled.csv']\n",
    "\n",
    "df_A = prepDF(Rhos_A, tileInfo,'Activation')\n",
    "fulldf_A = mergeDFs(df_A, prefixList, tiledfList)\n",
    "savename_A = '../Supplemental Tables - CSV/HHV-tiling/%s_ActivD2_bothseqs_combo_detailed_log2OnOff' % '-'.join(prefixList)\n",
    "df_A.to_csv(savename_A + '.csv', index=False)\n",
    "fulldf_A.to_csv(savename_A + '_reAddDups.csv', index=False)\n",
    "print('%s activation screen data and metadata have been compiled' % '-'.join(prefixList))\n",
    "\n",
    "df_R = prepDF(Rhos_R, tileInfo, 'Repression')\n",
    "fulldf_R = mergeDFs(df_R, prefixList, tiledfList)\n",
    "savename_R = '../Supplemental Tables - CSV/HHV-tiling/%s_ReprD5_bothseqs_combo_detailed_log2OnOff' % '-'.join(prefixList)\n",
    "df_R.to_csv(savename_R + '.csv', index=False)\n",
    "fulldf_R.to_csv(savename_R + '_reAddDups.csv', index=False)\n",
    "print('%s repression screen data and metadata have been compiled' % '-'.join(prefixList))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Create single master dataframe for the screen that has all tile information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtering dataframe for aHV|bHV|gHV Activation screen\n",
      "Cutoff for hits with 2 sigma is 1.0846347878236864\n",
      "Dataframe filtered\n",
      "Filtering dataframe for aHV|bHV|gHV Repression screen\n",
      "Cutoff for hits with 2 sigma is 0.6987727639587654\n",
      "Dataframe filtered\n"
     ]
    }
   ],
   "source": [
    "# Author: Connor Ludwig\n",
    "# Organization: Bintu Lab, Stanford University\n",
    "# Date: 10/8/2021\n",
    "\n",
    "# avoid running the above block again and instead import the files created above\n",
    "dfA = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/aHV-bHV-gHV_ActivD2_bothseqs_combo_detailed_log2OnOff_reAddDups.csv')\n",
    "dfR = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/aHV-bHV-gHV_ReprD5_bothseqs_combo_detailed_log2OnOff_reAddDups.csv')\n",
    "sigma = 2\n",
    "\n",
    "# print(dfA[dfA['Oligo ID'].str.contains('CoV')])\n",
    "dfA_HHV, dfA2_HHV, cutoffA_HHV = filterDF(dfA, 'aHV|bHV|gHV', 'Activation', sigma)\n",
    "dfR_HHV, dfR2_HHV, cutoffR_HHV = filterDF(dfR, 'aHV|bHV|gHV', 'Repression', sigma)\n",
    "\n",
    "df = pd.concat([dfA_HHV, dfR_HHV], ignore_index=True)\n",
    "df2 = pd.concat([dfA2_HHV, dfR2_HHV], ignore_index=True)\n",
    "\n",
    "inputDir = '../Supplemental Tables - CSV/HHV-tiling'\n",
    "savename = os.path.join(inputDir, 'HHV_ActivD2-ReprD5_detailed.csv')\n",
    "df.to_csv(savename, index=False)\n",
    "df2.to_csv(savename[:-4] + '_no-filter.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Screen Vizualization - Reproducibility Plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Function defined\n"
     ]
    }
   ],
   "source": [
    "# Author: Connor Ludwig\n",
    "# Organization: Bintu Lab, Stanford University\n",
    "# Date: 10/15/2021\n",
    "\n",
    "font = {'size':8,\n",
    "        'family':'Arial'}\n",
    "matplotlib.rc('font', **font)\n",
    "\n",
    "##################################################################\n",
    "\n",
    "def viz_reproducibility(df, library, screentype, cutoff, hits_to_highlight, colors, \n",
    "                        figsize, labelpos, savedir, lims=False):\n",
    "#     df = df[(df['Oligo ID'].str.contains(library)) & (df['Screen']==screentype)]\n",
    "    df = df[df['Screen']==screentype]\n",
    "    \n",
    "    # compute total number of proteins with hits prior to deduplication of tiles\n",
    "    totalprot = len(set(list(df[~df['Type'].str.contains('control')]['Protein ID'])))\n",
    "    totalprothit = len(set(list(df[(~df['Type'].str.contains('control')) & (df['Hit']==1)]['Protein ID'])))\n",
    "\n",
    "    df = df[df['Duplicate']==0]\n",
    "    df['Protein ID'] = df['Tile ID'].str.split('|').str.join('_').str.split('_').str[:-1].str.join('_')\n",
    "    \n",
    "    ###################################################################\n",
    "\n",
    "    df['Annotation'] = np.where(df['UniProt Protein Family'].isin(hits_to_highlight), 'Highlight', '')\n",
    "    print(set(list(df[df['Annotation']=='Highlight']['UniProt Protein Family'])))\n",
    "    df['Annotation'] = np.where((df['Tile Info'].str.lower().str.contains('random', na=False)) | (df['Type'].str.contains('random', na=False)), 'Random control', df['Annotation'])\n",
    "    \n",
    "    totalprotall = len(set(list(df[df['Annotation']!='random']['Protein ID'])))\n",
    "    print('Stats for %s %s screen' % (library, screentype))\n",
    "    print('Total proteins (including controls) detected: %d' % totalprotall)\n",
    "    print('Total proteins (excluding controls) detected: %d' % totalprot)\n",
    "    print('Total tiles passing filter of %s: %d' % (str(round(cutoff, 3)), len(df)))\n",
    "    print('Total hits: %d' % (len(df[df['Hit']==1])))\n",
    "    print('Total proteins encoding at least one hit: %d' % totalprothit)\n",
    "    print('Total non-hits: %d\\n' % (len(df[df['Hit']==0])))\n",
    "    \n",
    "    if lims == False:\n",
    "        lims = (min([min(list(df['R1'])), min(list(df['R2']))]) - 0.5, max([max(list(df['R1'])), max(list(df['R2']))]) + 0.5)\n",
    "\n",
    "    #################################################################\n",
    "\n",
    "    savename = os.path.join(savedir, library+'_'+screentype)\n",
    "    \n",
    "    if os.path.isdir(inputDir) == False:\n",
    "        print('Making directory called %s to store files and plots' % fig1Dir)\n",
    "        os.mkdir(fig1Dir)\n",
    "        \n",
    "    pal = sns.set_palette(sns.color_palette(colors))\n",
    "    greys = ['#BBBBBB', '#777777']\n",
    "    alpha = 1\n",
    "    markersize = 8\n",
    "\n",
    "    highlight = df[(df['Annotation']=='Highlight') & (df['Hit']==1)]\n",
    "    highlight['UniProt Protein Family'] = pd.Categorical(highlight['UniProt Protein Family'], categories=hits_to_highlight, ordered=True)\n",
    "    highlight = highlight.sort_values(by='UniProt Protein Family')\n",
    "    highlightX = list(highlight['R1'])\n",
    "    highlightY = list(highlight['R2'])\n",
    "    highlightInfo = list(highlight['Tile Info'])\n",
    "\n",
    "    # all tiles\n",
    "    if screentype == 'Activation':\n",
    "        score = 'ON:OFF'\n",
    "        pad = 4\n",
    "    elif screentype == 'Repression':\n",
    "        score = 'OFF:ON'\n",
    "        pad = 2\n",
    "    \n",
    "    fig, ax = plt.subplots(figsize=figsize)\n",
    "    sns.scatterplot(x='R1', y='R2', data=df[df['Annotation']==''], color=greys[0], alpha=alpha, linewidth=0, s=markersize)\n",
    "    sns.scatterplot(x='R1', y='R2', data=df[df['Annotation']=='Random control'], color=greys[1], alpha=alpha, linewidth=0, s=markersize)\n",
    "    sns.scatterplot(x='R1', y='R2', data=highlight, alpha=alpha, linewidth=0, s=markersize, hue='UniProt Protein Family', palette=pal)\n",
    "    \n",
    "    legend_elements = [Line2D([0], [0], marker='o', color='#FFFFFF', label=hits_to_highlight[i], markerfacecolor=colors[i], markersize=markersize/2) for i in range(len(hits_to_highlight))]\n",
    "    legend_elements.extend([Line2D([0], [0], marker='o', color='#FFFFFF', label='Random', markerfacecolor=greys[1], markersize=markersize/2),\n",
    "                           Line2D([0], [0], marker='o', color='#FFFFFF', label='All tiles', markerfacecolor=greys[0], markersize=markersize/2)])\n",
    "    ax.legend(handles=legend_elements, loc=4, frameon=False, bbox_to_anchor=(1.02, 0), labelspacing=0,\n",
    "              handlelength=0, borderpad=0.1, fontsize=6)\n",
    "    \n",
    "    x = [cutoff-pad, cutoff+pad]\n",
    "    y = [cutoff+pad, cutoff-pad]\n",
    "    plt.plot(x, y, ls='--', color='#444444', lw=1)\n",
    "    plt.xlim(lims)\n",
    "    plt.ylim(lims)\n",
    "\n",
    "    plt.xlabel('Rep 1, log$_2$(%s)' % score, labelpad=0.5)\n",
    "    plt.ylabel('Rep 2, log$_2$(%s)' % score, labelpad=0.5)\n",
    "    r1 = np.array(list(df[df['Hit']==1]['R1']))\n",
    "    r2 = np.array(list(df[df['Hit']==1]['R2']))\n",
    "    r, p = stats.spearmanr(r1, r2)\n",
    "    \n",
    "    if screentype == 'Activation':\n",
    "        plt.text(x=(lims[0] + 1.2), y=(lims[1] - 2.2), s=('n = %s tiles' % format(len(df), ',d')), fontsize=6)\n",
    "        plt.annotate(xy=labelpos, text='Activators', rotation=45, fontsize=6)\n",
    "        plt.xticks(ticks=[-5, 0, 5, 10], labels=[-5, 0, 5, 10])\n",
    "    elif screentype == 'Repression':\n",
    "        plt.text(x=(lims[0] + 0.7), y=(lims[1] - 1.3), s=('n = %s tiles' % format(len(df), ',d')), fontsize=6)\n",
    "        plt.annotate(xy=labelpos, text='Repressors', rotation=45, fontsize=6)\n",
    "        plt.xticks(ticks=[-2, 0, 2, 4, 6], labels=[-2, 0, 2, 4, 6])\n",
    "        plt.yticks(ticks=[-2, 0, 2, 4, 6], labels=[-2, 0, 2, 4, 6])\n",
    "#     plt.title(screentype, fontsize=8)\n",
    "    plt.tight_layout()\n",
    "    plt.savefig(savename + '_alltiles_annotated.png', dpi=1200)\n",
    "    plt.close()\n",
    "    return lims\n",
    "\n",
    "print('Function defined')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Reproducibility scatterplots featured in Fig. S2D & S2E"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'DBP', 'EBNA', 'VIRF'}\n",
      "Stats for HHV Activation screen\n",
      "Total proteins (including controls) detected: 917\n",
      "Total proteins (excluding controls) detected: 885\n",
      "Total tiles passing filter of 1.085: 36476\n",
      "Total hits: 445\n",
      "Total proteins encoding at least one hit: 165\n",
      "Total non-hits: 36031\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-4-a24d0e21448a>:55: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  highlight['UniProt Protein Family'] = pd.Categorical(highlight['UniProt Protein Family'], categories=hits_to_highlight, ordered=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'EBNA', 'IE', 'VIRF'}\n",
      "Stats for HHV Repression screen\n",
      "Total proteins (including controls) detected: 917\n",
      "Total proteins (excluding controls) detected: 885\n",
      "Total tiles passing filter of 0.699: 36506\n",
      "Total hits: 989\n",
      "Total proteins encoding at least one hit: 277\n",
      "Total non-hits: 35517\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-4-a24d0e21448a>:55: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  highlight['UniProt Protein Family'] = pd.Categorical(highlight['UniProt Protein Family'], categories=hits_to_highlight, ordered=True)\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_ActivD2-ReprD5_detailed.csv')\n",
    "colorsA = ['#F3CE48', '#F1815F', '#DB3982']\n",
    "limsA_HHV = viz_reproducibility(df, 'HHV', 'Activation', cutoffA_HHV, ['EBNA', 'VIRF', 'DBP'],\n",
    "                                colorsA, (1.73, 1.7), (2, 4.6), '../Individual Figure Panels/Figure S3')\n",
    "\n",
    "colorsR = ['#23BEDB', '#B276C2', '#DB3982']\n",
    "limsR_HHV = viz_reproducibility(df, 'HHV', 'Repression', cutoffR_HHV, ['EBNA', 'VIRF', 'IE'],\n",
    "                                colorsR, (1.7, 1.7), (-0.5, 2.5), '../Individual Figure Panels/Figure S3')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Domain Calling for Visualization as Tiling Plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Domain calling/plotting functions defined\n"
     ]
    }
   ],
   "source": [
    "# Author: Connor Ludwig\n",
    "# Organization: Bintu Lab, Stanford University\n",
    "# Date updated: 5/10/2022\n",
    "\n",
    "# NOTE: these functions use a few global variables, including:\n",
    "# - hits thresholds for activation and repression screens\n",
    "# - minimum and maximum screen scores for non-controls to set axis limits\n",
    "# - dictionary that maps UniProt IDs to protein and virus\n",
    "\n",
    "# make sure to run the above cells to get these global variables\n",
    "\n",
    "import mpl_axes_aligner\n",
    "from matplotlib.ticker import MultipleLocator\n",
    "from matplotlib.ticker import AutoMinorLocator\n",
    "\n",
    "# this function gets domain start and stop coordinates, as well as max tile score\n",
    "def domainLimits(temp_df):\n",
    "    span_starts = []\n",
    "    span_ends = []\n",
    "    scores = []\n",
    "    temp_df_hits = temp_df[temp_df['Hit']==1]\n",
    "    indices = range(0, len(temp_df_hits))\n",
    "    temp_df_hits.insert(0, 'index', indices)\n",
    "    temp_df_hits.set_index('index', inplace=True)\n",
    "    \n",
    "    if len(temp_df_hits) == 0:\n",
    "        return span_starts, span_ends, scores\n",
    "    temp_start = np.nan\n",
    "    temp_scores = []\n",
    "    # create range for all tiles excluding the last tile\n",
    "    for i in range(0, len(temp_df_hits)):\n",
    "        # if tile is last one\n",
    "        if i == (len(temp_df_hits) - 1):\n",
    "            # if temporary start of domain has been defined\n",
    "            if ~np.isnan(temp_start):\n",
    "                span_starts.append(temp_start)\n",
    "                span_ends.append(temp_df_hits.loc[i, 'Tile End'])\n",
    "                temp_scores.append(temp_df_hits.loc[i, 'Avg']) # FIXED - this line is crucial!\n",
    "                scores.append(max(temp_scores))\n",
    "            # if tile is the only tile (first and last)\n",
    "            elif int(temp_df_hits.loc[i, 'Tile ID'].split('_')[-1]) == 1:\n",
    "                span_starts.append(temp_df_hits.loc[i, 'Tile Start'])\n",
    "                span_ends.append(temp_df_hits.loc[i, 'Tile End'])\n",
    "                scores.append(temp_df_hits.loc[i, 'Avg'])\n",
    "        # if tile is not last one\n",
    "        else:\n",
    "            # if not consecutive\n",
    "            if (int(temp_df_hits.loc[i, 'Tile ID'].split('_')[-1]) + 1) != (int(temp_df_hits.loc[i+1, 'Tile ID'].split('_')[-1])):\n",
    "                # if a temporary start of the domain has been defined, then store that start and the end of this tile\n",
    "                if ~np.isnan(temp_start):\n",
    "                    span_starts.append(temp_start)\n",
    "                    span_ends.append(temp_df_hits.loc[i, 'Tile End'])\n",
    "                    temp_scores.append(temp_df_hits.loc[i, 'Avg']) # FIXED - this line is crucial!\n",
    "                    scores.append(max(temp_scores))\n",
    "                    # reset temporary start to nan\n",
    "                    temp_start = np.nan\n",
    "                    temp_scores = []\n",
    "                # if the tile is at the N-terminus, then store the start/stop of the domain\n",
    "                elif int(temp_df_hits.loc[i, 'Tile ID'].split('_')[-1]) == 1:\n",
    "                    span_starts.append(temp_df_hits.loc[i, 'Tile Start'])\n",
    "                    span_ends.append(temp_df_hits.loc[i, 'Tile End'])\n",
    "                    scores.append(temp_df_hits.loc[i, 'Avg'])\n",
    "            # if consecutive\n",
    "            else:\n",
    "                # if start of domain is not yet defined\n",
    "                if np.isnan(temp_start):\n",
    "                    temp_start = temp_df_hits.loc[i, 'Tile Start']\n",
    "                    temp_scores.append(temp_df_hits.loc[i, 'Avg'])\n",
    "                # if tile is in the middle of a domain\n",
    "                else:\n",
    "                    temp_scores.append(temp_df_hits.loc[i, 'Avg'])\n",
    "                    continue\n",
    "    \n",
    "    return span_starts, span_ends, scores\n",
    "\n",
    "###############################\n",
    "        \n",
    "# this function makes tiling plots\n",
    "def plotDomain(gene, temp_df, protDF, domainDict, indivDir_path, save_root, figsize, dpi):\n",
    "    # subset for activation\n",
    "    temp_dfA = temp_df[temp_df['Screen']=='Activation']\n",
    "    temp_dfA = temp_dfA.drop_duplicates(subset='Tile ID')\n",
    "    temp_IDsA = list(temp_dfA['Oligo ID'])\n",
    "    temp_dfA.set_index('Oligo ID', inplace=True)\n",
    "    \n",
    "    # get activation points and error\n",
    "    xA = [(1 + (temp_dfA.loc[ID]['Tile Start'] + temp_dfA.loc[ID]['Tile End'] - 1)/2) for ID in temp_IDsA]\n",
    "    xerrA = [(temp_dfA.loc[ID]['Tile End'] - temp_dfA.loc[ID]['Tile Start'] + 1)/2 for ID in temp_IDsA]\n",
    "    yA = [temp_dfA.loc[ID]['Avg'] for ID in temp_IDsA]\n",
    "    yerrA = list(temp_dfA['Standard Error'])\n",
    "    \n",
    "    # get activation domain info\n",
    "    domain_startsA, domain_endsA, scoresA = domainLimits(temp_dfA)\n",
    "    print('Activation domain starts/stops/scores:', domain_startsA, domain_endsA, scoresA)\n",
    "\n",
    "    # subset for repression\n",
    "    temp_dfR = temp_df[temp_df['Screen']=='Repression']\n",
    "    temp_dfR = temp_dfR.drop_duplicates(subset='Tile ID')\n",
    "    temp_IDsR = list(temp_dfR['Oligo ID'])\n",
    "    temp_dfR.set_index('Oligo ID', inplace=True)\n",
    "    \n",
    "    # get repression points and error\n",
    "    xR = [(1 + (temp_dfR.loc[ID]['Tile Start'] + temp_dfR.loc[ID]['Tile End'] - 1)/2) for ID in temp_IDsR]\n",
    "    xerrR = [(temp_dfR.loc[ID]['Tile End'] - temp_dfR.loc[ID]['Tile Start'] + 1)/2 for ID in temp_IDsR]\n",
    "    yR = [temp_dfR.loc[ID]['Avg'] for ID in temp_IDsR]\n",
    "    yerrR = list(temp_dfR['Standard Error'])\n",
    "    \n",
    "    # get repression domain info\n",
    "    domain_startsR, domain_endsR, scoresR = domainLimits(temp_dfR)\n",
    "    print('Repression domain starts/stops/scores:', domain_startsR, domain_endsR, scoresR)\n",
    "\n",
    "    # create axis objects and plot tile centers vs log2(ON:OFF) scores\n",
    "    fig, ax = plt.subplots(figsize=figsize)\n",
    "    ax2 = ax.twinx()\n",
    "    colors = ['#23BEDB', '#D7AC00', '#DB3982']\n",
    "    markersize = 4\n",
    "    \n",
    "    # the following stores extended domain and strongest tile info in domainDict\n",
    "    # it also draws horizontal spans that highlight the strongest tile in each domain\n",
    "    # activators\n",
    "    if len(domain_startsA) != 0:\n",
    "        for s, e, c in zip(domain_startsA, domain_endsA, scoresA):\n",
    "            domA = temp_dfA[(temp_dfA['Tile Start']>=s) & (temp_dfA['Tile End']<=e)]\n",
    "            startindex = int(s)\n",
    "            endindex = int(e)\n",
    "            domseq = protDF.loc[gene, 'Representative Sequence'][(startindex - 1):endindex]\n",
    "            max_avg = max(list(domA['Avg']))\n",
    "            max_coords = [domA[domA['Avg']==max_avg]['Tile Start'][0], domA[domA['Avg']==max_avg]['Tile End'][0]]\n",
    "            max_seq = domA[domA['Avg']==max_avg]['Tile Sequence'][0]\n",
    "            max_tileID = domA[domA['Avg']==max_avg]['Tile ID'][0]\n",
    "            \n",
    "            to_append = [gene, max_tileID, startindex, endindex, len(domseq), domseq,\n",
    "                         max_coords[0], max_coords[1], max_seq, c, 'Activation']\n",
    "            for k, v in zip(domainDict.keys(), to_append):\n",
    "                domainDict[k].append(v)\n",
    "            \n",
    "            ax.axvspan(xmin=max_coords[0], xmax=max_coords[1], color='#f5dc79', alpha=0.4, lw=0.8, zorder=-200)\n",
    "    \n",
    "    # repressors\n",
    "    if len(domain_startsR) != 0:\n",
    "        for s, e, c in zip(domain_startsR, domain_endsR, scoresR):\n",
    "            domR = temp_dfR[(temp_dfR['Tile Start']>=s) & (temp_dfR['Tile End']<=e)]\n",
    "            startindex = int(s)\n",
    "            endindex = int(e)\n",
    "            domseq = protDF.loc[gene, 'Representative Sequence'][(startindex - 1):endindex]\n",
    "            max_avg = max(list(domR['Avg']))\n",
    "            max_coords = [domR[domR['Avg']==max_avg]['Tile Start'][0], domR[domR['Avg']==max_avg]['Tile End'][0]]\n",
    "            max_seq = domR[domR['Avg']==max_avg]['Tile Sequence'][0]\n",
    "            max_tileID = domR[domR['Avg']==max_avg]['Tile ID'][0]\n",
    "            \n",
    "            to_append = [gene, max_tileID, startindex, endindex, len(domseq), domseq,\n",
    "                         max_coords[0], max_coords[1], max_seq, c, 'Repression']\n",
    "            for k, v in zip(domainDict.keys(), to_append):\n",
    "                domainDict[k].append(v)\n",
    "            \n",
    "            ax.axvspan(xmin=max_coords[0], xmax=max_coords[1], color='#bbdaee', alpha=0.4, lw=0.8, zorder=-150)\n",
    "     \n",
    "    # plot tiles, with activators on y-axis 1 and repressors on y-axis 2\n",
    "    ax.scatter(xA, yA, c=colors[1], zorder=-100, s=markersize)\n",
    "    for xi, yi, exi, eyi in zip(xA, yA, xerrA, yerrA):\n",
    "        ax.errorbar(xi, yi, xerr=exi, yerr=eyi, color=colors[1], zorder=-100, lw=0.8)\n",
    "    \n",
    "    ax2.scatter(xR, yR, c=colors[0], zorder=-50, s=markersize)\n",
    "    for xi, yi, exi, eyi in zip(xR, yR, xerrR, yerrR):\n",
    "        ax2.errorbar(xi, yi, xerr=exi, yerr=eyi, color=colors[0], zorder=-50, lw=0.8)\n",
    "    \n",
    "    ax.set_title(gene_uID_list_dict1[gene], fontsize=8)\n",
    "    ax.set_xlabel('Position along Protein')\n",
    "    colorsTitle = ['#01a4c2', '#c29d09']\n",
    "    ax.set_ylabel('Act. log$_2$(ON:OFF)', color=colorsTitle[1], labelpad=0)\n",
    "    ax2.set_ylabel('Repr. log$_2$(OFF:ON)', color=colorsTitle[0], rotation=270, labelpad=10)\n",
    "    ax2.axhspan(ymin=minScoreA - 0.5, ymax=cutoffR_HHV, color='#FFFFFF', alpha=0.6, zorder=-15)\n",
    "    ax2.axhline(y=cutoffR_HHV, linestyle='--', color='#666666', lw=1, zorder=0)\n",
    "    ax.set_ylim(minScoreA - 0.5, maxScoreA + 0.5)\n",
    "    ax.yaxis.set_minor_locator(MultipleLocator(1))\n",
    "    ax.set_yticks([-5, 0, 5, 10])\n",
    "    ax2.set_ylim(minScoreR - 0.5, maxScoreR + 0.5)\n",
    "    ax2.yaxis.set_minor_locator(MultipleLocator(1))\n",
    "    ax.xaxis.set_minor_locator(AutoMinorLocator(5))\n",
    "    mpl_axes_aligner.align.yaxes(ax, cutoffA_HHV, ax2, cutoffR_HHV, 0.5)\n",
    "\n",
    "    plt.tight_layout()\n",
    "    \n",
    "    # save figure and close current plot\n",
    "    fig_path = os.path.join(indivDir_path, gene_uID_list_dict2[gene] + save_root)\n",
    "    # do this because you cannot save files with '|' in the name\n",
    "    fig_path = fig_path.replace('|', '-')\n",
    "    fig.savefig(fig_path, dpi=dpi)\n",
    "    plt.close()\n",
    "    \n",
    "    return domainDict\n",
    "\n",
    "###############################\n",
    "\n",
    "# this function batch produces tiling plots while simultaneously storing domain info\n",
    "def extractDomains(df, protDF, lib_fullName, indivDir_path, save_root, figsize, dpi):\n",
    "    if 'herpesvirus' not in lib_fullName:\n",
    "        df = df[df['Type']==lib_fullName]\n",
    "    else:\n",
    "        df = df[(df['Library'].str.contains('herpesvirus')) & (~df['Type'].str.contains('control'))]\n",
    "    \n",
    "    # get hits and edit to avoid regex characters\n",
    "    tileHits = list(df[df['Hit'] == 1]['Oligo ID'])\n",
    "    geneHits = []\n",
    "    for hit in tileHits:\n",
    "        if '|' in hit:\n",
    "            geneHits.append('_'.join(hit.split('_')[1:3]))\n",
    "        else:\n",
    "            geneHits.append(hit.split('_')[1])\n",
    "\n",
    "    geneHits = set(geneHits)\n",
    "    geneHitsLen = str(len(geneHits))\n",
    "    \n",
    "    # find domains and construct data frame with domain info\n",
    "    domainDict = {'Protein ID':[],\n",
    "                  'Tile ID':[],\n",
    "                  'Extended Start':[],\n",
    "                  'Extended End':[],\n",
    "                  'Extended Length':[],\n",
    "                  'Extended Sequence':[],\n",
    "                  'Max Start':[],\n",
    "                  'Max End':[],\n",
    "                  'Max Sequence':[],\n",
    "                  'Max Score':[],\n",
    "                  'Effect':[]}\n",
    "    incr = 0\n",
    "    for gene in geneHits:\n",
    "        incr += 1\n",
    "        # make temporary dataframe for each gene\n",
    "        # '|' character is read as an 'or' by regex, so don't include when searching\n",
    "        if '|' in gene:\n",
    "            genesearch = '.'.join(gene.split('|'))\n",
    "        else:\n",
    "            genesearch = gene\n",
    "        temp_df = df[df['Oligo ID'].str.contains(genesearch)]\n",
    "        temp_df = temp_df.sort_values(by='Oligo ID')\n",
    "        \n",
    "        # make a plot with enrichment scores for each gene containing hit tiles\n",
    "        print('Protein %s of %s: plotting log2(ON:OFF) for %s' % (str(incr), geneHitsLen, gene))\n",
    "        domainDict = plotDomain(gene, temp_df, protDF, domainDict, indivDir_path, save_root, figsize, dpi)\n",
    "\n",
    "    finalDF = pd.DataFrame(domainDict)\n",
    "    print('Finished')\n",
    "    \n",
    "    return finalDF\n",
    "\n",
    "print('Domain calling/plotting functions defined')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Tiling plots featured in Supplemental PDF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making directory called ../Individual Figure Panels/HHV_indivProteins_allTiles_150-dpi to store plots\n",
      "Protein 1 of 346: plotting log2(ON:OFF) for P30005\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [291.0] [430.0] [2.823023037]\n",
      "Protein 2 of 346: plotting log2(ON:OFF) for P52378\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 3 of 346: plotting log2(ON:OFF) for Q2HR92\n",
      "Activation domain starts/stops/scores: [271.0] [370.0] [5.048030371]\n",
      "Repression domain starts/stops/scores: [271.0] [370.0] [1.203760152]\n",
      "Protein 4 of 346: plotting log2(ON:OFF) for P16784\n",
      "Activation domain starts/stops/scores: [881.0] [970.0] [3.394143029]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 5 of 346: plotting log2(ON:OFF) for F5HIC6\n",
      "Activation domain starts/stops/scores: [191.0] [350.0] [9.020424959]\n",
      "Repression domain starts/stops/scores: [191.0] [340.0] [3.6357068339999996]\n",
      "Protein 6 of 346: plotting log2(ON:OFF) for Q2HR80\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [31.0] [120.0] [1.611413513]\n",
      "Protein 7 of 346: plotting log2(ON:OFF) for P30024\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 8 of 346: plotting log2(ON:OFF) for P52523\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [51.0] [160.0] [4.288637273]\n",
      "Protein 9 of 346: plotting log2(ON:OFF) for Q96896\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [311.0] [430.0] [1.79505454]\n",
      "Protein 10 of 346: plotting log2(ON:OFF) for P04292\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 11 of 346: plotting log2(ON:OFF) for P04296\n",
      "Activation domain starts/stops/scores: [1071.0] [1170.0] [6.187997021]\n",
      "Repression domain starts/stops/scores: [121.0, 1071.0] [230.0, 1160.0] [1.7067741669999998, 1.940906101]\n",
      "Protein 12 of 346: plotting log2(ON:OFF) for P04294\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 13 of 346: plotting log2(ON:OFF) for Q2HR89\n",
      "Activation domain starts/stops/scores: [1.0] [90.0] [4.3459168539999995]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 14 of 346: plotting log2(ON:OFF) for P52442\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [391.0] [477.0] [1.540892949]\n",
      "Protein 15 of 346: plotting log2(ON:OFF) for Q1HVG4\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [591.0] [700.0] [3.776137505]\n",
      "Protein 16 of 346: plotting log2(ON:OFF) for P09295\n",
      "Activation domain starts/stops/scores: [91.0] [180.0] [4.880847403]\n",
      "Repression domain starts/stops/scores: [91.0] [180.0] [1.35931511]\n",
      "Protein 17 of 346: plotting log2(ON:OFF) for P09268\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 18 of 346: plotting log2(ON:OFF) for P81780\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 19 of 346: plotting log2(ON:OFF) for P09293\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 20 of 346: plotting log2(ON:OFF) for P09708\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0] [80.0] [1.746627251]\n",
      "Protein 21 of 346: plotting log2(ON:OFF) for F5HB81\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [131.0, 361.0] [220.0, 460.0] [1.22924986, 1.349641026]\n",
      "Protein 22 of 346: plotting log2(ON:OFF) for P89451\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 23 of 346: plotting log2(ON:OFF) for P09310\n",
      "Activation domain starts/stops/scores: [1.0] [90.0] [6.8444821560000015]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 24 of 346: plotting log2(ON:OFF) for P06763\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [51.0] [170.0] [3.2128019489999997]\n",
      "Protein 25 of 346: plotting log2(ON:OFF) for P09263\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 26 of 346: plotting log2(ON:OFF) for Q2HR82\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [191.0] [286.0] [4.112703085]\n",
      "Protein 27 of 346: plotting log2(ON:OFF) for P16788\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 28 of 346: plotting log2(ON:OFF) for F5HEF2\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 29 of 346: plotting log2(ON:OFF) for P09276\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 30 of 346: plotting log2(ON:OFF) for P52531\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 31 of 346: plotting log2(ON:OFF) for Q07286\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 32 of 346: plotting log2(ON:OFF) for P52340\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 33 of 346: plotting log2(ON:OFF) for Q69091\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 34 of 346: plotting log2(ON:OFF) for Q9QJ27\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 35 of 346: plotting log2(ON:OFF) for P52466\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 36 of 346: plotting log2(ON:OFF) for F5HGP1\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 37 of 346: plotting log2(ON:OFF) for P33479\n",
      "Activation domain starts/stops/scores: [1.0] [80.0] [9.795911745]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 38 of 346: plotting log2(ON:OFF) for C0H677\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [291.0] [430.0] [4.712676983]\n",
      "Protein 39 of 346: plotting log2(ON:OFF) for Q2HR75\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 40 of 346: plotting log2(ON:OFF) for Q2HRC9\n",
      "Activation domain starts/stops/scores: [131.0] [230.0] [7.185906719]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 41 of 346: plotting log2(ON:OFF) for Q6SW82\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 42 of 346: plotting log2(ON:OFF) for P08393\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [331.0] [440.0] [1.259168628]\n",
      "Protein 43 of 346: plotting log2(ON:OFF) for P52359\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 44 of 346: plotting log2(ON:OFF) for P04295\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 45 of 346: plotting log2(ON:OFF) for P89436\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [71.0] [160.0] [0.994237322]\n",
      "Protein 46 of 346: plotting log2(ON:OFF) for Q4JQX9\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [851.0, 2341.0] [940.0, 2430.0] [1.060004144, 1.71977471]\n",
      "Protein 47 of 346: plotting log2(ON:OFF) for P24827\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 48 of 346: plotting log2(ON:OFF) for P28275\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 49 of 346: plotting log2(ON:OFF) for P06492\n",
      "Activation domain starts/stops/scores: [1.0, 371.0] [80.0, 490.0] [1.801022726, 9.339976975]\n",
      "Repression domain starts/stops/scores: [71.0] [160.0] [1.5324096980000002]\n",
      "Protein 50 of 346: plotting log2(ON:OFF) for P16721\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [121.0] [210.0] [1.105516004]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Protein 51 of 346: plotting log2(ON:OFF) for Q4JQS7\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 52 of 346: plotting log2(ON:OFF) for Q77Q36\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 53 of 346: plotting log2(ON:OFF) for P0C6Z0\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 54 of 346: plotting log2(ON:OFF) for Q1HVF7\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [391.0] [480.0] [1.07559941]\n",
      "Protein 55 of 346: plotting log2(ON:OFF) for P0C6Z8\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 56 of 346: plotting log2(ON:OFF) for P09249\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 57 of 346: plotting log2(ON:OFF) for P09728\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [21.0] [140.0] [1.542364387]\n",
      "Protein 58 of 346: plotting log2(ON:OFF) for P24435\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 59 of 346: plotting log2(ON:OFF) for P16770\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0] [110.0] [1.721290631]\n",
      "Protein 60 of 346: plotting log2(ON:OFF) for P03234\n",
      "Activation domain starts/stops/scores: [291.0] [390.0] [4.781914256]\n",
      "Repression domain starts/stops/scores: [311.0] [410.0] [1.523000669]\n",
      "Protein 61 of 346: plotting log2(ON:OFF) for Q3KSV2\n",
      "Activation domain starts/stops/scores: [351.0] [451.0] [8.846633515]\n",
      "Repression domain starts/stops/scores: [31.0] [140.0] [2.3599193869999997]\n",
      "Protein 62 of 346: plotting log2(ON:OFF) for P09292\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 63 of 346: plotting log2(ON:OFF) for D5LX59\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 64 of 346: plotting log2(ON:OFF) for P24444\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 65 of 346: plotting log2(ON:OFF) for P52457\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 66 of 346: plotting log2(ON:OFF) for P03186\n",
      "Activation domain starts/stops/scores: [1221.0, 2001.0, 2031.0, 3021.0] [1310.0, 2090.0, 2120.0, 3120.0] [5.873405066, 3.4710367630000003, 2.527761644, 6.361502272999999]\n",
      "Repression domain starts/stops/scores: [241.0, 1071.0, 1991.0] [330.0, 1160.0, 2080.0] [0.8384802020000001, 1.474727444, 1.208991172]\n",
      "Protein 67 of 346: plotting log2(ON:OFF) for Q69535\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [751.0] [840.0] [1.013113242]\n",
      "Protein 68 of 346: plotting log2(ON:OFF) for P28284\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [341.0, 381.0, 591.0, 651.0] [440.0, 480.0, 690.0, 750.0] [1.401558714, 0.8953349340000001, 1.197690075, 2.9740637519999997]\n",
      "Protein 69 of 346: plotting log2(ON:OFF) for P10220\n",
      "Activation domain starts/stops/scores: [1811.0] [1900.0] [6.085847517]\n",
      "Repression domain starts/stops/scores: [1701.0] [1810.0] [2.147895103]\n",
      "Protein 70 of 346: plotting log2(ON:OFF) for P09299\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 71 of 346: plotting log2(ON:OFF) for P0C717\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [41.0] [130.0] [1.974962345]\n",
      "Protein 72 of 346: plotting log2(ON:OFF) for Q2HRD2\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [321.0] [420.0] [1.270460344]\n",
      "Protein 73 of 346: plotting log2(ON:OFF) for Q2HR71\n",
      "Activation domain starts/stops/scores: [141.0, 251.0] [280.0, 410.0] [7.41351146, 10.39664028]\n",
      "Repression domain starts/stops/scores: [261.0] [390.0] [2.8553556589999998]\n",
      "Protein 74 of 346: plotting log2(ON:OFF) for P06434\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0] [80.0] [4.064971451]\n",
      "Protein 75 of 346: plotting log2(ON:OFF) for Q69553\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 76 of 346: plotting log2(ON:OFF) for Q8AZJ5\n",
      "Activation domain starts/stops/scores: [291.0] [380.0] [4.0895191639999995]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 77 of 346: plotting log2(ON:OFF) for Q2HR83\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 78 of 346: plotting log2(ON:OFF) for P52362\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 79 of 346: plotting log2(ON:OFF) for P09250\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 80 of 346: plotting log2(ON:OFF) for P89469\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [181.0] [280.0] [1.583543362]\n",
      "Protein 81 of 346: plotting log2(ON:OFF) for F5HAI6\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 82 of 346: plotting log2(ON:OFF) for P52348\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 83 of 346: plotting log2(ON:OFF) for P08355\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [231.0] [370.0] [2.190530162]\n",
      "Protein 84 of 346: plotting log2(ON:OFF) for P09256\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [131.0] [230.0] [1.000856113]\n",
      "Protein 85 of 346: plotting log2(ON:OFF) for P36342\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 86 of 346: plotting log2(ON:OFF) for P09711\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [51.0] [150.0] [1.6840317180000002]\n",
      "Protein 87 of 346: plotting log2(ON:OFF) for P89462\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 88 of 346: plotting log2(ON:OFF) for Q2HR64\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1641.0] [1730.0] [0.942203984]\n",
      "Protein 89 of 346: plotting log2(ON:OFF) for Q2HRA1\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0] [80.0] [1.028519116]\n",
      "Protein 90 of 346: plotting log2(ON:OFF) for Q85228\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 91 of 346: plotting log2(ON:OFF) for P16818\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0] [80.0] [0.856300671]\n",
      "Protein 92 of 346: plotting log2(ON:OFF) for P52537\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 93 of 346: plotting log2(ON:OFF) for Q9QJ12\n",
      "Activation domain starts/stops/scores: [1.0] [80.0] [7.595415155]\n",
      "Repression domain starts/stops/scores: [1.0, 571.0] [80.0, 670.0] [2.6255174219999997, 2.53374805]\n",
      "Protein 94 of 346: plotting log2(ON:OFF) for P17149\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [441.0] [530.0] [1.229989952]\n",
      "Protein 95 of 346: plotting log2(ON:OFF) for P04485\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [71.0] [160.0] [1.75800178]\n",
      "Protein 96 of 346: plotting log2(ON:OFF) for F5HAQ7\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 97 of 346: plotting log2(ON:OFF) for P52342\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 98 of 346: plotting log2(ON:OFF) for P16769\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [11.0] [130.0] [2.147824838]\n",
      "Protein 99 of 346: plotting log2(ON:OFF) for P09265\n",
      "Activation domain starts/stops/scores: [1.0, 51.0] [100.0, 170.0] [7.863219076, 6.875394807999999]\n",
      "Repression domain starts/stops/scores: [61.0] [150.0] [1.0615146020000001]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Protein 100 of 346: plotting log2(ON:OFF) for P52385\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 101 of 346: plotting log2(ON:OFF) for P06437\n",
      "Activation domain starts/stops/scores: [1.0] [80.0] [1.971108862]\n",
      "Repression domain starts/stops/scores: [81.0] [180.0] [2.304961744]\n",
      "Protein 102 of 346: plotting log2(ON:OFF) for P52353\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 103 of 346: plotting log2(ON:OFF) for P03211\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [391.0, 421.0] [480.0, 510.0] [1.293116343, 1.793976895]\n",
      "Protein 104 of 346: plotting log2(ON:OFF) for P06725\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 105 of 346: plotting log2(ON:OFF) for P24443\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 106 of 346: plotting log2(ON:OFF) for P52538\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 107 of 346: plotting log2(ON:OFF) for P0C6N0\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 108 of 346: plotting log2(ON:OFF) for F5HIM6\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 109 of 346: plotting log2(ON:OFF) for Q6SW62\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [11.0, 361.0] [100.0, 450.0] [1.509469876, 1.054518972]\n",
      "Protein 110 of 346: plotting log2(ON:OFF) for P52474\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 111 of 346: plotting log2(ON:OFF) for P03235\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [81.0] [170.0] [0.995246289]\n",
      "Protein 112 of 346: plotting log2(ON:OFF) for P09258\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 113 of 346: plotting log2(ON:OFF) for P16827\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [751.0] [850.0] [1.0826943759999998]\n",
      "Protein 114 of 346: plotting log2(ON:OFF) for P52450\n",
      "Activation domain starts/stops/scores: [521.0] [610.0] [4.884718996]\n",
      "Repression domain starts/stops/scores: [511.0] [610.0] [1.8201193130000002]\n",
      "Protein 115 of 346: plotting log2(ON:OFF) for Q9J3N6\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [301.0] [400.0] [1.881408744]\n",
      "Protein 116 of 346: plotting log2(ON:OFF) for P16766\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 117 of 346: plotting log2(ON:OFF) for Q3KSS4\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [391.0, 421.0] [480.0, 510.0] [0.993197346, 1.388270039]\n",
      "Protein 118 of 346: plotting log2(ON:OFF) for P52549\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [271.0] [400.0] [2.2176235930000003]\n",
      "Protein 119 of 346: plotting log2(ON:OFF) for P09855\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 120 of 346: plotting log2(ON:OFF) for P03220\n",
      "Activation domain starts/stops/scores: [41.0] [130.0] [5.37480709]\n",
      "Repression domain starts/stops/scores: [41.0] [130.0] [1.424348842]\n",
      "Protein 121 of 346: plotting log2(ON:OFF) for P03198\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [391.0] [480.0] [0.93801277]\n",
      "Protein 122 of 346: plotting log2(ON:OFF) for P09731\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 123 of 346: plotting log2(ON:OFF) for P89459\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1781.0, 2391.0] [1870.0, 2480.0] [1.504853837, 1.563081845]\n",
      "Protein 124 of 346: plotting log2(ON:OFF) for P89431\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 125 of 346: plotting log2(ON:OFF) for Q9J3N2\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [21.0] [140.0] [2.184044889]\n",
      "Protein 126 of 346: plotting log2(ON:OFF) for Q9PWX1\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [291.0] [390.0] [2.195494194]\n",
      "Protein 127 of 346: plotting log2(ON:OFF) for P0CK57\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0, 31.0] [90.0, 120.0] [1.899364328, 1.6289693619999999]\n",
      "Protein 128 of 346: plotting log2(ON:OFF) for P06024\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [61.0, 311.0] [160.0, 400.0] [2.1422598369999997, 0.8435712940000001]\n",
      "Protein 129 of 346: plotting log2(ON:OFF) for Q9QJ45\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 130 of 346: plotting log2(ON:OFF) for P52534\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [11.0] [150.0] [4.900162611]\n",
      "Protein 131 of 346: plotting log2(ON:OFF) for F5HF23\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0] [95.0] [3.032371965]\n",
      "Protein 132 of 346: plotting log2(ON:OFF) for P24382\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 133 of 346: plotting log2(ON:OFF) for P09248\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 134 of 346: plotting log2(ON:OFF) for P03169\n",
      "Activation domain starts/stops/scores: [1.0] [90.0] [7.629990567]\n",
      "Repression domain starts/stops/scores: [401.0] [491.0] [1.87991627]\n",
      "Protein 135 of 346: plotting log2(ON:OFF) for P09271\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 136 of 346: plotting log2(ON:OFF) for Q69549\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 137 of 346: plotting log2(ON:OFF) for F5HGB6\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [361.0] [460.0] [3.236877063]\n",
      "Protein 138 of 346: plotting log2(ON:OFF) for Q69555\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 139 of 346: plotting log2(ON:OFF) for Q9J3N3\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 140 of 346: plotting log2(ON:OFF) for F5HDE4\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [111.0, 171.0, 261.0] [200.0, 290.0, 407.0] [0.7454717759999999, 1.78525774, 2.1609864659999998]\n",
      "Protein 141 of 346: plotting log2(ON:OFF) for Q6SW55\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 142 of 346: plotting log2(ON:OFF) for P52448\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [381.0] [470.0] [1.793434331]\n",
      "Protein 143 of 346: plotting log2(ON:OFF) for F5HIN0\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 144 of 346: plotting log2(ON:OFF) for Q69140\n",
      "Activation domain starts/stops/scores: [791.0] [880.0] [2.7623029089999998]\n",
      "Repression domain starts/stops/scores: [441.0] [550.0] [1.532939417]\n",
      "Protein 145 of 346: plotting log2(ON:OFF) for P89429\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [381.0] [480.0] [1.095215738]\n",
      "Protein 146 of 346: plotting log2(ON:OFF) for P06489\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 147 of 346: plotting log2(ON:OFF) for F5HEU7\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 148 of 346: plotting log2(ON:OFF) for P36320\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [271.0] [370.0] [2.47914102]\n",
      "Protein 149 of 346: plotting log2(ON:OFF) for F5HFD2\n",
      "Activation domain starts/stops/scores: [] [] []\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 150 of 346: plotting log2(ON:OFF) for P03226\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 151 of 346: plotting log2(ON:OFF) for Q6SW34\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 152 of 346: plotting log2(ON:OFF) for P10231\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 153 of 346: plotting log2(ON:OFF) for P28281\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [101.0] [186.0] [1.0795818590000001]\n",
      "Protein 154 of 346: plotting log2(ON:OFF) for P52437\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [901.0] [1000.0] [1.4395021209999999]\n",
      "Protein 155 of 346: plotting log2(ON:OFF) for P03222\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [401.0] [507.0] [3.142314005]\n",
      "Protein 156 of 346: plotting log2(ON:OFF) for P07645\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [281.0] [370.0] [3.1303119560000003]\n",
      "Protein 157 of 346: plotting log2(ON:OFF) for F5HCV3\n",
      "Activation domain starts/stops/scores: [531.0] [680.0] [10.91612574]\n",
      "Repression domain starts/stops/scores: [241.0, 551.0] [330.0, 680.0] [1.2517355559999999, 2.1791663740000002]\n",
      "Protein 158 of 346: plotting log2(ON:OFF) for P10205\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 159 of 346: plotting log2(ON:OFF) for F5HDB7\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 160 of 346: plotting log2(ON:OFF) for P09714\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0] [80.0] [1.4544308000000001]\n",
      "Protein 161 of 346: plotting log2(ON:OFF) for P10192\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [111.0, 181.0] [200.0, 270.0] [0.959120145, 1.808983604]\n",
      "Protein 162 of 346: plotting log2(ON:OFF) for F5HF68\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [151.0] [240.0] [1.042625909]\n",
      "Protein 163 of 346: plotting log2(ON:OFF) for Q6SVX2\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [401.0] [510.0] [1.460019265]\n",
      "Protein 164 of 346: plotting log2(ON:OFF) for P10189\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [571.0] [670.0] [0.98591857]\n",
      "Protein 165 of 346: plotting log2(ON:OFF) for Q9QR70\n",
      "Activation domain starts/stops/scores: [121.0] [210.0] [3.580613045]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 166 of 346: plotting log2(ON:OFF) for Q9QJ16\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 167 of 346: plotting log2(ON:OFF) for P10234\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 168 of 346: plotting log2(ON:OFF) for P03227\n",
      "Activation domain starts/stops/scores: [81.0] [170.0] [3.48859819]\n",
      "Repression domain starts/stops/scores: [91.0] [180.0] [1.9345429930000002]\n",
      "Protein 169 of 346: plotting log2(ON:OFF) for P30816\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 170 of 346: plotting log2(ON:OFF) for F5HGI9\n",
      "Activation domain starts/stops/scores: [21.0] [110.0] [3.10417078]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 171 of 346: plotting log2(ON:OFF) for Q2HRA7\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 172 of 346: plotting log2(ON:OFF) for P89439\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [131.0] [240.0] [2.465321184]\n",
      "Protein 173 of 346: plotting log2(ON:OFF) for P17147\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [481.0] [570.0] [0.8384158290000001]\n",
      "Protein 174 of 346: plotting log2(ON:OFF) for P03209\n",
      "Activation domain starts/stops/scores: [511.0] [605.0] [8.585619295]\n",
      "Repression domain starts/stops/scores: [521.0] [605.0] [3.0539706819999997]\n",
      "Protein 175 of 346: plotting log2(ON:OFF) for Q2HRD3\n",
      "Activation domain starts/stops/scores: [241.0, 1051.0] [330.0, 1132.0] [3.483817563, 5.6019440110000005]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 176 of 346: plotting log2(ON:OFF) for Q69022\n",
      "Activation domain starts/stops/scores: [351.0] [450.0] [7.313721592]\n",
      "Repression domain starts/stops/scores: [51.0] [150.0] [1.967592295]\n",
      "Protein 177 of 346: plotting log2(ON:OFF) for P52521\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 178 of 346: plotting log2(ON:OFF) for Q6SW48\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [81.0] [170.0] [0.9892663629999999]\n",
      "Protein 179 of 346: plotting log2(ON:OFF) for Q89882\n",
      "Activation domain starts/stops/scores: [1.0] [80.0] [8.290824561]\n",
      "Repression domain starts/stops/scores: [1.0] [80.0] [2.5707424430000003]\n",
      "Protein 180 of 346: plotting log2(ON:OFF) for F5HC79\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 181 of 346: plotting log2(ON:OFF) for P09273\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 182 of 346: plotting log2(ON:OFF) for Q00705\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 183 of 346: plotting log2(ON:OFF) for P09695\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [411.0] [500.0] [1.732126439]\n",
      "Protein 184 of 346: plotting log2(ON:OFF) for P36702\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 185 of 346: plotting log2(ON:OFF) for P03203\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [611.0] [700.0] [3.995393488]\n",
      "Protein 186 of 346: plotting log2(ON:OFF) for P28279\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 187 of 346: plotting log2(ON:OFF) for P13287\n",
      "Activation domain starts/stops/scores: [1.0] [80.0] [1.634497607]\n",
      "Repression domain starts/stops/scores: [281.0] [370.0] [1.302479043]\n",
      "Protein 188 of 346: plotting log2(ON:OFF) for P09715\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [411.0] [500.0] [1.732126439]\n",
      "Protein 189 of 346: plotting log2(ON:OFF) for Q2HR67\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [501.0] [610.0] [1.379308349]\n",
      "Protein 190 of 346: plotting log2(ON:OFF) for P52339\n",
      "Activation domain starts/stops/scores: [1031.0] [1131.0] [5.916692589]\n",
      "Repression domain starts/stops/scores: [1051.0] [1131.0] [1.345623126]\n",
      "Protein 191 of 346: plotting log2(ON:OFF) for P10190\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 192 of 346: plotting log2(ON:OFF) for P16731\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 193 of 346: plotting log2(ON:OFF) for P52525\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 194 of 346: plotting log2(ON:OFF) for P52445\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 195 of 346: plotting log2(ON:OFF) for P68336\n",
      "Activation domain starts/stops/scores: [11.0, 41.0, 361.0] [100.0, 160.0, 490.0] [2.90517328, 7.581438717999999, 9.301965747]\n",
      "Repression domain starts/stops/scores: [61.0] [160.0] [1.7507207790000001]\n",
      "Protein 196 of 346: plotting log2(ON:OFF) for P25215\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 197 of 346: plotting log2(ON:OFF) for P30118\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 198 of 346: plotting log2(ON:OFF) for Q6SW65\n",
      "Activation domain starts/stops/scores: [] [] []\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 199 of 346: plotting log2(ON:OFF) for P13158\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 200 of 346: plotting log2(ON:OFF) for P03204\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [441.0] [570.0] [3.300042972]\n",
      "Protein 201 of 346: plotting log2(ON:OFF) for P09282\n",
      "Activation domain starts/stops/scores: [1.0] [100.0] [1.5582638640000002]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 202 of 346: plotting log2(ON:OFF) for P16782\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 203 of 346: plotting log2(ON:OFF) for P89466\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 204 of 346: plotting log2(ON:OFF) for P52455\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 205 of 346: plotting log2(ON:OFF) for P08560\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [71.0] [170.0] [4.004786384]\n",
      "Protein 206 of 346: plotting log2(ON:OFF) for P52533\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [21.0] [150.0] [3.526194255]\n",
      "Protein 207 of 346: plotting log2(ON:OFF) for P24433\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [301.0, 381.0, 421.0] [390.0, 470.0, 510.0] [1.6973664130000001, 2.4002734919999997, 1.021886552]\n",
      "Protein 208 of 346: plotting log2(ON:OFF) for P52462\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 209 of 346: plotting log2(ON:OFF) for P29129\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0, 51.0] [90.0, 140.0] [3.610011858, 3.966988438]\n",
      "Protein 210 of 346: plotting log2(ON:OFF) for Q6SWD5\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 211 of 346: plotting log2(ON:OFF) for P16729\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1051.0] [1140.0] [1.074047518]\n",
      "Protein 212 of 346: plotting log2(ON:OFF) for P03189\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 213 of 346: plotting log2(ON:OFF) for P52440\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 214 of 346: plotting log2(ON:OFF) for P09251\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 215 of 346: plotting log2(ON:OFF) for P89440\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 216 of 346: plotting log2(ON:OFF) for P24434\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 217 of 346: plotting log2(ON:OFF) for P08318\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 218 of 346: plotting log2(ON:OFF) for P10185\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 219 of 346: plotting log2(ON:OFF) for P0DOE0\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 220 of 346: plotting log2(ON:OFF) for P16739\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 221 of 346: plotting log2(ON:OFF) for P89474\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [81.0] [180.0] [1.250414433]\n",
      "Protein 222 of 346: plotting log2(ON:OFF) for Q4JQT7\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 223 of 346: plotting log2(ON:OFF) for P24994\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [221.0] [310.0] [0.950703174]\n",
      "Protein 224 of 346: plotting log2(ON:OFF) for P03206\n",
      "Activation domain starts/stops/scores: [21.0] [110.0] [3.450202627]\n",
      "Repression domain starts/stops/scores: [1.0, 141.0] [100.0, 230.0] [2.644107679, 2.1896522259999998]\n",
      "Protein 225 of 346: plotting log2(ON:OFF) for Q9PX32\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [11.0] [100.0] [3.9271620539999996]\n",
      "Protein 226 of 346: plotting log2(ON:OFF) for P89467\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 227 of 346: plotting log2(ON:OFF) for Q00701\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 228 of 346: plotting log2(ON:OFF) for F5HGH5\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0] [100.0] [3.256558909]\n",
      "Protein 229 of 346: plotting log2(ON:OFF) for P52540\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 230 of 346: plotting log2(ON:OFF) for P10236\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [501.0] [590.0] [1.336247451]\n",
      "Protein 231 of 346: plotting log2(ON:OFF) for Q69138\n",
      "Activation domain starts/stops/scores: [681.0] [830.0] [6.692531606]\n",
      "Repression domain starts/stops/scores: [31.0, 771.0] [130.0, 910.0] [4.98669171, 2.2187539309999997]\n",
      "Protein 232 of 346: plotting log2(ON:OFF) for P89460\n",
      "Activation domain starts/stops/scores: [951.0] [1040.0] [5.6778933610000015]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 233 of 346: plotting log2(ON:OFF) for P14379\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [71.0] [180.0] [1.8259900690000002]\n",
      "Protein 234 of 346: plotting log2(ON:OFF) for P89455\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 235 of 346: plotting log2(ON:OFF) for P03200\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [101.0, 151.0] [210.0, 270.0] [1.35403772, 1.326232221]\n",
      "Protein 236 of 346: plotting log2(ON:OFF) for P12888\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 237 of 346: plotting log2(ON:OFF) for P12978\n",
      "Activation domain starts/stops/scores: [381.0] [487.0] [8.87737418]\n",
      "Repression domain starts/stops/scores: [71.0] [180.0] [4.309964462]\n",
      "Protein 238 of 346: plotting log2(ON:OFF) for P06477\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 239 of 346: plotting log2(ON:OFF) for P08546\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 240 of 346: plotting log2(ON:OFF) for F5HB39\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 241 of 346: plotting log2(ON:OFF) for Q69577\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [61.0] [143.0] [1.061312821]\n",
      "Protein 242 of 346: plotting log2(ON:OFF) for Q89730\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [51.0, 301.0] [150.0, 390.0] [1.886360229, 2.954238147]\n",
      "Protein 243 of 346: plotting log2(ON:OFF) for P16761\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 244 of 346: plotting log2(ON:OFF) for F5HAY6\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 245 of 346: plotting log2(ON:OFF) for Q1HVD3\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 246 of 346: plotting log2(ON:OFF) for P09260\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [631.0] [720.0] [1.080503194]\n",
      "Protein 247 of 346: plotting log2(ON:OFF) for P11870\n",
      "Activation domain starts/stops/scores: [1061.0] [1150.0] [7.6475899589999985]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 248 of 346: plotting log2(ON:OFF) for P03224\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0, 41.0] [100.0, 130.0] [1.5613038080000001, 1.092336935]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Protein 249 of 346: plotting log2(ON:OFF) for P89437\n",
      "Activation domain starts/stops/scores: [81.0] [170.0] [2.778054618]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 250 of 346: plotting log2(ON:OFF) for Q9QJ19\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 251 of 346: plotting log2(ON:OFF) for P16793\n",
      "Activation domain starts/stops/scores: [51.0] [140.0] [5.275112335]\n",
      "Repression domain starts/stops/scores: [41.0] [130.0] [1.76000181]\n",
      "Protein 252 of 346: plotting log2(ON:OFF) for Q9WT45\n",
      "Activation domain starts/stops/scores: [191.0] [300.0] [3.749427119]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 253 of 346: plotting log2(ON:OFF) for Q9QJ43\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 254 of 346: plotting log2(ON:OFF) for P06491\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [251.0, 741.0, 891.0] [370.0, 830.0, 1010.0] [0.956572735, 1.5456892469999999, 1.8693847]\n",
      "Protein 255 of 346: plotting log2(ON:OFF) for P68321\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 256 of 346: plotting log2(ON:OFF) for Q9J3M8\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [381.0] [470.0] [1.784128269]\n",
      "Protein 257 of 346: plotting log2(ON:OFF) for Q69567\n",
      "Activation domain starts/stops/scores: [1.0] [110.0] [8.05539343]\n",
      "Repression domain starts/stops/scores: [611.0, 651.0, 711.0, 761.0] [710.0, 740.0, 820.0, 850.0] [1.747800395, 2.800741285, 1.313447903, 0.774701083]\n",
      "Protein 258 of 346: plotting log2(ON:OFF) for P0CK48\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 259 of 346: plotting log2(ON:OFF) for Q04360\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 260 of 346: plotting log2(ON:OFF) for Q2HRD4\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [121.0] [230.0] [1.831054054]\n",
      "Protein 261 of 346: plotting log2(ON:OFF) for F5H8Y5\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 262 of 346: plotting log2(ON:OFF) for P16799\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [131.0] [220.0] [1.7507572009999999]\n",
      "Protein 263 of 346: plotting log2(ON:OFF) for Q2HRB6\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [251.0] [340.0] [1.076295918]\n",
      "Protein 264 of 346: plotting log2(ON:OFF) for P16783\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0] [80.0] [1.979593748]\n",
      "Protein 265 of 346: plotting log2(ON:OFF) for P52379\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 266 of 346: plotting log2(ON:OFF) for Q2HR78\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 267 of 346: plotting log2(ON:OFF) for Q9QJ55\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0] [80.0] [0.7750615129999999]\n",
      "Protein 268 of 346: plotting log2(ON:OFF) for P10238\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 269 of 346: plotting log2(ON:OFF) for P09245\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [111.0] [200.0] [1.938959018]\n",
      "Protein 270 of 346: plotting log2(ON:OFF) for P09257\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 271 of 346: plotting log2(ON:OFF) for P36384\n",
      "Activation domain starts/stops/scores: [1061.0] [1170.0] [5.99465074]\n",
      "Repression domain starts/stops/scores: [1071.0] [1160.0] [1.965282882]\n",
      "Protein 272 of 346: plotting log2(ON:OFF) for P09286\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0, 311.0] [80.0, 410.0] [1.3948445269999998, 1.918880334]\n",
      "Protein 273 of 346: plotting log2(ON:OFF) for P24381\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [31.0, 211.0] [130.0, 300.0] [0.835261313, 1.1219398409999999]\n",
      "Protein 274 of 346: plotting log2(ON:OFF) for P16719\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [41.0] [130.0] [1.854549777]\n",
      "Protein 275 of 346: plotting log2(ON:OFF) for Q4JQU7\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 276 of 346: plotting log2(ON:OFF) for P52529\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [251.0, 311.0] [370.0, 430.0] [1.8651161230000002, 2.2100271559999998]\n",
      "Protein 277 of 346: plotting log2(ON:OFF) for Q9J3N5\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 278 of 346: plotting log2(ON:OFF) for P09246\n",
      "Activation domain starts/stops/scores: [1071.0] [1160.0] [5.9870291510000015]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 279 of 346: plotting log2(ON:OFF) for P89453\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 280 of 346: plotting log2(ON:OFF) for F5HGU6\n",
      "Activation domain starts/stops/scores: [181.0] [270.0] [2.349929838]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 281 of 346: plotting log2(ON:OFF) for Q6SW67\n",
      "Activation domain starts/stops/scores: [491.0] [580.0] [3.507352553]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 282 of 346: plotting log2(ON:OFF) for P06473\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 283 of 346: plotting log2(ON:OFF) for P0CK50\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 284 of 346: plotting log2(ON:OFF) for P09264\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [491.0, 531.0] [590.0, 630.0] [0.928396837, 1.7383200669999999]\n",
      "Protein 285 of 346: plotting log2(ON:OFF) for P09722\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 286 of 346: plotting log2(ON:OFF) for P10200\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 287 of 346: plotting log2(ON:OFF) for P16810\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 288 of 346: plotting log2(ON:OFF) for P0C726\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 289 of 346: plotting log2(ON:OFF) for Q77PU6\n",
      "Activation domain starts/stops/scores: [1.0] [130.0] [10.09150184]\n",
      "Repression domain starts/stops/scores: [781.0, 851.0] [870.0, 1000.0] [3.314365365, 1.923917505]\n",
      "Protein 290 of 346: plotting log2(ON:OFF) for P09713\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 291 of 346: plotting log2(ON:OFF) for P19893\n",
      "Activation domain starts/stops/scores: [1.0] [120.0] [9.268958443999999]\n",
      "Repression domain starts/stops/scores: [151.0] [260.0] [4.2485207439999995]\n",
      "Protein 292 of 346: plotting log2(ON:OFF) for P09277\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [771.0] [870.0] [2.154436017]\n",
      "Protein 293 of 346: plotting log2(ON:OFF) for P52465\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 294 of 346: plotting log2(ON:OFF) for Q6SWD3\n",
      "Activation domain starts/stops/scores: [31.0] [111.0] [2.484493181]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 295 of 346: plotting log2(ON:OFF) for P30003\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [601.0] [710.0] [1.44774726]\n",
      "Protein 296 of 346: plotting log2(ON:OFF) for P10226\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 297 of 346: plotting log2(ON:OFF) for P03179\n",
      "Activation domain starts/stops/scores: [511.0] [660.0] [7.322433823]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Repression domain starts/stops/scores: [531.0, 1211.0] [640.0, 1300.0] [2.404403908, 2.15999445]\n",
      "Protein 298 of 346: plotting log2(ON:OFF) for P52468\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [421.0] [510.0] [1.133301163]\n",
      "Protein 299 of 346: plotting log2(ON:OFF) for P04291\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 300 of 346: plotting log2(ON:OFF) for Q6SWC8\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [21.0] [150.0] [1.487569945]\n",
      "Protein 301 of 346: plotting log2(ON:OFF) for P03214\n",
      "Activation domain starts/stops/scores: [1.0] [80.0] [1.091819145]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 302 of 346: plotting log2(ON:OFF) for Q69545\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [81.0] [170.0] [1.213473399]\n",
      "Protein 303 of 346: plotting log2(ON:OFF) for P52351\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [371.0] [480.0] [2.443144863]\n",
      "Protein 304 of 346: plotting log2(ON:OFF) for P08354\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 305 of 346: plotting log2(ON:OFF) for Q6SWB8\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 306 of 346: plotting log2(ON:OFF) for P03210\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 307 of 346: plotting log2(ON:OFF) for P10221\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 308 of 346: plotting log2(ON:OFF) for P10230\n",
      "Activation domain starts/stops/scores: [591.0] [690.0] [4.294716359]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 309 of 346: plotting log2(ON:OFF) for Q1XBW5\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 310 of 346: plotting log2(ON:OFF) for Q2HR73\n",
      "Activation domain starts/stops/scores: [301.0, 601.0] [390.0, 720.0] [6.185088795, 8.970479692]\n",
      "Repression domain starts/stops/scores: [161.0, 291.0, 491.0] [260.0, 380.0, 630.0] [1.186172642, 1.544401151, 2.45119432]\n",
      "Protein 311 of 346: plotting log2(ON:OFF) for Q4JQW9\n",
      "Activation domain starts/stops/scores: [701.0] [810.0] [4.935263797]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 312 of 346: plotting log2(ON:OFF) for F5HAZ3\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 313 of 346: plotting log2(ON:OFF) for P52347\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [501.0] [590.0] [0.851634022]\n",
      "Protein 314 of 346: plotting log2(ON:OFF) for Q3KSQ2\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 315 of 346: plotting log2(ON:OFF) for Q9QR71\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [231.0] [320.0] [3.094764454]\n",
      "Protein 316 of 346: plotting log2(ON:OFF) for P68324\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [581.0] [670.0] [1.3758984630000002]\n",
      "Protein 317 of 346: plotting log2(ON:OFF) for Q89660\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 318 of 346: plotting log2(ON:OFF) for P52464\n",
      "Activation domain starts/stops/scores: [1.0] [80.0] [3.589329605]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 319 of 346: plotting log2(ON:OFF) for P10228\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [171.0] [260.0] [2.067083378]\n",
      "Protein 320 of 346: plotting log2(ON:OFF) for P08392\n",
      "Activation domain starts/stops/scores: [21.0] [150.0] [5.4646004]\n",
      "Repression domain starts/stops/scores: [301.0] [390.0] [1.332132173]\n",
      "Protein 321 of 346: plotting log2(ON:OFF) for P13290\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 322 of 346: plotting log2(ON:OFF) for P90493\n",
      "Activation domain starts/stops/scores: [371.0] [470.0] [5.097037379]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 323 of 346: plotting log2(ON:OFF) for Q9QJ11\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 324 of 346: plotting log2(ON:OFF) for P10201\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [41.0] [140.0] [2.1927194919999997]\n",
      "Protein 325 of 346: plotting log2(ON:OFF) for P52438\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 326 of 346: plotting log2(ON:OFF) for P03218\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [31.0] [120.0] [2.305562766]\n",
      "Protein 327 of 346: plotting log2(ON:OFF) for P08543\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 328 of 346: plotting log2(ON:OFF) for P04413\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [271.0] [370.0] [2.378622272]\n",
      "Protein 329 of 346: plotting log2(ON:OFF) for P10210\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 330 of 346: plotting log2(ON:OFF) for P52517\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 331 of 346: plotting log2(ON:OFF) for Q2HRC8\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 332 of 346: plotting log2(ON:OFF) for P10186\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 333 of 346: plotting log2(ON:OFF) for P17151\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 334 of 346: plotting log2(ON:OFF) for P12977\n",
      "Activation domain starts/stops/scores: [701.0] [850.0] [7.321095275]\n",
      "Repression domain starts/stops/scores: [791.0] [930.0] [2.638298322]\n",
      "Protein 335 of 346: plotting log2(ON:OFF) for Q89893\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [171.0] [260.0] [0.9971794540000001]\n",
      "Protein 336 of 346: plotting log2(ON:OFF) for P06726\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [51.0, 381.0] [140.0, 480.0] [2.02243373, 3.818779417]\n",
      "Protein 337 of 346: plotting log2(ON:OFF) for Q3KST0\n",
      "Activation domain starts/stops/scores: [751.0] [840.0] [5.072182016]\n",
      "Repression domain starts/stops/scores: [441.0] [570.0] [3.300042972]\n",
      "Protein 338 of 346: plotting log2(ON:OFF) for P28277\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [131.0, 581.0] [240.0, 670.0] [4.263314238, 1.060120304]\n",
      "Protein 339 of 346: plotting log2(ON:OFF) for P16800\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [251.0] [345.0] [1.419435861]\n",
      "Protein 340 of 346: plotting log2(ON:OFF) for Q9QJ39\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 341 of 346: plotting log2(ON:OFF) for F5HF49\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 342 of 346: plotting log2(ON:OFF) for P52357\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 343 of 346: plotting log2(ON:OFF) for Q89743\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 344 of 346: plotting log2(ON:OFF) for P17887\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 345 of 346: plotting log2(ON:OFF) for P03188\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Protein 346 of 346: plotting log2(ON:OFF) for Q77Z83\n",
      "Activation domain starts/stops/scores: [1.0] [110.0] [8.179664852]\n",
      "Repression domain starts/stops/scores: [] [] []\n",
      "Finished\n"
     ]
    }
   ],
   "source": [
    "# the following creates dictionaries to map UniProt ID to protein and virus names\n",
    "# this is specifically to produce tiling plot titles and save names\n",
    "uID_to_gene_df = pd.read_csv('../Supplemental Tables - CSV/uID_protein_virus_mapping_HHV_expanded.csv')\n",
    "uID_to_gene_df = uID_to_gene_df.rename(columns={'UniProt ID':'Protein ID'})\n",
    "uID_list = list(uID_to_gene_df['Protein ID'])\n",
    "uID_to_gene_df['Virus Protein (ProteinID)'] = uID_to_gene_df['Virus'] + ' ' + uID_to_gene_df['Protein'] + ' (' + uID_to_gene_df['Protein ID'] + ')'\n",
    "uID_to_gene_df['Protein_Virus_ProteinID'] = uID_to_gene_df['Protein'] + '_' + uID_to_gene_df['Virus'] + '_' + uID_to_gene_df['Protein ID']\n",
    "gene_uID_list1 = list(uID_to_gene_df['Virus Protein (ProteinID)'])\n",
    "gene_uID_list_dict1 = dict(zip(uID_list, gene_uID_list1))\n",
    "gene_uID_list2 = list(uID_to_gene_df['Protein_Virus_ProteinID'])\n",
    "gene_uID_list_dict2 = dict(zip(uID_list, gene_uID_list2))\n",
    "\n",
    "###############################\n",
    "\n",
    "dpi = 150\n",
    "figsize = (4, 1.8) # original smaller format was around (2.6, 1.5)\n",
    "df = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_ActivD2-ReprD5_detailed.csv')\n",
    "indivDir_path = '../Individual Figure Panels/HHV_indivProteins_allTiles_%d-dpi' % dpi\n",
    "save_root = '_tileScores_HHV.png'\n",
    "HHVprot = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_full-length-proteins_uID_sequence.csv')\n",
    "HHVprot = HHVprot.rename(columns={'Cluster ID':'Entry'})\n",
    "HHVprot = HHVprot[['Entry', 'Representative Sequence']]\n",
    "HHVprot = HHVprot.set_index('Entry')\n",
    "\n",
    "if os.path.isdir(indivDir_path) == False:\n",
    "    print('Making directory called %s to store plots' % indivDir_path)\n",
    "    os.mkdir(indivDir_path)\n",
    "\n",
    "# compute minimum and maximum log2(ON:OFF) for plots\n",
    "minScoreA = limsA_HHV[0]\n",
    "maxScoreA = limsA_HHV[1]\n",
    "minScoreR = limsR_HHV[0]\n",
    "maxScoreR = limsR_HHV[1]\n",
    "\n",
    "# extract domains and make tiling plots\n",
    "domDF_HHV = extractDomains(df, HHVprot, 'herpesvirus', indivDir_path, save_root, figsize, dpi)\n",
    "domDF_HHV = pd.merge(domDF_HHV, uID_to_gene_df, on='Protein ID', how='left')\n",
    "domDF_HHV = domDF_HHV.drop_duplicates(subset=['Protein ID', 'Max Score', 'Effect'])\n",
    "domDF_HHV.to_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_domains.csv', index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Add estimated percent ON or OFF based on ground truth individual validation experiments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Complete\n"
     ]
    }
   ],
   "source": [
    "### update all files with columns that estimate % ON or OFF\n",
    "# parameters from the correlation plots later in this notebook\n",
    "# HHV = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_ActivD2-ReprD5_detailed.csv')\n",
    "HHVA = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/aHV-bHV-gHV_ActivD2_bothseqs_combo_detailed_log2OnOff_reAddDups.csv')\n",
    "HHVR = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/aHV-bHV-gHV_ReprD5_bothseqs_combo_detailed_log2OnOff_reAddDups.csv')\n",
    "HHV_doms = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_domains.csv')\n",
    "\n",
    "# parameters from model fit in 'Viral Figure 2 + S3 (HHV tiling, notebook 1).ipynb'\n",
    "HHVA_params = [96.92073446, 5.70115069, 0.9198081, 1.06626647]\n",
    "HHVR_params = [105.77107461, 2.17386321, 1.05766339, -4.35079577]\n",
    "HHVA_cutoff = 1.0846347878236864\n",
    "HHVR_cutoff = 0.6987727639587654\n",
    "\n",
    "def est_activity(scores, params, cutoff):\n",
    "    activities = []\n",
    "    for s in scores:\n",
    "        activities.append(params[0]/(1 + np.exp(-params[2]*(s-params[1])))+params[3])\n",
    "    activities = [i if i >= cutoff else 0 for i in activities]\n",
    "    return activities\n",
    "\n",
    "def add_est_activity(df, cutoff, screentype, params):\n",
    "    if screentype == 'Activation':\n",
    "        colname = 'Est. %ON'\n",
    "    elif screentype == 'Repression':\n",
    "        colname = 'Est. %OFF'\n",
    "        \n",
    "    df['R1 ' + colname] = est_activity(list(df['R1']), params, cutoff)\n",
    "    df['R2 ' + colname] = est_activity(list(df['R2']), params, cutoff)\n",
    "    df['Mean ' + colname] = np.mean([df['R1 ' + colname], df['R2 ' + colname]], axis=0)\n",
    "    df['Std ' + colname] = np.std([df['R1 ' + colname], df['R2 ' + colname]], axis=0)\n",
    "    return df\n",
    "\n",
    "def dom_est_activity(domdf, alldf, screentype):\n",
    "    if screentype == 'Activation':\n",
    "        colname = 'Est. %ON'\n",
    "    elif screentype == 'Repression':\n",
    "        colname = 'Est. %OFF'\n",
    "        \n",
    "    domdf = domdf[domdf['Effect']==screentype]\n",
    "    domdf = pd.merge(domdf, alldf[['Tile ID', 'R1 ' + colname, 'R2 ' + colname, 'Mean ' + colname, 'Std ' + colname]], on='Tile ID', how='left')\n",
    "    return domdf\n",
    "    \n",
    "\n",
    "HHVA = add_est_activity(HHVA, HHVA_cutoff, 'Activation', HHVA_params)\n",
    "HHVA.to_csv('../Supplemental Tables - CSV/HHV-tiling/aHV-bHV-gHV_ActivD2_bothseqs_combo_detailed_log2OnOff_reAddDups_estON.csv', index=False)\n",
    "HHVR = add_est_activity(HHVR, HHVR_cutoff, 'Repression', HHVR_params)\n",
    "HHVR.to_csv('../Supplemental Tables - CSV/HHV-tiling/aHV-bHV-gHV_ReprD5_bothseqs_combo_detailed_log2OnOff_reAddDups_estOFF.csv', index=False)\n",
    "\n",
    "HHV_doms = pd.concat([dom_est_activity(HHV_doms, HHVA, 'Activation'), dom_est_activity(HHV_doms, HHVR, 'Repression')])\n",
    "HHV_doms = HHV_doms.sort_values(by=['Effect', 'Max Score'], ascending=[True, False])\n",
    "HHV_doms.to_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_domains_estON-OFF.csv', index=False)\n",
    "print('Complete')\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Breakdown of proteins with effector domains by virus"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Barplot featured in Fig. 2D"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    Virus  Count\n",
      "2     EBV     93\n",
      "3    HCMV    238\n",
      "4   HHV6A    103\n",
      "5   HHV6B    102\n",
      "6    HHV7     63\n",
      "7    HSV1     85\n",
      "8    HSV2     79\n",
      "9    KSHV     85\n",
      "10  SuHV1     37\n",
      "11    VZV     70\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-22-09029f38a56b>:26: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  HHVdomActDup['Virus'] = HHVdomActDup['Original']\n",
      "<ipython-input-22-09029f38a56b>:38: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  HHVdomRepDup['Virus'] = HHVdomRepDup['Original']\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAANIAAACNCAYAAADCQSRgAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjMuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8vihELAAAACXBIWXMAAAsTAAALEwEAmpwYAAAkuElEQVR4nO2deVxVdf7/n29QQjFc0MKk1NxSG0vHXWPQLCnHsjS1QkXclVxKc8mFgbHcJpVRcE1Em9BvariWmZo/UxttmTI1dVyLwZQQNxCU9++Pc8ArsiUXLlzP8/Hgwb3nnvP5vD/33vf9bK/zfouqYmFhUTBcHG2AhYUzYDmShYUdsBzJwsIOWI5kYWEHLEeysLADliNZWNgBp3Ikf39/Baw/6++P/hWYUvYopLjw//57luZbfnK0GfckK/dF5eu8pMe326W+Zq98Y5dy7IVTOZLFvcexY8cYN24c3t7eNGvWjF27dnHz5k1EhCFDhtCiRQtCQ0MZPHgwb7zxBhUrVuTq1assXbqUTZs2sWbNGj788MPlwBzgP8A/gWtAdaC3qqbkx45CGdqJSKCI9LR5HiUiVUVkmYgsFZHNIvKq+dpxERltc+5SEVkgIiEi8qLN8bEi0rkw7LUouSQlJTFr1izmzJnD2rVr+fHHH/Hw8KBcuXI0bNgQgMTERM6fP88777zDggULqFChAidPnkREWLZsGcA84CWgLNATqAyk59eJoGjnSFHAe6raD/grUMU8/l/AF0BEPIGK5vEIINA87gq0BzYWnbkWJYGmTZvi5uZGp06daNGiBTNnziQiIgJ/f3/Cw8M5ePAgjz/+OA0bNqRRo0Zs3rwZFxcX6tWrR5cuXdizZw/AYmA7IMCLqtoXOCkivvm1ozAdaYjZE0UBbYFSqnoUQFXTVTXcPE+BPSLSCngd+NA85zfgvIg8CnQG1mo2wkARGSgiB0TkwI1LiYXYHIviyPfff4+7uztbt27lwIEDfPXVVwB4eXmRmprKunXr6NKlCwBhYWEcO3aMuXPnAvDFF1/g6+sL8GfgHeBZoJlZ9P8Az/zaUZiOFKmqgaoaCOwGXESkPoCIuIhIpIhkzNGigF5AY8B2FjkXGAD0AKKzq0RVF6lqU1VtWsqzYnanWDgxqampDBo0iEGDBlG7dm3OnTvHiBEjWLBgAUOGDCEhIQEvLy+io6NZsWIF+/fvJyAggKNHj3Ly5En69OkD8AGwEtgG+IpIOPAn4NP82iGFof4WkUAgRVVjzOdRwERgNpACeACrVTVGRD5VVX8R+RewE9gKjFPVwea1/wKOqmpIXvV61GmoDcNX2709FnlTwlftpKAFFMqqnapGZXkeaD58JZtz/c3/r9kcHmzz+mtZr7GwKG4USo/kKJo2baoHDhxwtBkWJY8C90hOpWywsHAUTrUhezgpucQrG/I71yjO5DYPKm6KBHth9UgWFnbAIT2SuarXDzhtHjoIBAH/BsoDu4A1wAxV7WZeUw34h6r2vKNAixJHVmnP+fPnOXXqFElJScyePZsqVaoQGhrKsGHDGDhwIA8++CAPPfQQEyZMIDY2ljVr1uDq6srIkSNp3Lixo5vj0KHdfJvl8RpADVUdbO4tfaWqM0UkXUSqqeqvGCt54bmUZ1GCyJD2+Pj40KlTJ+677z42bNjAjh07WLx4MRMmTCAxMZFdu3bh6+vLiBEjCA8PZ8eOHZnSnu+//55169YVC0dy5NDOVvnwMtDOfLwTYy8JDMcZICJuQCNV3eMIQy3sj620x8/PjwceeAAAHx8f4uLiMqU9zz//PKdPn2b48OEcPnyYc+fOZUp7BgwYQPv27R3cEgNHOpKt8mEtsMN83hZoKSLVVXU38ATwKrAiu0IsiVDJxFba880333DhwgUAfvnlFx566KFMac+FCxcydXMPPvgg1atXz5T2fPPNN0ydOtXBLTFw5NBumIj81XzsDjQSkZUY2rtzwFnztdXAKKBVdoWo6iJgERjKhkK12MJuZEh7vLy8ePTRR/Hx8WHo0KEkJiayYMECpkyZgpeXF2lpaYwdO5Z169bh6elJmzZtWLJkCX369EFECAgIcHRTACfbkHUGiZC1/O0QrA1ZC4vigFP1SJZEyOIusXokC4vigCURKmbYa450t7crFNM5TLHHqRzJwn7s2bOHuXPnUq5cOapXr05aWhoJCQkkJycze/ZsKlSoQGhoKMHBwUyaNImyZcty+vRpoqOjWbVqFR999BHe3t60a9cu4+Y5p6ZQh3a2QVDM/Z5IEZliBjhZKiJh5mtRIuJtc92nIuInIrNtjjUXkX8Upr0Wt0hMTGTJkiUsXbqU3bt388QTTxAREUGDBg349ttvM89xc3MjJiaGCxcu4OLigru7O7t27aJatWqoKi1btnRwS4qGopgjiekw96vqEOApYKQZBOVHEbkvu4tUdSfwJxEpYx4agiURKjI6depEuXLlmDp1Kq+//jrdunVjwYIFxMTEUL9+/UzlgaoSGxvLsmXLqFmzJrt27SIoKIjIyEhmzZrF22+/7eimFAlFMbQbC/wOTDafDwPGmxGD4oEb5vG5IpJsPq5s/o8CeojIZowVxgyRayYiMhAYCOD2QNVCacC9yOXLlxk5ciSvvfYavr6+7Nq1i8GDB9O6dWumT5+Ol5cXQ4cOZevWrZw5c4a2bdtStWpVLl26xKFDh2jVqhWenp4406pwbhSFI4UD/wdsFJEJgK+qTgAQkb9zS7EwQlXjzeMZQSdWATGAD0bssTuwlA2Fw4gRIzh27BjLli0jOjqa+++/n5iYGBISEhg7dizR0dF4eXnRoUMHAgMDOXnyJMnJyQwdOpTz58/Tr18/0tPTGT9+vKObUiQUhSNdU9XLItIfWA7sF5GPMaJZpgLf5nShqqaJyH6guar+vQhstTD54IMPcn29SZMmAJQvX55169bd9lrfvn3p27dvodlWHClUR7INgqKqx4DWOZwamOU6f5vH0wrDNgsLe2IpGywsLGWDhUXxwKk2ZEuqsqEwgitaCoWixeqRLCzsQKH3SDmEL/4AY4P1Ckb44tHAVGCuqn5vnhcBLMDYIyoFPAi8paonCttmZ+P48eO88sorfPfddwwbNozU1FTKli3LlClTqFSpEqGhoTRs2JANGzYA8Pnnn7Nt2zZ27drFzp07qVy5Mn369KFp06YObknxxVFDuz7AUlXdJiL1gGoYccGHYMRyKA94AyeAT1V1o4h0BZ4BFjrI5hJJfHw8S5YswcPDg8TEROLi4li3bh0//PADs2fPJiwsjMTERLp27UrXrl2JjY2ladOm1K9fnzfeeIPPPvsMVaVHjx6sWbPG0c0pthSVIw0RkYwl7bbA00AXM3FYKeB9Vf2viHiLyP1AX2Chql7B2MitjZGRol/Wgi1lQ+54e3szbdo0/P39qVixIk8//TQDBw6kRo0anDt3LlPqA3Djxg0iIiLYsmULABMmTKBv377UqFGD69evO7IZxZ6icqTILEO7ECBYVa+aTjIe6A8sAQKAv6jqHPP8LhiOF6iq17IWbCkb/hgeHh4sWrSInTt3oqqsW7eOoUOHArB+/Xq6d++Oi4sxdT579ixRUVFcu3aNXr16OdLsYo+jhnZLgFUikgiUAWaYxzdjBIf8AMB0ssXAZ8AiEfk/VY11gL1Ow5kzZxg0aBBXrlxh4cKFTJw4ES8vLwD27dtH7969M8+tUKECvXr1Ii0tjbCwMEeZXCJwqg3Zkhr8xFr+djjWhqyFRXHAqXokSyJkcZdYPZKFRXHAkggVA+wxR7LmRI7Fbj2SbXwG83mUGXshX7EYRKSJiCwWkRnm3lDG60tFJNJedt4rHD9+PDNLw6xZsxgzZgw9e/bkhx9+ACA0NJTff/+dpKQkunbtyr59+wDYtm0bvXv3plevXuzZs4eUlBQCAgIYPnz4PXPb+N2QqyOJyIsi8pCIfCgig+xRYS6xGMYCZzAUDfvM+h/GWB73FpGH7FH/vYCtmgHgk08+4fTp08THx+Pj4wMYgUsqVarE1KlTKVu2bOa177//PkuWLGHRokVMnz6dixcvMmjQIMLDwzl69Ki1MZsDeQ3tXsSYiG0DmuSjvKwKhuPkMxaDiDwGDAeuAyuBvwJvmI/LYATSH5OfRt3r2KoZACZOnEjHjh3ZunUr8+bN4+WXX85UM8yYMYOQkJDMa1UVNzc3AFJSUvD29sbb25uBAwfi4uKCSIHn5U5JXkM7BboCXwH56RFsU7XsNo+NsDl2wTy2CuiMIe3JiMXwP+Cy+ZcqIh4YsqAuwHPAayJSIWuFVlqX3ElISCAyMhIRyQxOkpEyJTvc3d1JTU0lOTkZd3d34uLiOHz4MIsWLcLPz4/NmzcXbQNKCHn1SIsxVNcCzLFXpTnEYpiKoWi4jOFcfTFSXYYDiMgwjAhEU7OUZUmEcsHLy4tHHnmEgQMHcu3aNd59913ef//9TDVDVkaOHEn//v1JS0tj0qRJuLu7M2rUKCpXrkxSUtJtygeLW+S6jyQiUzB6JQBUNbQojLpb7mVlg7VqVyAKPF7Nq0fKCIvlBbxQ0MosLJyVfCsbRGSJqvYvZHsKhKVssLhLCrdHEpEtGEM7F+DfBa3MwsJZyWtoF6uqC4rEEjtQnJQNuc17rJQrzkdejuQrIieAiwCqavVKxQjbWAwzZ87k3LlzXLp0ibfffpvatWvnGIvh119/JTo6GlVlyJAhtG6dU9xOi/ySlyP9DNjm5SiwI4nIZ0AvVf3NfL4dSMEItO8JNMLYsP0u4yY+ERkLHFLVDQWt31mwVS+cO3eO7du3U7duXcqWLcsjjzwCkGMshrfeeotPPvmEmzdv0rNnT2JjrXslC0qOG7Ii0gtIxlAaXMf4stuD+UCQWUdt4KiqPg8MBlwx1BQRmGGMRcQVaA9stFP9TkGGeqFcuXKcOnUKNzc35s6dS6NGjVi5cmW2sRgybinPUC+UKVOGlBR7faz3NrkpG/pifJkvAnuBr+1U5wbAT0RcMDZY55o5klYAoar6H7O3Oi8ij2IoINZqDsuLlrLBcKry5csDUKVKFdLT029TL2SNxZBVvWBRcHJ0JFVtj/ElrgcMsFeFpkNsAF4CHgCOYQzl5qiqrbPONevtAUTnUt4iVW2qqk1LeVa0l5kliurVq+Pj40NwcDCxsbH06NGDhISE22IxtGjRIvP8DPVCUFAQkyZNcpTZTkW+9pFE5GXgXVV9zC6VipQFvgcGAU9iaO5sl6T6mjKif2EM/ULyU25xUjZYq3Z3EhUVxdKlSzNz0vr6+jJs2LC7Li8+Pp6VK1cyevToO16Li4vjo48+4q233rrt+KlTp/jkk08YOXKk7eHC20cSkcoYceT+gtGDNC5oZRmYYbXqmk93YASHzO681+xVp0XxYNiwYfTsady21r9/f8LCwjhx4gSpqakEBgby66+/snnzZmrVqsWJEyd48skn2bt3L+Hh4Xz33Xds3LiRa9eu0aJFC7p06cLx48eJioris88+o379+pw9e5bFixcTEhLCjBkz2Lt3L4sXLyY9PZ0XX3yRl156ibi4OE6dOkWNGjXs1q7cVu1WAvNUdbrdaitk6pcvw7+fa+hoMwyem+loC4o9f/7znxk6dCiqSlpaGp07d6Znz5506NCBgQMH0rp1a1atWsWSJUv497//Tb169QgMDOTnn3/mo48+uk3B3qFDB/r160eHDh0AOHnyJBUqVODcuXNcv36drl270rx5cwBatGjBp59+yuDBg+3WltzmSP6qaq2UWRQae/fupV27dgCISGa+WU9Pz9v+lypVivT0dMLCwoiLi6Nly5Z35KbNuImxVCmjb7jvPiPHd7169Zg8eTIXL17MHAKWLl06c+HFXlgxG+xIflXcuZF1/lTS50VZmT9/Phs3biQlJYU2bdrQoUMHBgww1rLGjRvH6dN35NvOpFq1anz55ZeZUWJz4+GHH+bixYucP3+eOXPmULVqVXx9fQHYvXs3wcHB9msUThaOy9GLDZYjFR8SEhJYvnw5b7755m3HT548ydatWxk06LbICcU/HFcOQVF8ReQjM9jJv8y4EMtE5Emb8yJEpJGIuIrIeFPdcE9jG9AkKCiIPn36EBgYyNdfG7sGGQFNhg0bxpgxY+jevTspKSksX74cf39/AgMDWb58uSObUGR4eXnd4UQANWvWzOpEdsFRce0y0roMAP7GrbQugwAy0rqo6g8YS+NlscOvRkkma0CTH3/8EQ8PD8qVK0fDhsYCS2JiIm5ubsTExHDhwgVcXFxwd3dn165dVKtWDVWlZcuWuVVjcZcU67QuAKoaKSJ+3K75y+ReSetiG9BEVZk5cyZ+fn5s3LiR8PBwXnjhBR5//HFUldjYWNq2bcv48ePZtWsXQUFBNGvWjKSkJPr3729p6wqBouqRsgZFCQGWqOoIjJ5ovHmebVqXz/JT8L2obLh8+TKHDh0CjCFMampqpiRo69at7N+/HyAz2MlXX31FqVKl8PT0zHOSbnF3FOu0LhbZ4+npyZEjRxgxYgQXL15kxowZvPfee3h5edGhQwcCAwM5efIkycnJDB06lPPnz9OvXz/S09MZP3583hXcBXE7nyA99XyOr7u4VeEhv/8UqI4zZ85kKtvzw+XLl7l58yYVKlQoUL35wVq1syP38qrdL1vzjtbm82xcrq///vvvNGzYkI8//pg2bdrc8fpzzz3Hli1bGD16NLNmzcqzvr/97W907NgxP/PCQg9+YmFRZCxcuJCwsDBmzZpF8+bNefPNN3F1dSUpKYlXX32Vo0ePsnfvXg4ePMjatWu5cOECAwcOpH///kyaNImQkBA8PDz473//y8qVK/niiy84d+4cPj4+vPXWW1SpUoUyZcowc+ZM6tati6+vL6GhoTz0UMGD+DqVIzlcImTJgu6atLQ01q5dy969e1m/fj2RkZHUqVOH4cOHc/DgQerUqUOdOnVo1aoVAC+88AI9evTgtddeIzk5OTPz+pUrV5g/fz4HDx6kffv2+Pv7889//pNx48bRuHFjhg8fzk8//USlSpVYsmSJ3ey30rpYFAtWr17NzZs3CQ4O5saNG0yZMiVTxhMXF8fVq1dvO79UqVLUrVuX6dOn07NnTw4ePEhkZCQVK1akbt26qGpmeOX09PTMslxcXFBVu8+bnKpHKikSoZI6D8oNF7cqeS425MaiRYvYvHkz3t7eqCpt27Zl7dq1HD16lGvXrvH000+TkpJyW8jkfv368fTTTxMSEsLRo0e5ePEia9eu5aeffsLX15fatWszY8YMZs+ezYQJE6hatSoeHh6Zdw7bE7stNohIIJCSJXu5N0Y28njz2KfANOBFVR1lHmuOcfPehxiZKRKB46q6SETigO0YK3tnMq7JiZKy2JCbI9kGNJk0aRIJCQkkJycze/ZsKlSoQGhoKMHBwbi6uhIUFMSYMWNo2bIl27Ztuy2gSd26dRk2bBgVK1bk6tWrLF26NDM4vsUdFH+JUFb+SFoX4AdVDVDVrkBrM36D05JVvfDEE08QERFBgwYN+Pbbb4H8p2M5d+4c77zzDgsWLKBChQqcPHnSIW26V7C3Iw0xtXRRGAoGMGIyZBzLmtblAcy0LsBjGMHwhwPvmuf9KSNhGXAESLezvcUK24AmAN26dWPBggXExMRQv3792wKazJgxg1q1amVemzWgScOGDWnUqBGbN2/GxcWFevXqOaRN9wr2niNFZjO0G5FlaAdGWpcYwIc707pcB1LNYz+aaghEZAngC3xpW6GzSoTS0tLYu3cvgwcPpnXr1kyfPh0vL6/MSEBZyQhocvPmzcyAJmFhYXh6ejJ37tyiNP2exCGLDflM6wLQSERWYvRE9wPfZlOWU6Z1KV26NKtXryYmJoaEhATGjh1LdHR0vtOxREdHs2LFCpo3b05AQACTJ0+mbt262V5rDxofuMj5tJzf/iqlhe+aVii0+h2NpWywI/fyqp3P3rxDof3SKnstZFRUFB9++CG1atXixo0bXL16lejoaEqXLm1vM3PCUjZYOAf9+vXLDIoydOhQIiMj+fnnn7l58yZNmzbNDMH8wgsvcObMGWbMmMFf/vIXHnvsMUaMGMHChQtxdXXl+vXrhIeHM3r0aNzc3IiPj2f58uWMGjXqtudz5szh6NGjXL58mZiYmCcxMkPWA3ap6h9O/u1UjmQpG0oukZGRrF+/nkOHDvHqq68SGxtLs2bNANixYwe1a9emffv2jBw5ktmzZ/PVV19x8+ZNli5dSmRkJAkJCdSoUYOEhAQOHjzIiRMneOaZZ+jatSsictvz69evs2fPHtauXUtCQgIxMTETgEPAclX9NFdDc8CpHMmi5DJkyBB69uzJ4sWLOXjwIDdu3GDixImUKVOGqKgowFAoACQnJ+Pm5papTkhPT+fZZ58lKCiI9evX88ADDzBlyhRSU1OZNm0akyZNuu35mDFjMlUPpuIhY/X64t3a71SOVBjKhqzzHtv5jTPMbYobAwYMIDg4mKCgIAIDAylVqhSdO3cGYNOmTVy5coUbN27cFjk2ICCAoKAgDhw4wLVr1+jUqRMhISG4u7tTvnx5Hn30Ud5+++3M5w0aNKBFixYEBwdz7do1MO7S7loQu63FhjzIjyP98ssvTJ48mYoVK6KqhIWFMW3aNNq0aUNsbCz33Xcf7dq1o2PHjvTv359KlSrh7u7OjBkzcBYKe9Vu586d7Nu3j3Hjxt11GblQvBcb7CAbugTUBMoDlVX1qcK09245cuQI27dvp1mzZrRq1YqNGzfSqVMnpk6dSqNGjfj1119p3LgxFy9eZNCgQTz11FN06dKF69evZ8ZfK+kU9tK2n58ffn5+hVpHQSgWQztV3SkiE0WkjKomY8iGQkzFAyLyAYbioVjy8MMPs337dmrWrIm/vz8NGjSge/fuHDx4kFWrVhEfH8/EiROJjo7G29ubgQMH4uLikjlOtyj5FIXWriCyIUSkI3BMVc9mV3hxSOsyb948kpKSEBHKli2LiCAiVK9eHXd398xN1Li4OA4fPsyiRYvw8/O7TclsUbIpih6pILIhMCIKDcyp8OKgbOjbty+TJ0/mkUceoWXLljz1lDECDQ4Opk+fPri6umZOdkeNGkXlypVJSkqid+/ejjDXohAoFkM7yFE2BOCpqpccZVd+aNKkSWaeVlu6detGt27dbju2atWqojKrSLkw+U30cs4fk9zvSeXQ9wtUx2+//Ub58uWL5byyUId2qhqV0RuZzwPN4PzxNsf8bR5PU9WXs5TxfGHaaGEfcnOivF6PioriqaeeIiAggA4dOrBjx45szxs7diyJiYmEhISwb9++O17/6aefCAgIoHXr1pnlHT58+I815C5xquXvpk2b6oEDBxxtxj3J+VH98zynyuzsYyRERUXh7u5Oz549OXDgAEuWLGHevHkMHjyYsmXLkpCQwHvvvUfnzp15/fXXuXbtGocPH8bd3Z2aNWsSEhKSY3nPP/88H3/8MefPn2fmzJlcuXIFb29vPDw8ePTRR+ncuTPly5ePAG4C92Esas3CuAPBG+ijqjfyaluxGdpZ3NtERkayZcsWvv76a8LDw/n8889p2LAho0aNYu3atWzYsIHGjRvTu3dvFixYQN++ffH39+fZZ5/NtdygoCBWrlzJ8ePHGTZsGNOnTyc4OBgfHx9efPFFLl26BOAFnDL/Pw48CnwOrAHy1dM4lSPZS9kw/0rOiwCWmqFwyJAIpaSk0KxZM6ZNm3ZHwBLb7YIMeVBeeY5eeuklXnnlFcqUKUP9+vWBW1Kj9PT0jMdbVfUDEXkB+A1D6eAGjAMmY6RpzRWnciSLksv8+fPZtGkTycnJ9OrVi44dO7Ju3TpGjx7NpUuXmDlzJpcvX+a9996jYsX8h6Z2dXWlTp06mQnNwMjaUbp0afr160e7du0IDg7uJCJNMZI1bMIIqZ0CJAHZbrtkxa5zJDPv7GzTiPuB+ar6/7KcEwJ8qqr7zOefAkOBGarazTxWDfiHqvYUkdrA/6lqnjls7SURyk+PNH/+fPbv309aWhq7d+/m0KFDTJs2jSZNmrBmzRpcXV0ZOXIkv//+OytWrADg888/Z9u2bZm/jM5EUaza3Q0RERHs37+fZcuWARAYGMi0adPw9va+zbyC1mNvR3oG43bwyRhd43CglqoOFpFxGEFN/MjiSKrqLyKrgVGq+quIhAFbgBPASKCtqra9o8IsFKUjZTB27Fh69OjBsWPHqF69OvHx8XTq1Invv/+eDRs2EBoaCkBsbCxnz561e6Y4C7tQvLR2qvq5iHhi9EplgP05nDpJRDKCoD1m/g8HBojIu0AjVZ1kHh9ns2lbrDhy5AhJSUk0adKEFStW0L17d0SEL7/8khEjRjBnzhwAbty4QUREBFu2bHGswRaFhl0dSUR6AUdUdaSIlAI+w1gNAWNFJIOwLEM7VHW3iLwFvAqs+AN1Oiz4SUREBGPGjOHq1at4eHggInzxxRe0b9+eb775Bn9/f/z8/Fi/fj3du3e3ewJgi+KDvRcbvgDCRSQFY2j3L+CvIjIPY0lxUx7XrwZGAa3yW6EjJULHjx+nVq1arF69muefN/aNT548SZ8+fRARAgICANi3b58lB3JynGpD1hFzJAunoORFWrWwcEacah/JfsFPrF7H4o/hVEM7EbkM/OxoO0wqAxccbYSJZUv2ZNhywVY8fTc4VY8E/KyqTR1tBICIHLBsuRNntcWaI1lY2AHLkSws7ICzOdIiRxtgg2VL9jilLU612GBh4SicrUeysHAITrFql3HbBfA78JOqzneQHZm3fIjIGKAGRnDLUaqac6Zi+9vRGhgBXAFOA8kOtKUORgDQeAwRcxVH2WJj04fABuBhu9miqiX+DwgFWpuPNwOlHWCDN8YXZjfgDmwwj7cDJhSxLZ2A+83HWx1sS1OMaLmlHW2LWe+bwEKgpz1tcZahnTe37mRMxPiFKVJUNV5Vx2H0ApUwblkG+AV4qIht2QRcEZF3MLLFO9KWAxiBRDYBOx1pi4h0xsg4sRdjWmM3W5xiaIeRDd0Hw5kqUYD0HHbiN27dNuIDxBVl5SJyPzAHQ33/JfCSA215Ejirqs+KyBpufeeK3BYgAOOHNiMz9RV72eIUq3Yi4g28j5F/9oCqLnagLRl3/I7A+MAqAoNVNakIbfgAqIMxP7qJkXvXUbY0B94GEjCSIvziKFtsbArECIfwoL1scQpHsrBwNM4yR7KwcCiWI1lY2AHLkSws7IDlSMUMEblfRCoUUV2PFEU9Zl0PF1VdjqDEO5KIBIrIbjNqESKyQERq/MEyQkSkpR1sqSciO8zQtxnH/iMiK0VkhYhsEpFaeRTzJrdClGUtf1ZBbcxSzkLzebbhzkSkhogcNe1fKSLLzeORIrJIRCaLyIciUiWP+qoD7/wB+6JE5GOzzjUi4pvfa3Mp8zkRebqg5eSEs+wjxWOoCkZnHLBZhs5QHERhfJh7gYbAd8ATwEzzktEiEo+xJ/UPYAbG+1MB48u9DDgHfKSq2806RgN1MaLKTgeCMIKu2+YS+Z+qBpjnvwo8IyKTgDSMQO3NMELlVgLGAk8DD4rIr6a9NzAkPm8Dj4uIn3nebtP+VzH2jG7LniAirsDHqvqSiGwD3sVYCm9kltMBqCsirYDKIhKO4cDjVPVbG/u3q+pgm/e1MfCs+f4MBdYDT4jIS4ArcAAjGtQ/MJaYU833o5mI1DffI9v2hma8F6pqG7kmWFXjRcQNYzP3GRGZmc218UBVjKX1/2EoKbpihB2uCNQG3sNY6k4xk92dxdi0P4IRMi4EuAr8oKoR3AXO4kgfAy3MDzM39qvqZBH5EsO5/Mw/gIVqBLjcCPwINAD+g+EkLYFyQE9VvQYgIh4YsqSXRcQLiAQigDhVPWZTZ1Xzw0vHcIjxZnn/wIhe00ZVh4lIE+ANYDvwKRBsvn4FY0/INmjfAVWdau4XVSOb7AmqelNELpg6xIvAMxif91ygk6puE5FjqrpXRNIwtHltMeRFto7UzrQf4Kiqvmtet9B0wvkY6omMYKDtMEY6G1R1vYj8GWMPKeNOVLcs7QUjPPWROz8uUNVUEfldRP6Uw7UfmG1+R1XfEJGVGHq+bRgh4ToB7TH21DKIAk6a7/N+wBPYmKXdfwhncSQwfrHXY9MmEXHh9sCUGcGpk81f7RvcGt5eNf9fx/gAMpzOF0OpkJrhRCYu3Er5kU7Ow+T/qWqg7QExsipcNG1LtylDbMp0AVap6nYz8KZtgtwMW1MxcvrklD1hAzAV44veDUhX1V/kziTQSaqqpkO5Znlth22PlAOlgL9j/FAEYkTZzWhXDW7FaHDJpr2QixLF7JEeMM/P7tpLGNrGFPP5DbMNYRi98DcYP0S2XFXVdBFJx/hsx2Okc5kHdMyjrdniNI6kRurMAdz6VVmHIZHJVzYBoL853NmBMZToJiLzMYYEfbOp77KIfG0GvyyL8WX2ynpeHjb/JCI3RWQ2xlBjMvAXjB+FUcAcEemB8cGvyMYBwBgWDSL77AlbMYakIzB6iqzvhbuI5JURsb35K5/BG9mc8y7Gr/wNDOddixEo9DkMp/8cY0jpAWRt79+zKQ9gnohcxxgJ/D2H9yqna0sD14DnMIZ3l7kV8TcrZTCGfqeAPTmckyeWssHCwg6U+FU7C4vigOVIFhZ2wHIkCws7YDmShYUdsBzJwsIOWI5kYWEHLEeysLAD/x+MHO9rfl4FNAAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 216x144 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "### first run block that plots number of proteins per virus in order to calculate rate\n",
    "import matplotlib.patches as mpatches\n",
    "from pandas.api.types import CategoricalDtype\n",
    "\n",
    "u90 = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/uniref_reviewed_HHV+SuHV.csv')\n",
    "u90 = u90.rename(columns={'Cluster ID':'UniProt ID'})\n",
    "dfmap = pd.read_csv('../Supplemental Tables - CSV/uID_protein_virus_mapping_HHV_expanded.csv')\n",
    "u90 = pd.merge(u90, dfmap, on='UniProt ID', how='left')\n",
    "u90['Count'] = 1\n",
    "u90 = u90[['Virus', 'Count']].groupby('Virus', axis=0).sum().reset_index()\n",
    "u90 = u90[~u90['Virus'].str.contains('CeHV')]\n",
    "print(u90)\n",
    "\n",
    "vir2sub = {'HSV1':'alpha', 'HSV2':'alpha', 'VZV':'alpha', 'SuHV1':'alpha',\n",
    "           'HCMV':'beta', 'HHV6A':'beta', 'HHV6B':'beta', 'HHV7':'beta',\n",
    "           'EBV':'gamma', 'KSHV':'gamma'}\n",
    "u90['Subfamily'] = u90['Virus'].map(vir2sub)\n",
    "u90['Virus'] = u90['Virus'].astype(CategoricalDtype(categories=['SuHV1', 'HSV1', 'HSV2', 'VZV', 'HCMV',\n",
    "                                                                'HHV6A', 'HHV6B', 'HHV7','EBV', 'KSHV'], ordered=True))\n",
    "u90 = u90.sort_values(by='Virus')\n",
    "\n",
    "HHVdom = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_domains_estON-OFF.csv')\n",
    "HHVdomAct = HHVdom[HHVdom['Effect']=='Activation']\n",
    "# make dataframe with proteins belonging to multiple virus\n",
    "HHVdomActDup = HHVdomAct[(HHVdomAct['Virus'])!=(HHVdomAct['Original'])]\n",
    "HHVdomActDup['Virus'] = HHVdomActDup['Original']\n",
    "HHVdomActDup = HHVdomActDup.drop_duplicates('Protein ID')\n",
    "HHVdomActDup['Protein ID'] = HHVdomActDup['Protein ID'] + '-1'\n",
    "# don't multi-count proteins with multiple domains\n",
    "HHVdomAct = HHVdomAct.drop_duplicates('Protein ID')\n",
    "# merge dataframes\n",
    "HHVdomAct = pd.concat([HHVdomAct, HHVdomActDup])\n",
    "HHVA_list = list(HHVdomAct['Protein ID'])\n",
    "\n",
    "HHVdomRep = HHVdom[HHVdom['Effect']=='Repression']\n",
    "# make dataframe with proteins belonging to multiple virus\n",
    "HHVdomRepDup = HHVdomRep[(HHVdomRep['Virus'])!=(HHVdomRep['Original'])]\n",
    "HHVdomRepDup['Virus'] = HHVdomRepDup['Original']\n",
    "HHVdomRepDup = HHVdomRepDup.drop_duplicates('Protein ID')\n",
    "HHVdomRepDup['Protein ID'] = HHVdomRepDup['Protein ID'] + '-1'\n",
    "# don't multi-count proteins with multiple domains\n",
    "HHVdomRep = HHVdomRep.drop_duplicates('Protein ID')\n",
    "# merge dataframes\n",
    "HHVdomRep = pd.concat([HHVdomRep, HHVdomRepDup])\n",
    "HHVR_list = list(HHVdomRep['Protein ID'])\n",
    "\n",
    "HHVB_list = [b for b in HHVA_list if b in HHVR_list]\n",
    "HHVA_list = [a for a in HHVA_list if a not in HHVB_list]\n",
    "HHVR_list = [r for r in HHVR_list if r not in HHVB_list]\n",
    "\n",
    "HHVdoms = pd.concat([HHVdomAct, HHVdomRep])\n",
    "HHVdoms['Count'] = 1\n",
    "HHVB = HHVdoms[HHVdoms['Protein ID'].str.contains('|'.join(HHVB_list))].drop_duplicates('Protein ID').groupby('Virus').sum().reset_index()\n",
    "HHVA = HHVdoms[HHVdoms['Protein ID'].str.contains('|'.join(HHVA_list))].drop_duplicates('Protein ID').groupby('Virus').sum().reset_index()\n",
    "HHVR = HHVdoms[HHVdoms['Protein ID'].str.contains('|'.join(HHVR_list))].drop_duplicates('Protein ID').groupby('Virus').sum().reset_index()\n",
    "\n",
    "HHVfull = HHVB[['Virus', 'Count']]\n",
    "# print(HHVfull[HHVfull['Virus']=='SuHV1'])\n",
    "HHVfull = HHVfull.rename(columns={'Count':'Bifunctional'})\n",
    "for hhvdf, hhvtype in zip([HHVR, HHVA], ['Repressor', 'Activator']):\n",
    "    hhvdf = hhvdf[['Virus', 'Count']]\n",
    "    hhvdf = hhvdf.rename(columns={'Count':hhvtype})\n",
    "    HHVfull = pd.merge(HHVfull, hhvdf, on='Virus', how='outer').fillna(0)\n",
    "\n",
    "HHVfull['Bifunctional'] = HHVfull['Repressor'] + HHVfull['Bifunctional']\n",
    "HHVfull['Activator'] = HHVfull['Activator'] + HHVfull['Bifunctional']\n",
    "HHVfull = HHVfull.sort_values(by='Activator', ascending=False)\n",
    "HHVfull = HHVfull[~HHVfull['Virus'].str.contains('CeHV', regex=True)]\n",
    "\n",
    "totalprots = dict(zip(list(u90['Virus']), list(u90['Count'])))\n",
    "HHVfull['Activator'] = HHVfull['Activator'].astype(int)\n",
    "HHVfull['Total'] = HHVfull['Virus'].map(totalprots)\n",
    "HHVfull['Rate'] = HHVfull['Activator'].astype(str) + '/' + HHVfull['Total'].astype(str)\n",
    "labels = list(HHVfull['Rate'])\n",
    "xpos = [x + 0.5 for x in list(HHVfull['Activator'])]\n",
    "ypos = list(range(len(labels)))\n",
    "\n",
    "# colors = ['#E3B919', '#1BC6EE', '#CE5F24']\n",
    "colors = ['#E3B919', '#1BC6EE', '#EA7063']\n",
    "\n",
    "plt.figure(figsize=(3,2))\n",
    "barA = sns.barplot(data=HHVfull, x='Activator', y='Virus', color=colors[0])\n",
    "barB = sns.barplot(data=HHVfull, x='Bifunctional', y='Virus', color=colors[2])\n",
    "barR = sns.barplot(data=HHVfull, x='Repressor', y='Virus', color=colors[1])\n",
    "\n",
    "for l, x, y in zip(labels, xpos, ypos):\n",
    "    plt.annotate(text=l, xy=(x, y), ha='left', va='center')\n",
    "\n",
    "sns.despine()\n",
    "\n",
    "A_bar = mpatches.Patch(color=colors[0], label='Activator')\n",
    "B_bar = mpatches.Patch(color=colors[2], label='Both Types')\n",
    "R_bar = mpatches.Patch(color=colors[1], label='Repressor')\n",
    "\n",
    "plt.legend(handles=[A_bar, R_bar, B_bar], loc='lower right', bbox_to_anchor=(1.1, 0),\n",
    "           frameon=False, title='Domain(s)', handlelength=1)\n",
    "plt.xlabel('Number of Proteins with Effector Domains')\n",
    "plt.xlim(0, xpos[0] + 5)\n",
    "plt.tight_layout()\n",
    "plt.savefig('../Individual Figure Panels/Figure 2/HHV_domain_virus_distribution_bar_annotated.png', dpi=1200)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### vTR + HHV Overlap"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Venn diagram featured in 2E"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "67\n",
      "147 34 31\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAPEAAADnCAYAAADPRLCPAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjMuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8vihELAAAACXBIWXMAAAsTAAALEwEAmpwYAAAWbklEQVR4nO3de2ydd33H8ff33Hx8qeMkzt3NPWmalDZJ1yalF0pXaLmOVQg06KDsjwIS2xiIPyYExtImTWOahjYJFU1rhQaIVYgJwUpb2rSiTUIKNGmTNPfEiZMmzqW248uxz+W3P57jxHV87OP4PM/v+T3P9yVZkY/TPl87/pzf8/yuYoxBKeWuhO0ClFIzoyFWynEaYqUcpyFWynEaYqUcpyFWynEaYqUcpyFWynEaYqUcpyFWynEaYqUcpyFWynEaYqUcpyFWynEaYqUcpyFWynEaYqUcpyFWynEaYqUcpyFWynEaYqUcl7JdgJoBkTogWemrgAFGMKYQXFEqaBriMBJJArOAFqAJqK/wUd2dlEgBGAJy5T/HfvQBPUAfun+xk0T/3SwTaQBayx9zgTlAM15LGqQSXpgvAReBC8BFjMkFXIeaJg1x0ESywBKgrfxnk92CpvQOcBroAt7GmLzletQ4GmK/ebfGC7ka2la7Bc1ICejGC/VpoBtjSnZLUhpiP4gkgKXAGuBGotv3MAIcAw5jzNu2i4krDXEticzHC+4qIGu5mqBdBo7gBbrHci2xoiGeKZEmYC2wGq83WXmdYofwAj1su5io0xBfL5G5wEZgJcH3JLuiALwFvIkx/baLiSoN8XSJLMIL742WK3FJCTgK7MGYS7aLiRoNcbVElgO3AQssV+K6k3hh1o6wGtEQT8UL7x3AbMuVRM1ZYCfGdNsuxHUa4kpEWoGtwGLbpUTcUWAXxly2XYirNMTjeTOqtgA32S4lRorAG8Drulhj+jTEo0QEWI9365yxXE1c9ePdYh+zXYhLNMQAInOA+3F7SmSUnAJexphB24W4QEMscite61tpXa6yIwf8FmOO2y4k7OIbYpFGvNZ3ieVK1OQOAdsxZsR2IWEVzxCLrATuBepsl6Kq0g9s07HlicUrxCIZ4G68RQrKLQavB/s1Xf74bvEJsUgz8DC6SMF1Z4DndWHFVfEIschi4APo7XNU9AG/1iWPnuiHWGQdcA+6PW/UjAC/wZgu24XYFt0Qe5M3tgC32i5F+aYE7MCYfbYLsSmaIRZJAw8Ay2yXogKxH28YKpYdXtELsdcD/RFgnu1SVKCOAS/GMcjRek7UAMfZSuCB8iaFsRKdb1gDrLwg/2ncghyNb1YDrK5aQcyC7P43qgFW11oBPBiXILv9TWqAVWXL8YIc+Z1I3Q2x9y77QTTAqrLleHPlI83dEHurkHT/KzWV9eU145HlZohFNqF7YKnqbS0vP40k90IssgJvJw6lpuP+8qkdkeNWiK/uhaXUdKWAhxCpt11IrbkTYm8r2YeAtO1SlLOagA9EbejJpW/mfcANtotQzlsIbLZdRC25EWJvTbCuSFK1sgmRhbaLqJXwh9jbVucu22WoSBHg/eXJQs4Ld4i92TYPoM/BqvZuAN5ru4haCHeIYRMw33YRKrLWRmH8OLwhFplHxDogVCjdWz5IwFnhDLFIEng/Ya1PRUkdcJ/tImYirCF5D7o/tArOjYgstV3E9QpfiEUa8J6FlQrSXa5OAglj0XeivdEqeLPw7gCdE64Qe51Za22XoWJrc/lO0CnhCnFExu2Us9J4d4JOCU+IRVYDC2yXoWJvLSJOzU0IR4i9IaUttstQqsyp38VwhNh7DnZ6wF1FyiJEnLkrTNkuoDw/OtJ7IKnaKgmmkKZUTGKKKUqFFKaYgkQJMjkSdcMkU4UZN1AbgWdrUK7v7IfY2yN4lu0iVDhdbibfO4d8zxxMXwuJgWZSw/WkgeRk/12iQDGbo5gdoDjnAqX5Z0i1XCSTMFS7he0yRGZjzDsz/y78Zf9ANZFHgFa7RaiwMGAuLGT41ApK3YvJFDK1a2gSRUrNPQzPPUdpcSfpWT1MtRTxMMZsq9X1/WI3xCJL8DZ/VzF3fgG5k6sx3YvIFNOTt7K10tRLbvkhSm3HyaaKE95+l4CfYszlIOq5XrZD/BFgib0ClE0GzOnlDB26heTgDdTZqiNRoLj4FMOr95FqunxN67wPY161UliV7IVYpBV4xM7FlW1dyxk8cBupXMOUt7TBKWHaOhla/0fqMiNX7gYKwI8xJmeztMnYDPF9wDo7F1e2XG5m5A/3YPpn2Wt5p5IsUFyzl+GVB6gvd4Rtx5i9tuuqxE6IRVLAoxCid2HlKwPm0C0MHtlAvUmEZn7CpLIDjGzcSam1mwGM+ZnteiqxNcS0Ag1wbAw0kf/9vRQvt7g1oSfXSGbnA5iVByiuTcvcVN5ctF3TRGy1xNqhFROnVjD4xh1kTdKN1reSYylO/u0qtpl2M2y7lvGC/8F6S730NMMYOLyewT1baXA9wAALSixOGD4hHTLbdi3j2fjhroKqZ80oR735JwwcvA3n1uZW0lgidfsAGeBj0hGug9lshVhFVEkwr93LYOcat55/q/HBXgyQBT4qHRKaWYbBhlikCd1HOtJ+fy9D59qi0wKPtXGAOjEYvB0yPyod4Vh3HHRL3Bbw9VSADtzKQPeSaAYYIGtIrhtipPxpBnhYOsT6IX9Bh1h7pCPq9FKGjmyI3i30eFv7KYz5NAs8JB1idTWghljNWM8chndvDe8MrFraNHjN3Io5eAcdWBNciL250tnArqcCkU9T3PU+ElEYRqrG0mEy2RKlcS+vkA7ZaKMeCLYl1lY4gnZvZWQkG599wpMgmwaYaMLH7dIhLUHXAxpiNQPnFjN0ro1623UEbUs/E01zTAL3Bl0LBBVibzfLyJzMrqCYoPTGnaHY3ilwt137XDxqkXTIzYEWQ3AtcSvh2M9L1cih95Ar73UVO60FMg1FihW+vEU6JNBOvqBCPCeg66gA5NMUj98U707KVcPkK3wpA9wWZC0aYjVtR29muBST3uhKVueu6aEe6xbpkMDe5DTEalpKgulcrWvBV02+WU8K2BBMJRpiNU1dKxjK12n/xrLhKXfk3CAdEsiunf6HWKQR4jGbJw6OrA9mO9mwW5Sf8o0sC9wYRC1BtMTaCkfEO3MZtrm1bJjUGZILRt41j3oia4KoRUOsqnZm6ZS/tLGyunIP9ail0iG+9x8EEeLmAK6hAnB+kT4Lj7V4ZMKZW2Ml8TaF9FUQIY7dtLwoGq6j2N+svdJjzSlMGWIIYLqxhlhV5WwbI4jujTbW7EJVPw/fpxtriFVVzi2pqtWJlZZiVSFukg7xdbMEDbGqSv8sHVoar3nikxQn4mtr7G+IveNaYjlJPmpyMV3sMJkbqg+xrztj+t0Sx3qSfFTkshTjPld6Io3Fqu9Omvysw+9/mMjufBgn/c1TjofGUhoS2VJVfQVOh1hvwSKgv3nSFTux1lCquK54LKc7tvQWLAKGGrVnupJkdT+ZRukQ34bn/J6BE7lxxV/BvC/AF7vhHwB+CG3fgM+fg3/8LNz/FiwvQPIErO6Dv7dcbk0ULfdL7zrN/Cd380hjmt7lLZx4/HZ2/vOrPCRgvnkfz9qsLWHGvMEdZD7P8whZelnACT7GjvJXBG/2li/TVv0OcaRa4teg+V/hngzeboevww1PwN2N0A/wI3gJ4AF45O/g+xZLralSwu6bcU+O+sc38/S6Vnq++ixf+o9dZEaKZDLJK6cxWPOuAA1Qz4d4mjZ6+AFfgishBh+zEKmQ+e0O6HsBfl4Hw72QfBw+8WP4X+Hqu/HPYeEA1H8BTtqstabE7u30B1fRmU1R/Otn+EpbM8e+dhcvb1zIAZs1XTG2Jd5MJ2mKfJ+vMJdjQZXgd4gj2yHyb7CuHxo/BZ/shkVfhLsA/gXe9x14zm51tZWobmaSb148Tltjhvx/fpzvnbnMsq6+8Ix6lGRMhvbQRpY8X+V7XGIZF95Vp29Z8Pt2OrIdIu2wrx32AayCv3mifOt0DuZ/CM7bra62EtUNo/hmpEjqn17h0WyKgZYs59uaGdx12mZFVxXG3qXkSfE/PEodAzRxnlYGx/xV30Isxvj47yPSBnzYvwuoIBy7iaH9m3X67EQ+v4rCpdSUjeGgaTf/7VcNft9O6ySBCLihN3qjDLWSq25lV7+fNfgd4iGf//8qAI19uvhhIgUoDSar+tkM+FmHhlhNqWGQtFx7EmDsDSSrmq0FTrfExuTxaYBbBatuqOpf2NjoT1Td4XfJzzqCGCfW1jgCGi/rm/F4vdW3xGf9rENDrKqy4Ex0hwuvV2+qqp/JkGk3vX7WEUSIJz/wQjlh4SldkTbeO9WF+JzfdQQR4sGp/4oKu4ZB0tkB+3OVw+RSdX32Z3wuI5AQ+9ozp4Iz76yO+4/VnZ5yjNiA/3Oogwixrz1zKjiLT+p48VhHslPO1Dpt2o3vd6IaYlW11rPU1Q1pawyQh1JXZsp+giNB1OJ/iI3pQ8eKI0FAlh/SEAOcT5M3k0+5LADHg6glqPXE2hpHxIqDZBMFnfjRWTflDLaDpt0E8oanIVbTkiqSaDvu7WwSZ0cnD3EJeCOoWjTEatrW7COD5TXGth3JTtrJd8i0m8tB1aIhVtNWP0SqrTPeM/EOZyt2apWAPwZZS1AhPk+Et+qJow1/oC6Zj+ezcW+SfF+qYku827SbQOdGBBNibzVTpLasibt0nuTNu+P5bLyvvuJoSw8Bt8IQ7G6XXQFeSwVg+REaWi7G77Z6V1PF/oCXTbsJ/I4zyBCHZGszVUubXyEVtyGn1xqpm+Dlfabd+L7YYSJBhrgb3XMrchoGSW/awci79l+OsDNphid4Hu4GdtqoB4IMsTEl4O3ArqcCs6iL+rV743FbvbvxmufhAeA5026s3Y0EfQKEPhdH1Nq9NCw8Gf1lpzub3tUKF/ECbPX71hCrmtm8nfrmS9HdBCIPpb31ZMqfloBtpt1YH3UJNsTG9OB1w6sIShjkrhdJN78TzSAfqGc4nyCBF+DfmHYT2HlLk7FxoFogy7OUHek8ybufIzP3bPSekZ+fheHqLfQJy+VcYSPERy1cUwUoWSKxdRvZxSei84w8IpReuYEE8KxpN6E68TL4EBvTi87eijwB2byDhpVv+Xv6QVA6M1zIJ/iFaTeh69fx90C1ileVDcDdwV9Y2XB+IbnX7yI5UnnRQKjNO8Pgujd4YdYlE8ohUlshrgMeBd2zKS4KKUq7t5A7uzQ8ZwtPJVmguG43wysOU8SYn9iupxI7IQYQeRBYaefiypa32xjas4V0IeP72djXz2CWdDK0/o/U1Q2TBH6PMYEvbKiWzRAvAT5i5+LKpkKK0qFbGDqxhmyp8pI+K1ouMnTr70g2914ZDzbATzDBLi+cDnshBhB5BGi1V4CyaSRD8fAGcidXkS2m7Ya55SJDq/Z7U0jHfekoxrxgpagq2Q7xSuBBewWoMCgkKXWuIde1nOTlFjJUd3D3jKVGKCzpZHjlATKN/RU73X6GMReDqOd62Q6xAJ8Gmu0VocIkV0/h9DJGziwj0TubuloHOpMj33KR/OKTsLiT+oSZ9P9/CmOeqeX1/WA3xAAi64F77Bahwmi4juLFBeR75lDsm430N5PK1ZOuNthSolSXozDrEoV5b8O8t0k3DkxrmOsXGOPrsaS1EIYQJ4HPwDXPIkpdo5ig1N9MoZCmVExBIYUpeh+CwWSHkLocifpBktmhGfWAn8WYX9SscB/Z7+Y3pojIm8CdtktR4ZcskZjVc6Xn2E+7A7hGTdiYOz2R/aDHZqrQuIQJ1/zoyYQjxMaMYGGXQKUq2GW7gOkIR4g9e4Fe20Wo2DvlUisMYQqxtwfXDttlqFhz8ncwPCEGyu+Ap2yXoWJrb3n3GaeEK8SeHeiRLyp4QzjaL2N/iGk8Y3oQ2Q/cYrsU5aYcyFr4GsA5uHEunE1B/jF47pdwexfcmIVBQIag4avw9DfhiXIHq3PsT/aYiEgGbzqmTgBRM7IUvv5f8NSDcBFgMzz2F7D9G3AI4BmY95fwhQuwhlCGYWphvJ0eHXJ62XYZyh3r4Ms/hDaAL8HWD8NHq/nvdkPLZa9H2skAQxhvp0cZcxKRt4CbbZeiwu/PYfuTsOVz0PU83PkE/KjS3/0ufPp7kOuB1np4bQT+KsBSay6cLfFVO4A+20Wo8PsWvLkf1r0KLQZk9PZ5It+An3bBdz8ET17wHtlCt/nddIQ7xMYUgBchHod1qevXAKV1cPBx+NTD1R1uNvw0fAs4DHzb5/J8Fe4QAxjTDbxuuwwVfl+G7Ydhw7er+335LcYMAF8HPiciN/lcnm/C2Ts9nkgC+Dgw33YpKhIOYcxLtouolfC3xDA6JfNFYNh2Kcp5vcCrtouoJTdCDGBMH/AC+nysrt8I8CzGROqwe3dCDGBMF/A722UoJxngRRfnRk/FrRADGPMGcNB2Gco5u1xbYlgt90Ls+S1w2nYRyhlvYcwe20X4xc0Qex1dzwPv2C5Fhd5J4BXbRfjJzRDD6PzqZ4DQHq+hrDsPvODyvOhquBtioHw+zi/RIKtrXQD+L2o90RNxO8QwOvSkQVZjnQd+iTGxmFfgfohBg6zGOg/8ytUF/tcjGiEGDbIC6CZmAYYohRg0yPHWjfcMHKsAgysLIKZLpAl4CJhruxQViC7gN3EMMEQ1xAAiaeABYJntUpSv9gE7ynMHYim6IYbR84+3ALfaLkXVXAnYjjH7bRdiW7RDPEpkHd4ZyNHqA4ivYbzbZ516S1xCDCCyGPgAUGe7FDUjvcCvMUbP7SqLT4gBRJrxnpN1hxA3dQLb4tqBVUm8Qgyjz8mbgU3o7bUr8nidVwdsFxJG8QvxKJH5wPuBWbZLUZM6h9f66tbFFcQ3xAAiKeC9wDrbpahrlIA/ALujvgpppuId4lEiy4D70LOfwqIHbyudC7YLcYGGeJR3iNsdwHpALFcTVwVgN7AHY4qWa3GGhng8kbl4Y8oLbJcSM8fxOq903vs0aYgrEVmFN9uryXYpEXcB2IkxZ2wX4ioN8WREksB7gI1Axm4xkTMAvIYxh2wX4joNcTW85+X1wC1Ag+VqXNcH7ME7SkWfe2tAQzwdXsu8FrgNaLZcjWvO44X3uA4Z1ZaG+Hp4s75W4N1mt9otJvRO4fU26zOvTzTEM+UtrLgJL9Qpy9WExTBwFG/T9oqHfava0BDXircJwQpgDbCY+I01F/E2aj8MnIzzIv2gaYj9INKIF+Y1wGzL1fjtLF5wj8Vli9iw0RD7TWQW0AYswWuhXR+qyuGdg3Ua6NLJGfZpiIMkksBby7wEL9jzCP9yyAJea9sFnNZn3PDRENvkjT+3jvtoxl6wi3iH1F3Em0l1EejW59tw0xCHjddat4z5aMJbXTX243p7wfPA0LiPPrxVQz1An47hukdD7CKvJ3w00MlKfwswwAjec+yQzpCKJg2xUo4Le6eKUmoKGmKlHKchVspxGmKlHKchVspxGmKlHKchDgERuV9Enhr32ksi8piIfGfc6ydEpFlETok342v09dkiclC8tc4qRjTEDjLeaQjbgIfHvPwp4CdGB/5jR0PsrieBz4z5/LPl11TM6E4U4fGwiLw05vON5T8fE5H7x7y+sPznS8C/i0gT3sqonDGm098SVRhpiMPj18aYx0Y/GRPop4wx3xnz+gkAY4wRkaeBPwNWoa1wbOnttNueAj4JPAT83G4pyhYNscPKt8/1wG5jTM52PcoOXcWklOO0JVbKcRpipRynIVbKcRpipRynIVbKcRpipRynIVbKcRpipRz3/6j8iU20PX/6AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# !{sys.executable} -m pip install matplotlib_venn\n",
    "from matplotlib_venn import venn2\n",
    "\n",
    "vTRprot = pd.read_csv('../Supplemental Tables - CSV/vTR-CoV-tiling/uniprot_vTR-census_wo-BSL4.csv')\n",
    "vTRList = list(vTRprot['Entry'])\n",
    "\n",
    "HHVprot = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_full-length-proteins_uID_sequence.csv')\n",
    "HHVList = list(HHVprot['Cluster ID'])\n",
    "\n",
    "shared = [i for i in vTRList if i in HHVList]\n",
    "print(len(shared))\n",
    "dom_vTR = pd.read_csv('../Supplemental Tables - CSV/vTR-CoV-tiling/vTR_domains_estON-OFF.csv')\n",
    "dom_HHV = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_domains_estON-OFF.csv')\n",
    "\n",
    "vA = list(set(list(dom_vTR[dom_vTR['Effect']=='Activation']['Protein ID'])))\n",
    "hA = list(set(list(dom_HHV[dom_HHV['Effect']=='Activation']['Protein ID'])))\n",
    "vR = list(set(list(dom_vTR[dom_vTR['Effect']=='Repression']['Protein ID'])))\n",
    "hR = list(set(list(dom_HHV[dom_HHV['Effect']=='Repression']['Protein ID'])))\n",
    "\n",
    "dom_vTR2 = vA + vR\n",
    "dom_vTR2 = list(set([v for v in dom_vTR2 if v in HHVList]))\n",
    "dom_HHV2 = hA + hR\n",
    "dom_HHV2 = list(set(dom_HHV2))\n",
    "dom_vTR_HHV2 = list(set([v for v in dom_vTR2 if v in dom_HHV2]))\n",
    "# nondom_vTR = list(set([v for v in dom_vTR if v not in dom_HHV]))\n",
    "print(len(dom_HHV2)-len(dom_vTR_HHV2), len(dom_vTR2), len(dom_vTR_HHV2))\n",
    "\n",
    "dom_new = [v for v in dom_HHV2 if v not in dom_vTR_HHV2]\n",
    "dom_new_df = dom_HHV[dom_HHV['Protein ID'].isin(dom_new)].drop_duplicates(subset='Protein ID')\n",
    "dom_new_df = dom_new_df[['Protein ID', 'Virus', 'Protein', 'Gene', 'Location']]\n",
    "dom_new_df.to_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_specific_effector_proteins.csv', index=False)\n",
    "\n",
    "venn2(subsets = (147, 3, 31), set_labels = ('HHV', 'vTR'))\n",
    "plt.savefig('../Individual Figure Panels/Figure 2/Venn_diagram_of_hit_proteins_vTR-vs-HHV.png', dpi=1200)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Domain-level Analyses - Strip Plots of Domain Scores"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Stripplots featured in Fig. 2F&G"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['gamma', 'beta', 'alpha']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-25-1d1ea56b4dcb>:37: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.\n",
      "  tempdf = df.groupby(by='UniProt Protein Family')['Max Score', 'HHV+vTR'].max().sort_values(by='Max Score', ascending=False).reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[<matplotlib.collections.PathCollection object at 0x0000023CF64522B0>, <matplotlib.collections.PathCollection object at 0x0000023CF6452640>, <matplotlib.collections.PathCollection object at 0x0000023CF6452A30>, Text(-1, -1, 'abc')] ['gamma', 'beta', 'alpha', 'Shared with vTR']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-25-1d1ea56b4dcb>:57: UserWarning: Legend does not support Text(-1, -1, 'abc') instances.\n",
      "A proxy artist may be used instead.\n",
      "See: https://matplotlib.org/users/legend_guide.html#creating-artists-specifically-for-adding-to-the-legend-aka-proxy-artists\n",
      "  plt.legend(handles = [handles[i] for i in order], labels = [labels[i] for i in order],\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Activation complete\n",
      "['gamma', 'beta', 'alpha']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-25-1d1ea56b4dcb>:37: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.\n",
      "  tempdf = df.groupby(by='UniProt Protein Family')['Max Score', 'HHV+vTR'].max().sort_values(by='Max Score', ascending=False).reset_index()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Repression complete\n"
     ]
    }
   ],
   "source": [
    "font = {'size':8,\n",
    "        'family':'Arial'}\n",
    "matplotlib.rc('font', **font)\n",
    "\n",
    "from matplotlib.text import Text\n",
    "\n",
    "alpha = '#9AC42D'\n",
    "beta = '#63A8DB'\n",
    "gamma = '#804571'\n",
    "\n",
    "###############################\n",
    "\n",
    "dom = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_domains_estON-OFF.csv')\n",
    "\n",
    "subfamDict = {'SuHV1':'alpha', 'HSV1':'alpha', 'HSV2':'alpha', 'VZV':'alpha', 'HCMV':'beta',\n",
    "              'HHV6A':'beta', 'HHV6B':'beta', 'HHV7':'beta', 'EBV':'gamma', 'KSHV':'gamma', 'CeHV1':'alpha'}\n",
    "colorDict = {'alpha':alpha, 'beta':beta, 'gamma':gamma}\n",
    "dom['Subfamily'] = dom['Virus'].map(subfamDict)\n",
    "dom['HHV+vTR'] = np.where(dom['Protein ID'].isin(dom_vTR_HHV2), 1, 0)\n",
    "\n",
    "def plot_domEffects(df, screentype, lower_lim, savename):\n",
    "    df = df[df['Effect']==screentype].sort_values(by='Max Score', ascending=False)\n",
    "    famorder = list(df.drop_duplicates(subset=['Subfamily'])['Subfamily'])\n",
    "    pal = sns.color_palette([colorDict[f] for f in famorder])\n",
    "    print(famorder)\n",
    "    \n",
    "    if screentype == 'Activation':\n",
    "        plt.figure(figsize=(4.63, 1.26))\n",
    "        labelname = 'Activation\\nlog$_2$(ON:OFF)'\n",
    "        color = '#D7AC00'\n",
    "        plt.yticks(ticks=[5, 10], labels=[5,10])\n",
    "    elif screentype == 'Repression':\n",
    "        plt.figure(figsize=(4.9,1.4))\n",
    "        labelname = 'Repression\\nlog$_2$(OFF:ON)'\n",
    "        color = '#23BEDB'\n",
    "    \n",
    "    tempdf = df.groupby(by='UniProt Protein Family')['Max Score', 'HHV+vTR'].max().sort_values(by='Max Score', ascending=False).reset_index()\n",
    "    if len(tempdf) > 40:\n",
    "        toplot = list(tempdf['UniProt Protein Family'])[:40]\n",
    "        df = df[df['UniProt Protein Family'].isin(toplot)]\n",
    "    \n",
    "    xlabs = list(tempdf['UniProt Protein Family'])\n",
    "    labcolors = list(tempdf['HHV+vTR'])\n",
    "    \n",
    "    sns.stripplot(data=df, x='UniProt Protein Family', y='Max Score', hue='Subfamily', s=2.5, palette=pal)\n",
    "    plt.xticks(rotation=90)\n",
    "    lower_y, upper_y = plt.gca().get_ylim()\n",
    "    plt.ylim(lower_lim, upper_y)\n",
    "    plt.ylabel(labelname)\n",
    "    \n",
    "    if screentype == 'Activation':\n",
    "        handles, labels = plt.gca().get_legend_handles_labels()\n",
    "        handles.append(Text(-1, -1, 'abc'))\n",
    "        labels.append('Shared with vTR')\n",
    "        print(handles, labels)\n",
    "        order = [2, 1, 0, 3] # order of genome types when ranking data from strongest to weakest domain\n",
    "        plt.legend(handles = [handles[i] for i in order], labels = [labels[i] for i in order],\n",
    "                   frameon=False, loc='upper left', bbox_to_anchor=(1.04,1), title='Subfamily',\n",
    "                   markerscale=0.3, borderpad=0, handletextpad=0, handlelength=2.2, labelspacing=0.25)\n",
    "        plt.xlabel('')\n",
    "    elif screentype == 'Repression':\n",
    "        plt.legend([], frameon=False)\n",
    "        plt.xlabel('Viral Protein Family')\n",
    "        \n",
    "    labcolDict = {1:'#ab70c1', 0:'k'}\n",
    "    for xtick, xlab, c in zip(plt.gca().get_xticklabels(), xlabs, labcolors):\n",
    "        xtick.set_label(xlab)\n",
    "        xtick.set_color(labcolDict[c])\n",
    "        \n",
    "    plt.tight_layout()\n",
    "    plt.savefig(savename, dpi=1200)\n",
    "    plt.close()\n",
    "    print('%s complete' % screentype)\n",
    "\n",
    "savenameA = '../Individual Figure Panels/Figure 2/HHV_Activation_domains_ranked_protein-label-color.png'\n",
    "plot_domEffects(dom, 'Activation', cutoffA_HHV, savenameA)\n",
    "\n",
    "savenameR = '../Individual Figure Panels/Figure 2/HHV_Repression_domains_ranked_protein-label-color.png'\n",
    "plot_domEffects(dom, 'Repression', cutoffR_HHV, savenameR)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Biological process GO terms for the 147 new effectors"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Barplot featured in Fig. 2H"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                         BP  Count\n",
      "1                  Multiple     29\n",
      "2           DNA replication     18\n",
      "3               Viral entry     11\n",
      "4           Gene expression     10\n",
      "5        Immune suppression     10\n",
      "6           Virion assembly      7\n",
      "7                DNA repair      2\n",
      "8   Modulation of apoptosis      2\n",
      "9   Protein phosphorylation      2\n",
      "10              Proteolysis      2\n",
      "0                     Other      5\n",
      "0            None annotated     49\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAALMAAACxCAYAAACP3Iz4AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjMuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8vihELAAAACXBIWXMAAAsTAAALEwEAmpwYAAAb30lEQVR4nO2de5xd09nHv78kEtQtcQkvkVDXSrUhr6JVtFpBabX9uLRoFBGCugvxalwiUfcSEremlVRcQlFxeb2K8tIKoS7VyKuuLQnJkCFI5Hn/eNbO7Dmz5+w9lzNn5sz6fj7zObP3WXuddc48s87a6/dcZGZEIrVAj2oPIBJpL6IxR2qGaMyRmiEac6RmiMYcqRmiMUdqhmjMXZBhw4YZ0J1+ChGNuQvy3nvvVXsInZJozJGaIRpzpGaIxhypGaIxR2qGXtUeQKTlzJ8/n2uuuaZDX3OXI19tcm4zm9ChY8gjzsyRmiEacwWR1EdS/2qPo7sQjbkFSBouaZ6kPuF4kKSlktYtabeLpNHAAcCukoZI+mlGf4MkTeqY0dc+0ZhbzlPAD8LvhwIPA1MAJK0raUqq7XeA/YD+wABJYyVdK+lUSccnjSQNkDRJ0pWSLpSkir+LGiQac8u5DdhXUk/cSN8q0/ZB4Bbgk9S535rZr4Adgd7h3DGAgHpgY2C90o4kjZA0S9Ks+vr6tr+LGiQac8tZDLwLHAnck5yU1ANYs6Rtll9B8pmvACxLnbvZzEYDfwAWll5kZteY2VAzG7rKKqu06Q3UKtGYW8f1wInAzHD8BPB7YHhJu9eAw4A1UueOlXQZcD+wNJy7MpyfDAwxs8WVGHStE/eZW4CZTUkdbhIehzfT/OHw+EhyQtJY4GIzezLVbmR43LfNA+zmKEZndz2GDh1qs2bNqvYwOpJCN8RxmRGpGeIyowtSCTl7xIgR7dpfNYgzcwcQlcCOIdeYJfUIP1tL6tsRgypKUOT+LGmqpJsljQrnp0i6INXuvtTv4yXNaOPrjpW0vaSLCl5SVgmMtA9Flhm/BSYCpwN1wM8qOaBWMNHMpgNIuk5SssuwsaR9zOyupKGklYEhwKuStjWzp1PPPQ68DFyO7yF/DvQBjsN3JmYCawHXpl57cLh2HLASsD6+FTc+PL8p8AtcCVwZuAZXAr8KjAY+AF7BhZibgD8C2wI/N7O6Nn4u3Y4iy4x3ge2AXwEfVnY4beZpYMvw+0nAKEkbpZ4fDtyF7xOPLrm2p5kdBnwdFz8+BlbBDfYjMzsXGBv6XY6kzYCVzOxEYAzQEzfOe4D5wPY0VQJHA0eb2ZHh9VYF5prZOOBZYOvWfgDdmSLG3BufWT7DZ5/OzA7A38PvnwFHAJOBPsHfYSTwtXB+J0mbpq6tC489gAeCGncz8DYNW0MrhX7T9KZByVsL+ApurJ8Cz4drS/c/e6TOWTj+KDXuJn+XKGfnU8SYjwe+j//BTqnoaFrHqLBmvg2YbWZzkyfM7A3gEmAgsBfwoJn9zMxG4suBUzP6mwrsJekq4If47NpX0uXA+cCF6cZm9gLQS9Kv8SXFP/HZ+Xv4rLwmTZXACcAVki4FHsWXG2WJcnY+uaKJpBtJrZnNrLOtmSuOpPvMbFi1x5EwcOBAGzNmTLv22cm35tpNNOlKa+aK0JkMOdI8RXYzkjXzE3T+NXO3YO211+7sM2lVKGLMJ+N39O/iN06RSKekiDH/GvgCvrV0LvDzio4okktr5OzuMJMXWTN/CrxpZnfjjumRNiJpVUlrVHsctUYRY+4JDJI0HA8Tqkkk3S9pndTx7ZK+lic/B0n9gBa+3InAFq0ZZ6R5iiwzTgd+iqtUh1Z2OFVlIr6EmhAk8Xn4De+A4FS/Ob4nvC7QF3fOH1/aiaQBuBK4FP8mOxV4CbgRl6pPAb4N9Je0BfBj4Engm8DuuCL4dTO7oLTvSHmKzMxnAX/BlbD23dzsXNwN7BJi+UbhPhppfmtmV+PS9J24H8e3MvrJCk59z8zOD9d9A3gI+F1o/0czOw94AN81OoLG/h9AVACLUEjONrPZZnY7fiNYk5irR3fj4UvrmNnfS5rUSVoJvwnuifuBZG3mZwWnlkrVaaWqLjxeAxwLLDKzBRnjiwpgDkWWGStJ+h7+B1i9wuOpNr/BHX2ObOb5z3EHpD3wpcYiXKpOcyVwmaT9cQelGzPSYMzFlx/LPfrM7ENJ9cBVbXsL3ZcixnwSsD8unhxb2eFUFzP7GNgsdfwwDYGpCXvm9PE6JcGpiYKYuKoGpqXbSPol8JaZvdTScUecIsa8D+5cDvA+7kkWaWfM7Oxqj6GrU8SYv2VmuwEEz7BozFUmytnZFDHmxBcYYLVKDiYSaQtFjHkKfpcvmm5XRarAJ0+/zRyVBspk09kSgleSIsa8h5l9r+IjiUTaSBFj3lbS+YT90JDBssMJS51zgQ3x/do+wMlm9m41xtMaJA0BvmRm03IbR1pMEWP+ZcVHUYxDgbfN7EyAEKjaX1Jv8uXjjfHtsp7ALDO7LulU0ll4aFM/4GzgBOAOXKl7HtgbeAcXPl7FldCzgWfwoNXl/QJ/woNePwL+hidHTB+/RIzOrhhlFUBJRwJnAEvN7BEze6Rc+wqzHW4cyZ7subhfQxH5+HRc4KgDdk06lLQlMAz/J1gE7ILvq48G+plZkl/jyhCdvV84fsLMTsjotx9+k/wn4L8zjhNaHJ2dlrMX8nFLPrduQ56cvRsemNkZnPKfwWfJZE/2THzJUUQ+7gWch8/gD6X67IEb0WhgEvACLtl/hs/6SrVLP9aFx9J+5+EGbrgSWHqcft0WRWen5ey+rNzsh9SdyVtmfGxmSyQty2nXEVwHnCfpdjz/xAp4XOI88uXj8/FdmaX4zgwAZvaipIWhrsh6wNG4nHwSnrbgxND0LElL8Hwb5fpdCfekew3434zjhCQ6ez4Fo7Mj+ZSNzpZ0L75mPgs4B8DM/toxQ+schBolo83snWqPJWGw1rPbCyaWqpGtuULR2XnGXHrzZ2Z2TltGFWk7MT9zNmWXGdFfINKViPmZuyAtCWjtTj4cZY05HROXYGbzKjecSKT15M3MSRya0ZAAsE2pBiTtAmxvVht3Ji1F0kVmdnK1x1GL5K2ZD5XUD08CI1wEaBeCUY/BMyVtBczGM2heiDvArwcswfd95+DiwoHArWY2TF7idwK+NXYa8Fi4/kA82+dmeKLCq8zsL+E1v0hTRW57M5sQtucm4PvBz+DBCA/ge9l7h/77mtlZkubgW2rn4vvIifp4Bh6/9z6ej/lYfEvxfWBlMzsKGCxpVXxfO2k3Eq9KNRP4MnBpMuZIcYrEAE7FjWVf4Kh2fv2nzOwsPBXsBDz97C7huRm4tLyRmY3HlbEtM/oAl6jH4Srg+rhxLAYW4EGiCc0pcml64EZ6Bq4uAtxhZpcCq0saCCwws8Pxfem0+rgOMACXqK/H98LTxwkHAbeZ2XH4P9TuwLLwHq4Kx5EWUsSY/w//gG8Bnmvn108SMS42s6X4DJeM6UMzW0JDgu7lz2VUQ02rZ6vjxjYauBQPPE0oVeSW4gZHqj/RoBqmK6hCQ37mutT5tPr4IZ7ObC5wEW7cy49T9yA9Un0vC68Z8zO3kSK7GV/AZ57j6BwBrXfg1VDfbOb5RcB9km7AZ/wzU8+VKnLPAWeGCJqkXnWiLK6CG+QWwEGStsYzO/07pTA2Cl4FbsX/Wd7F8zQvwJdNyXFSRngqcKWkb+DefxNp+BbIxMyuwSO4GThwYCzemEGR/My9gRXxP+qijBD8mqI0F3PI5PRJSTBqVWlJfuYa2Zprt/zMVwMb4LPyCW0ZUVegNBezmU3pTIYcaZ4iy4wPQ/j7S5L2qvSAIvnEgNZsihhzT0nH4DdNq1Z4PJFIqymyZu6FO56vANxvZp93xMAizZO3Zq7BWbvtjkaSLsQFjaSzY8jJ6BOJVIuyN4Bmdgoeezccr8y6X7n2HUnRfMotKAncISSlilPHg4L6GGkjRdbMDwJv4Zv7m0v6fVDtqk3RfMqDw1JpEp70cE1cyZwGPAVsBPzBzO4EL9oe2n4EfBFX6w7ASwevh0vwu5ccD6JxYOtcmpfqAU6W9A7wBi5GIWk8nkjxSVyA+YHlrQEjjSiyNfcnM9vDzPbCBYveFR5TUYrmUwaXtF8M8vEM4GDcJ2ICHknz/dR1vfB64Unx+MG4sdbjWUL/nXGcFTBbTqqfbGbH4AnGVwznrgBG4Mu4e6Mht5wiM3N/SV/GdzP+A1e1qo6ZmaRG+ZQlpctU1KV+z5KPl5jZZ5JK5ePB+Mz9a9zBSXgqgEX4kmuTjOMksHUxviSDEqleUlqqT6TrT/E0uZjZv0Kc4QjcWaoRkkaE5+jXr938vWqKIsZ8HD7z9cFnoCY+zlUkL59ywv3AvmH9vBqeS6O5LE0f4t52P8SXCI/izkJfwZcRM/Bg1/TxK2QEzJbhcEm74Q5PS1Ln7wB2Dql1GxHl7HyKbM19G3eWETC+yrkzapbgEnsScFheAETcmsumyMw8HJ/FhN90RWOuAM0kNo+0gCLGvCQRSlJJUSJVJMrZ2RQx5ock3Y/fAE6t8HgikVaTl2tuO/yO/ix8C2tORwwqUp7WlBvuDuTNzHvQuMwXQLfKaBTpOuTJ2WfjmTS3wvdfZ5Rr39UIpYL/LGmqpJsljQrnp0i6INXuvtTv4yW1++fQ2WT3rkiRNfOpeIAouBx7eOWGUxUmJs73kq4L0jjAxpL2MbPltfokrQwMAV6VtK2ZPZ167nG8auvl+L735/je/HH4LsVMXA28Ft/rHonvMS8MqQcGh+25s4FnQsrcSAsoImfXm9lCM1uIK1y1zNM0RICfBIwKSc0ThuP+E9fjOZbT9DSzw/CUCGvifiCr4N9oH4X8zmNDv/NCHw/hdbTTPJFlyDGgNZ8ixjxH0gxJtwKvV3pAVWYHIIlx/AzPSz2ZhopbI4GvhfM7Sdo0dW1deOwBPBAitm/GM+0nW5pJdPcv8JjK2TREn5f204hYbjifPH/mNfByCjfhf6TPOmBMHc2oUE55ReBxM5ubbKeb2RuSLsFTLewFPGhmJwIEv5BTaZqIfSpwg6ShuDPTPUBfSZfjcZTn4f4kO+IZ8leU1LPC77FbkJfS9nQ8M9DyRmbWpvRc3ZHSiO+2ksjZ3Ug4aZeUtuODJ9pC3Ke3uSxAkTK0pyFHmidvZp6K33n3Bv4M7BBrAlafmGw8m7wbwJ5mtg9e22Qc8F6bhxWJVIg8Y058besKto90AFHOzibPOLeWNBPYJhTr+XIHjCkXSQOqPYY8JK0T4gkjHUSenP1VM9szPO5hZkOaaxuk4XnJHzBEHS8NeZSbJS0Vl5xvFMWcOj8QDxZtFwlY0hmSpklau619lXAB0LfA668rKSYfbwfau6bJU8APcLHgUIKzuaSDgW/hN5JTgH/gQZ7v4hHQy7evJB1AQ5Anko7GVbT/wCOqVwf+U15dtUWJuzPaXgDsD9xlZvNDm81wL8GPgRXN7BBJT9BYju5F03LBM/DspBvhWUS3AQ6R9EBJ2xtwyft9PI7wemATSTvge9Y9gDvN7I7W/AG6M+1tzLfhsXa3Af3xFAUAB5vZdyWtgMfIzQYmmNnfJG2T0+fTwIu4Orcn7rswNASwQkPi7jsknUgqcbekb4bjJAt9advNwlgmpl6vHs92vw6e/haCHC2pL3AJruQdbWYLJN2BxxjONrOLJe2LR2jPBn4HXFbS9ubQ94M0zh3dH/flmEGGZ2IMaM0nz5/5XkkzJd0VfvJmi8X4bHskrnyVkq6NkrC0ZCxr0pjz8e3Bv2Zcm1xXNHF3VttSDgR2xr893g/nSuXorHLBpQnJk+ez2p6CR7lPxoNiCa93Dh5M22T5FOXsfPJEkz0kXYx/7S4DxhXo83rgdjwJy4/CuamSrg2/T8C/bs+X9B6QrKnfkHQ1vs5Mr6Pn4844fXCjXojfmG6X9E3xxN1ZbX9c0uZfYdz9gJUlrUlTOXo1mpYL3j6s4VfBU/8OwqPZS0sLL8aXGW/gM3OSu29t4Hg8D8ejzYw/Ug4zK/sD/CY8Crgur30t/gD35Tw/CJjUUePZcMMNbfLkydaNKPS5FEk1MAxPirIMuMLMHqrYf1akEFEBzKbIDeAc3If3BfyrMRLplBRR9Mbi0RUf466QkUinpIgxL8BLQTwPvFPh8UQKEOXsbIoY8/vAzpKm0DQqok1I2rA9++usdJf3WW2KGPMtwGG4qjWtuUapSOcbJU2XdHaBvic309fPJX2pwPXpazJl8Rb2MSVPfi9z7S6SSuMCEyaHNjECu4I0ewMoaQPcB2IoLlMLl4e/Uaa/dKTzw5IG4XvOs3AZ93gaZN1ngc2CjLsNqVrXeL3ql0JoUlZC8EE0lo9H44WELsfl8YnA88DF+D71YnzPt1Eda0mvhHMb4vvHAOdJWh1X9W4mWyp/Ga+9Pc/MbpJ0Cy7TI2lHUpHXeMjZZpL2xGX5pLzGyvhe9mm4WPImvo/9spnFe5NW0Kwxm9lb8mzu36EhwmRhc+0DR0naHVe1Lgvn5pjZCEnTaSzrTsaN+nk8Ves9+PZfutZ1khB8XfxG9M7Uc7OtQT7+UXjN03EFcSye2HuCmc2WV2DdHE9N+yT+zwXwupldIGkIQSrG1bc5uC/GCjSVynua2WGSVgKmS/obXpI5SUObRF5vAJxsZidLesXMZko6LviU9DazUUHKPzZcNwWv4nofGTfaUc7OJ08BfEPSB3hqLuHKV6liluZqSxWADDNoXTjMknUJ/S4ws9FhbbkVHgENzScEh6by8Wdm9rGk1XDDLpWue+CpedfE61jvl9EHQJ2ZLQt+H1nyd134bBZLehaf/Y8gOEzhkdcv4JE5WfcYzUnqH4XXXZZxTczPXIAia+af4n4DN+KzTmtJZN1LgUfN7APcO24nGmpdX4mH5hchkY+/idcAKeVK4NTQ5iP8W+B0vORDUsd6E3n09eH4cqOUqcCPw7JgY3zpkWY6bvzpOt5v4JHXo2iIvK6XdAiAmb0IfB4+h+No+AaLtJU8iRBff+6Eh9pfUlRarOQP7SQfkyNT51w7GF8abdnR7z/K2a2Xs4/CHWLOx294flKR/6pIYaKcnU2RZcY/zeyvZrYbflMUiXRKmjVmSQNC/N8Vwad5JrBPxw0t0hxRAcym3Nbcm8CeYfvoEzzHxosdNrJOiKQNzSw6W3VSinjN7Y87spukejP7rwqPqcOQNBxXN1/D95T/YWa/LHPJZDwBe16/g4DRZjYyp90e+Jbi/xQccqQMRYx5JfPKpkj6VYXHUw3SquVrkvam5YrlpzQNcM0qITyBVNAqrvh9Ik9yvrx8sZnNrexbrk2KGPM6kr5CQ4XWWqNUtdzeWq5YDqZpgCt4CeFxuGx9L02DVhO1cxCNyxdHWkERYz4Jl1yFiw61xtWpmXkQnjcZWqZYbp3V1pqWEB6I+2F8HZfNE6MvLV98Zekgo5ydTxFj3gYPtuyBh/l3l5S2jQJRzewDSaWK5VrAmaVt8eVGwvISwvJEM8fTNGi1tHxxEyzK2fnkqSr42nEA/hXZv6gaE38M3NnpbrxQfVQAW0+hz6XIzPwq8G8zW5rbMtIIiyWEO5QicvZduFfZEgAz27MDxhUpQ5SzsykyMx+M3+DEdLaRTk0RY74e32Ndgt+pP1LJAUXyaU7O7kY1TjIpYsxPm9n4/GaRLCQNsMb+zpEKUcSY95a0Be7gPtDM9qrwmLok8tCUsfhecm88jOpcfG9+pNq54lSkKUWM+UA8GeBp+F5oJJtDgLct+HaE5IyvAs+GuL/NJV2BxyIejsvh++J7y7OAucRSw20ir6jlDnjcXC88kHODDhlV12Q7PHczAGb2mKSHgVfMc0m/a2bHyhOv74yXLn4qNN8VN+YnzCu7NiEqgPnk7VDMwH0QfkjMM5fHbFJ1sEPEd/ozqwuPS/DZuBee3mAMXkM73aYJFvMz55K3zNgKTxz+BNBD0toWyiVEmnADME7SNNxgl+FR7dNTuaTTnI+nF1iKq4SRNpKXamAhMCFEOO+PZzfatSMG1tUws2VkO2LtGB6HhXbTU8+VRns/3P4j6z4UEkLMbKmZTTOzaMiRTkuunB3pfEQ5O5soUUdqhvYunRbpALpDdHZrpPk4M7czKlNmOOZprixd0phDLujHJPUKx5NCyFNnoFyZ4cx81GnaI890d6UrLzPewcOVltedVtOyxuvjaWj/DgwwsyMknYVnAu0HnG0hElrZZYYfJ1W2GE8zsDbwIQ37yJPCNWviW3NJmeHHaD5P82r4lt0qeFjVc5SUX460nC45MwduwxOM75s6d7CZHYpLxSeFcw+a2TnARsFHYhiefHwRHtaUkJQZfhAPOIVQthhPJ7B7OHe7mZ2Op939DvCieSqGGcDeNJQZTvI0PwR828yexqXtmcCp+D/APOC7eGm6CcEn419Zb1bSCEmzJM2qr69v0QfVXejKMzO4UdxF0/eRlDWGhtLDS/F/3rnmkdVb48nMExKHqj/QUGY4q2xxaX+luZaTvc5yeZqXAWeE19sLn/kTMsPTLAa05tKljdnMlkg6AngmnCotazywpP2LkhZKmoQnXDk69XRWmeE87scL31+ELx1OAVbFlxtJnuataZqn+Qo89/OKeMrgq2hafjnSQqJo0gUZOHCgjRkzptrDqCglW3OFRJNozF2QqABm05VvACORRkRjjtQM0ZgjNUM05kjNEI05UjNEY47UDHFrrgsiaRFeaLQIawHvtaD7ztj+5SI5R6Ixd0EkzTKzoe3dtqu3j8uMSM0QjTlSM0Rj7pq0JGaqpfFVXbZ9XDNHaoY4M0dqhi7tz9zdkLQ+cDGwAI9wmVim7SbArWY2RNIpeK3B1YET0inWJO2IBxLUA6/jUTiZbUP7TXFf8XfwxI9rl2sfrpmGpyAbkNP3hniwxbN4Ra4FeX03omgln/hT/R+8huCO4feZwArNtFs3GNxjeADA3eH8rsAZJW33AlYNvz9Qrm04PxTYCC/PXKT9iXgg7wEF2h6MBzxMwUPQyrYv/YnLjK7FukCShX8hPmM1wczeMU+NW49HzswLT71FSZVdM7sHj4AZA0wr1za0n4WHkd2D58Zrtn0o3VxHSLyZ1zdetXY4XmvyhALtGxGXGV2LN4ANcIPuR5kUuCnm4ZHjhGsbBcxKWhUvs/x7vF7Nvs21De2/CrxpZt+VNIMGG8pqfxD+T7d5OK4v0xZgCJ6jepmkT2gw4ObaNx5bmMYjXQBJ6+IJzRcBs8zs2pz295nZMEm/wA2qLzDSzD5ItbkBL0L/OvA5Hk+Z2Ta03w4PJH4fT7nwVrn24ZrheFBv/5y+t8UrNMzHa5T3yeu70fXRmCO1QlwzR2qGaMyRmiEac6RmiMYcaTck9ZHUv1qvH7fmIoRsquPxLbw18Jx4p7WiqwOAT4HpeQ0rQdzNiCDpGGCxmV0fjg/CE6+UZlT9xMymp7b8SrOkHovn7/uJmZXm16s4cWaOAAwGLkoOzGyqpAeCMLIC7leRNdsuM7Nxkr6JZ0l9EDf4DjdkiGvmiPMcsFtyIOlUGlJiJRlVl+L+GNCgKJZmSa3q13ycmSPgeakvl3QTbpCv0zSj6qvAtZKG0mDUpbwGnBmWIXWVHXJT4po5UjPEZUakZojGHKkZojFHaoZozJGaIRpzpGaIxhypGaIxR2qG/wdo2fcH6rY9vgAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 180x180 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "font = {'size':8,\n",
    "        'family':'Arial'}\n",
    "matplotlib.rc('font', **font)\n",
    "\n",
    "hgo = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_specific_effector_proteins_GO_info.tsv', sep='\\t')\n",
    "pID = list(hgo['Entry'])\n",
    "GObp = list(hgo['Gene Ontology (biological process)'])\n",
    "GOcc = list(hgo['Gene Ontology (cellular component)'])\n",
    "GOmf = list(hgo['Gene Ontology (molecular function)'])\n",
    "\n",
    "def extract_GO(ID, GO):\n",
    "    new_ID = []\n",
    "    new_GO = []\n",
    "    new_code = []\n",
    "    for p, g in zip(ID, GO):\n",
    "#         print(p, g)\n",
    "        if type(g) == float:\n",
    "            new_ID.append(p)\n",
    "            new_GO.append(g)\n",
    "            new_code.append(g)\n",
    "        else:\n",
    "            g_sub = g.split(';')\n",
    "            for s in g_sub:\n",
    "                new_ID.append(p)\n",
    "                new_GO.append(s.split('[')[0][:-1])\n",
    "                new_code.append(s.split('[')[1][:-1])\n",
    "    df = pd.DataFrame({'UniProt ID':new_ID, 'GO':new_GO, 'Code':new_code})\n",
    "    df['Count'] = 1\n",
    "    df = df.fillna('-')\n",
    "    df = df.groupby(by=['GO', 'Code']).sum().reset_index()\n",
    "    return df\n",
    "\n",
    "GObp_df = extract_GO(pID, GObp)\n",
    "GOcc_df = extract_GO(pID, GOcc)\n",
    "GOmf_df = extract_GO(pID, GOmf)\n",
    "\n",
    "GObp_df.to_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_specific_effector_proteins_GO_BP_agg.csv', index='False')\n",
    "GOcc_df.to_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_specific_effector_proteins_GO_CC_agg.csv', index='False')\n",
    "GOmf_df.to_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_specific_effector_proteins_GO_MF_agg.csv', index='False')\n",
    "\n",
    "GOBP = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_specific_effector_proteins_GO_info_ANNOTATED.csv').fillna('None annotated')\n",
    "GOBP['Count'] = 1\n",
    "GOBP = GOBP.groupby('BP').sum().sort_values(by='Count', ascending=False).reset_index()\n",
    "\n",
    "tocut = 11\n",
    "GOBP = pd.concat([GOBP.head(tocut), pd.DataFrame({'BP':'Other', 'Count':[sum(list(GOBP.tail(len(GOBP) - tocut)['Count']))]})])\n",
    "GOBP = pd.concat([GOBP.tail(len(GOBP) - 1), GOBP.head(1)])\n",
    "GOBP = GOBP.replace(to_replace=['Regulation of gene expression', 'Suppression of immune response', 'Multifunctional'],\n",
    "                    value=['Gene expression', 'Immune suppression', 'Multiple'])\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(2.5, 2.5))\n",
    "\n",
    "for x, y, bp in zip(list(GOBP['Count']), list(range(len(GOBP['BP'])))[-1::-1], list(GOBP['BP'])):\n",
    "    if bp == 'Gene expression':\n",
    "        ax.barh(y, x, color='#D90077')\n",
    "    elif bp == 'Multiple':\n",
    "        ax.barh(y, x, color='#D90077') # for stacked bar effect, first plot full count in gene expression color\n",
    "        ax.barh(y, x-4, color='#999999') # add gray bar on top with height of non-gene expression count (there are 4 gene expression counts)\n",
    "    else:\n",
    "        ax.barh(y, x, color='#999999')\n",
    "\n",
    "print(GOBP)\n",
    "# sns.barplot(data=GOBP, x='Count', y='BP', color='#999999')\n",
    "plt.ylabel('Annotated Biological Process')\n",
    "plt.yticks(ticks=list(range(len(GOBP['BP'])))[-1::-1], labels=list(GOBP['BP']))\n",
    "plt.xlabel('Count')\n",
    "plt.xticks(ticks=[0, 10, 20, 30, 40, 50], labels=[0, 10, 20, 30, 40, 50])\n",
    "sns.despine()\n",
    "plt.tight_layout()\n",
    "plt.savefig('../Individual Figure Panels/Figure 2/HHV_specific_effector_proteins_GO_BP_barplot.png', dpi=1200)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### RNA-seq for U84, RL5A, and U8"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Volcano plots for Fig. 2I-K"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "loaded\n",
      "Functions defined\n"
     ]
    }
   ],
   "source": [
    "from matplotlib.ticker import MultipleLocator\n",
    "\n",
    "font = {'size':8,\n",
    "        'family':'Arial'}\n",
    "matplotlib.rc('font', **font)\n",
    "print('loaded')\n",
    "\n",
    "def volcano(df, symbol, savename):\n",
    "    df['-log10(padj)'] = -np.log10(df['padj'])\n",
    "    df['-log10(padj)'] = np.where(df['-log10(padj)']==np.inf, 300, df['-log10(padj)'])\n",
    "    df['-log10(padj)'] = np.where(df['-log10(padj)']>50, 50, df['-log10(padj)'])\n",
    "    df['Color'] = np.where((df['-log10(padj)']>-np.log10(0.05)) & (df['log2FoldChange']<-1), -1, 0)\n",
    "    df['Color'] = np.where((df['-log10(padj)']>-np.log10(0.05)) & (df['log2FoldChange']>1), 1, df['Color'])\n",
    "    \n",
    "    pal = sns.color_palette(['#00b0c7', '#999999', '#DAA520'])\n",
    "    plt.figure(figsize=(1.8, 1.5))\n",
    "    sns.scatterplot(data=df, x='log2FoldChange', y='-log10(padj)', hue='Color', alpha=0.5, linewidth=0, s=5, palette=pal)\n",
    "    plt.axvline(x=1, ls=':', color='#333333', lw=0.5)\n",
    "    plt.axvline(x=-1, ls=':', color='#333333', lw=0.5)\n",
    "    plt.axhline(y=-np.log10(0.05), ls=':', color='#333333', lw=0.5)\n",
    "    plt.xlim(-10, 10)\n",
    "    plt.xticks([-9, -6, -3, 0, 3, 6, 9])\n",
    "    plt.axes().xaxis.set_minor_locator(MultipleLocator(1))\n",
    "#     matplotlib.axis.xaxis.set_minor_locator(MultipleLocator(1))\n",
    "    plt.xlabel('log$_2$(Fold Change)')\n",
    "    plt.ylim(-5, 55)\n",
    "    plt.ylabel('-log$_{10}$(Adj. p-value)')\n",
    "    plt.legend('', frameon=False)\n",
    "    plt.title(symbol + ' vs. negative control', fontsize=8)\n",
    "    plt.tight_layout()\n",
    "    sns.despine()\n",
    "    \n",
    "    plt.savefig(savename, dpi=1200)\n",
    "    plt.close()\n",
    "\n",
    "def GOterms(df, symbol, direction, savename):\n",
    "    df['Fraction'] = df['Significant']/df['Annotated']\n",
    "    df['-log10(p)'] = -np.log10(df['result1'])\n",
    "    plt.figure(figsize=(6,3.5))\n",
    "    sns.scatterplot(data=df.sort_values(by='Fraction', ascending=False), x='Fraction', y='Term', hue='-log10(p)', size='Significant')\n",
    "    plt.legend(loc='center left', bbox_to_anchor=(1, 0.5), ncol=2)\n",
    "    plt.xlim(0,1)\n",
    "    plt.xlabel('Fraction of Genes in Set')\n",
    "    plt.title('%s - GO Terms for %sregulated Genes' % (symbol, direction), fontsize=8)\n",
    "    plt.tight_layout()\n",
    "    plt.savefig(savename, dpi=1200)\n",
    "    plt.close()\n",
    "\n",
    "print('Functions defined')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-10-9db2822f935a>:23: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.\n",
      "  plt.axes().xaxis.set_minor_locator(MultipleLocator(1))\n",
      "<ipython-input-10-9db2822f935a>:23: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.\n",
      "  plt.axes().xaxis.set_minor_locator(MultipleLocator(1))\n",
      "<ipython-input-10-9db2822f935a>:23: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.\n",
      "  plt.axes().xaxis.set_minor_locator(MultipleLocator(1))\n"
     ]
    }
   ],
   "source": [
    "U84deseq = pd.read_csv('../Supplemental Tables - CSV/RNA-seq/DEseq/CL457-CL048_dox.csv')\n",
    "U8deseq = pd.read_csv('../Supplemental Tables - CSV/RNA-seq/DEseq/CL456-CL048_dox.csv')\n",
    "RL5Adeseq = pd.read_csv('../Supplemental Tables - CSV/RNA-seq/DEseq/CL418-CL048_dox.csv')\n",
    "\n",
    "volcano(U84deseq, 'U84', '../Individual Figure Panels/Figure 2/U84_volcano_replotted_unlabeled.png')\n",
    "volcano(U8deseq, 'U8', '../Individual Figure Panels/Figure 2/U8_volcano_replotted_unlabeled.png')\n",
    "volcano(RL5Adeseq, 'RL5A', '../Individual Figure Panels/Figure 2/RL5A_volcano_replotted_unlabeled.png')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Small tiling plots for Fig. 2I-K"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Domain calling/plotting functions defined\n"
     ]
    }
   ],
   "source": [
    "# Author: Connor Ludwig\n",
    "# Organization: Bintu Lab, Stanford University\n",
    "# Date updated: 5/10/2022\n",
    "\n",
    "# NOTE: these functions use a few global variables, including:\n",
    "# - hits thresholds for activation and repression screens\n",
    "# - minimum and maximum screen scores for non-controls to set axis limits\n",
    "# - dictionary that maps UniProt IDs to protein and virus\n",
    "\n",
    "# make sure to run the above cells to get these global variables\n",
    "\n",
    "import mpl_axes_aligner\n",
    "\n",
    "# this function gets domain start and stop coordinates, as well as max tile score\n",
    "def domainLimits(temp_df):\n",
    "    span_starts = []\n",
    "    span_ends = []\n",
    "    scores = []\n",
    "    temp_df_hits = temp_df[temp_df['Hit']==1]\n",
    "    indices = range(0, len(temp_df_hits))\n",
    "    temp_df_hits.insert(0, 'index', indices)\n",
    "    temp_df_hits.set_index('index', inplace=True)\n",
    "    \n",
    "    if len(temp_df_hits) == 0:\n",
    "        return span_starts, span_ends, scores\n",
    "    temp_start = np.nan\n",
    "    temp_scores = []\n",
    "    # create range for all tiles excluding the last tile\n",
    "    for i in range(0, len(temp_df_hits)):\n",
    "        # if tile is last one\n",
    "        if i == (len(temp_df_hits) - 1):\n",
    "            # if temporary start of domain has been defined\n",
    "            if ~np.isnan(temp_start):\n",
    "                span_starts.append(temp_start)\n",
    "                span_ends.append(temp_df_hits.loc[i, 'Tile End'])\n",
    "                scores.append(max(temp_scores))\n",
    "            # if tile is the only tile (first and last)\n",
    "            elif int(temp_df_hits.loc[i, 'Tile ID'].split('_')[-1]) == 1:\n",
    "                span_starts.append(temp_df_hits.loc[i, 'Tile Start'])\n",
    "                span_ends.append(temp_df_hits.loc[i, 'Tile End'])\n",
    "                scores.append(temp_df_hits.loc[i, 'Avg'])\n",
    "        # if tile is not last one\n",
    "        else:\n",
    "            # if not consecutive\n",
    "            if (int(temp_df_hits.loc[i, 'Tile ID'].split('_')[-1]) + 1) != (int(temp_df_hits.loc[i+1, 'Tile ID'].split('_')[-1])):\n",
    "                # if a temporary start of the domain has been defined, then store that start and the end of this tile\n",
    "                if ~np.isnan(temp_start):\n",
    "                    span_starts.append(temp_start)\n",
    "                    span_ends.append(temp_df_hits.loc[i, 'Tile End'])\n",
    "                    scores.append(max(temp_scores))\n",
    "                    # reset temporary start to nan\n",
    "                    temp_start = np.nan\n",
    "                    temp_scores = []\n",
    "                # if the tile is at the N-terminus, then store the start/stop of the domain\n",
    "                elif int(temp_df_hits.loc[i, 'Tile ID'].split('_')[-1]) == 1:\n",
    "                    span_starts.append(temp_df_hits.loc[i, 'Tile Start'])\n",
    "                    span_ends.append(temp_df_hits.loc[i, 'Tile End'])\n",
    "                    scores.append(temp_df_hits.loc[i, 'Avg'])\n",
    "            # if consecutive\n",
    "            else:\n",
    "                # if start of domain is not yet defined\n",
    "                if np.isnan(temp_start):\n",
    "                    temp_start = temp_df_hits.loc[i, 'Tile Start']\n",
    "                    temp_scores.append(temp_df_hits.loc[i, 'Avg'])\n",
    "                # if tile is in the middle of a domain\n",
    "                else:\n",
    "                    temp_scores.append(temp_df_hits.loc[i, 'Avg'])\n",
    "                    continue\n",
    "                \n",
    "    return span_starts, span_ends, scores\n",
    "\n",
    "###############################\n",
    "        \n",
    "# this function makes tiling plots\n",
    "def plotDomain_small(gene, temp_df, protDF, domainDict, indivDir_path, save_root):\n",
    "    # subset for activation\n",
    "    temp_dfA = temp_df[temp_df['Screen']=='Activation']\n",
    "    temp_dfA = temp_dfA.drop_duplicates(subset='Tile ID')\n",
    "    temp_IDsA = list(temp_dfA['Oligo ID'])\n",
    "    temp_dfA.set_index('Oligo ID', inplace=True)\n",
    "    \n",
    "    # get activation points and error\n",
    "    xA = [(1 + (temp_dfA.loc[ID]['Tile Start'] + temp_dfA.loc[ID]['Tile End'] - 1)/2) for ID in temp_IDsA]\n",
    "    xerrA = [(temp_dfA.loc[ID]['Tile End'] - temp_dfA.loc[ID]['Tile Start'] + 1)/2 for ID in temp_IDsA]\n",
    "    yA = [temp_dfA.loc[ID]['Avg'] for ID in temp_IDsA]\n",
    "    yerrA = list(temp_dfA['Standard Error'])\n",
    "    \n",
    "    # get activation domain info\n",
    "    domain_startsA, domain_endsA, scoresA = domainLimits(temp_dfA)\n",
    "    print('Activation domain starts/stops/scores:', domain_startsA, domain_endsA, scoresA)\n",
    "\n",
    "    # subset for repression\n",
    "    temp_dfR = temp_df[temp_df['Screen']=='Repression']\n",
    "    temp_dfR = temp_dfR.drop_duplicates(subset='Tile ID')\n",
    "    temp_IDsR = list(temp_dfR['Oligo ID'])\n",
    "    temp_dfR.set_index('Oligo ID', inplace=True)\n",
    "    \n",
    "    # get repression points and error\n",
    "    xR = [(1 + (temp_dfR.loc[ID]['Tile Start'] + temp_dfR.loc[ID]['Tile End'] - 1)/2) for ID in temp_IDsR]\n",
    "    xerrR = [(temp_dfR.loc[ID]['Tile End'] - temp_dfR.loc[ID]['Tile Start'] + 1)/2 for ID in temp_IDsR]\n",
    "    yR = [temp_dfR.loc[ID]['Avg'] for ID in temp_IDsR]\n",
    "    yerrR = list(temp_dfR['Standard Error'])\n",
    "    \n",
    "    # get repression domain info\n",
    "    domain_startsR, domain_endsR, scoresR = domainLimits(temp_dfR)\n",
    "    print('Repression domain starts/stops/scores:', domain_startsR, domain_endsR, scoresR)\n",
    "\n",
    "    # create axis objects and plot tile centers vs log2(ON:OFF) scores\n",
    "    fig, ax = plt.subplots(figsize=(1.2,0.8))\n",
    "    ax2 = ax.twinx()\n",
    "    colors = ['#23BEDB', '#D7AC00', '#DB3982']\n",
    "    markersize = 4\n",
    "    \n",
    "    # the following stores extended domain and strongest tile info in domainDict\n",
    "    # it also draws horizontal spans that highlight the strongest tile in each domain\n",
    "    # activators\n",
    "    if len(domain_startsA) != 0:\n",
    "        for s, e, c in zip(domain_startsA, domain_endsA, scoresA):\n",
    "            domA = temp_dfA[(temp_dfA['Tile Start']>=s) & (temp_dfA['Tile End']<=e)]\n",
    "            startindex = int(s)\n",
    "            endindex = int(e)\n",
    "            domseq = protDF.loc[gene, 'Representative Sequence'][(startindex - 1):endindex]\n",
    "            max_avg = max(list(domA['Avg']))\n",
    "            max_coords = [domA[domA['Avg']==max_avg]['Tile Start'][0], domA[domA['Avg']==max_avg]['Tile End'][0]]\n",
    "            max_seq = domA[domA['Avg']==max_avg]['Tile Sequence'][0]\n",
    "            \n",
    "            to_append = [gene, startindex, endindex, len(domseq), domseq,\n",
    "                         max_coords[0], max_coords[1], max_seq, c, 'Activation']\n",
    "            for k, v in zip(domainDict.keys(), to_append):\n",
    "                domainDict[k].append(v)\n",
    "            \n",
    "            ax.axvspan(xmin=max_coords[0], xmax=max_coords[1], color='#f5dc79', alpha=0.4, lw=0.8, zorder=-200)\n",
    "    \n",
    "    # repressors\n",
    "    if len(domain_startsR) != 0:\n",
    "        for s, e, c in zip(domain_startsR, domain_endsR, scoresR):\n",
    "            domR = temp_dfR[(temp_dfR['Tile Start']>=s) & (temp_dfR['Tile End']<=e)]\n",
    "            startindex = int(s)\n",
    "            endindex = int(e)\n",
    "            domseq = protDF.loc[gene, 'Representative Sequence'][(startindex - 1):endindex]\n",
    "            max_avg = max(list(domR['Avg']))\n",
    "            max_coords = [domR[domR['Avg']==max_avg]['Tile Start'][0], domR[domR['Avg']==max_avg]['Tile End'][0]]\n",
    "            max_seq = domR[domR['Avg']==max_avg]['Tile Sequence'][0]\n",
    "            \n",
    "            to_append = [gene, startindex, endindex, len(domseq), domseq,\n",
    "                         max_coords[0], max_coords[1], max_seq, c, 'Repression']\n",
    "            for k, v in zip(domainDict.keys(), to_append):\n",
    "                domainDict[k].append(v)\n",
    "            \n",
    "            ax.axvspan(xmin=max_coords[0], xmax=max_coords[1], color='#bbdaee', alpha=0.4, lw=0.8, zorder=-150)\n",
    "     \n",
    "    # plot tiles, with activators on y-axis 1 and repressors on y-axis 2\n",
    "    ax.scatter(xA, yA, c=colors[1], zorder=-100, s=markersize)\n",
    "    for xi, yi, exi, eyi in zip(xA, yA, xerrA, yerrA):\n",
    "        ax.errorbar(xi, yi, xerr=exi, yerr=eyi, color=colors[1], zorder=-100, lw=0.8)\n",
    "    \n",
    "    ax2.scatter(xR, yR, c=colors[0], zorder=-50, s=markersize)\n",
    "    for xi, yi, exi, eyi in zip(xR, yR, xerrR, yerrR):\n",
    "        ax2.errorbar(xi, yi, xerr=exi, yerr=eyi, color=colors[0], zorder=-50, lw=0.8)\n",
    "    \n",
    "    uID = gene.split('(')[-1]\n",
    "    \n",
    "    titleDict = {'P52534':'HHV7 U84', 'F5HF23':'HCMV RL5A', 'P52523':'HHV7 U8'}\n",
    "    ax.set_title(titleDict[uID], fontsize=8, pad=0)\n",
    "    colorsTitle = ['#01a4c2', '#c29d09']\n",
    "    ax.set_yticks([])\n",
    "    ax2.set_yticks([])\n",
    "    ax.set_ylabel('Act.', color=colorsTitle[1], labelpad=1)\n",
    "    ax2.set_ylabel('Repr.', color=colorsTitle[0], rotation=270, labelpad=8)\n",
    "    ax2.axhspan(ymin=minScoreA - 0.5, ymax=cutoffR_HHV, color='#FFFFFF', alpha=0.6, zorder=-15)\n",
    "    ax2.axhline(y=cutoffR_HHV, linestyle='--', color='#666666', lw=1)\n",
    "    ax.set_ylim(minScoreA - 0.5, maxScoreA + 0.5)\n",
    "    ax2.set_ylim(minScoreR - 0.5, maxScoreR + 0.5)\n",
    "    mpl_axes_aligner.align.yaxes(ax, cutoffA_HHV, ax2, cutoffR_HHV, 0.5)\n",
    "\n",
    "    plt.tight_layout()\n",
    "    \n",
    "    # save figure and close current plot\n",
    "    fig_path = os.path.join(indivDir_path, gene_uID_list_dict2[gene] + save_root)\n",
    "    # do this because you cannot save files with '|' in the name\n",
    "    fig_path = fig_path.replace('|', '-')\n",
    "    fig.savefig(fig_path, dpi=1200)\n",
    "    plt.close()\n",
    "    \n",
    "    return domainDict\n",
    "\n",
    "###############################\n",
    "\n",
    "# this function batch produces tiling plots while simultaneously storing domain info\n",
    "def extractDomains(df, protDF, lib_fullName, indivDir_path, save_root):\n",
    "    if 'herpesvirus' not in lib_fullName:\n",
    "        df = df[df['Type']==lib_fullName]\n",
    "    else:\n",
    "        df = df[(df['Library'].str.contains('herpesvirus')) & (~df['Type'].str.contains('control'))]\n",
    "    \n",
    "    # get hits and edit to avoid regex characters\n",
    "    tileHits = list(df[df['Hit'] == 1]['Oligo ID'])\n",
    "    geneHits = []\n",
    "    for hit in tileHits:\n",
    "        if '|' in hit:\n",
    "            geneHits.append('_'.join(hit.split('_')[1:3]))\n",
    "        else:\n",
    "            geneHits.append(hit.split('_')[1])\n",
    "\n",
    "    geneHits = set(geneHits)\n",
    "    geneHitsLen = str(len(geneHits))\n",
    "    \n",
    "    # find domains and construct data frame with domain info\n",
    "    domainDict = {'Protein ID':[],\n",
    "                  'Extended Start':[],\n",
    "                  'Extended End':[],\n",
    "                  'Extended Length':[],\n",
    "                  'Extended Sequence':[],\n",
    "                  'Max Start':[],\n",
    "                  'Max End':[],\n",
    "                  'Max Sequence':[],\n",
    "                  'Max Score':[],\n",
    "                  'Effect':[]}\n",
    "    incr = 0\n",
    "    for gene in geneHits:\n",
    "        incr += 1\n",
    "        # make temporary dataframe for each gene\n",
    "        # '|' character is read as an 'or' by regex, so don't include when searching\n",
    "        if '|' in gene:\n",
    "            genesearch = '.'.join(gene.split('|'))\n",
    "        else:\n",
    "            genesearch = gene\n",
    "        temp_df = df[df['Oligo ID'].str.contains(genesearch)]\n",
    "        temp_df = temp_df.sort_values(by='Oligo ID')\n",
    "        \n",
    "        # make a plot with enrichment scores for each gene containing hit tiles\n",
    "        print('Protein %s of %s: plotting log2(ON:OFF) for %s' % (str(incr), geneHitsLen, gene))\n",
    "        domainDict = plotDomain_small(gene, temp_df, protDF, domainDict, indivDir_path, save_root)\n",
    "\n",
    "    finalDF = pd.DataFrame(domainDict)\n",
    "    print('Finished')\n",
    "    \n",
    "    return finalDF\n",
    "\n",
    "print('Domain calling/plotting functions defined')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Protein 1 of 3: plotting log2(ON:OFF) for P52534\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [11.0] [150.0] [4.900162611]\n",
      "Protein 2 of 3: plotting log2(ON:OFF) for F5HF23\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [1.0] [95.0] [2.821301289]\n",
      "Protein 3 of 3: plotting log2(ON:OFF) for P52523\n",
      "Activation domain starts/stops/scores: [] [] []\n",
      "Repression domain starts/stops/scores: [51.0] [160.0] [4.288637273]\n",
      "Finished\n"
     ]
    }
   ],
   "source": [
    "# the following creates dictionaries to map UniProt ID to protein and virus names\n",
    "# this is specifically to produce tiling plot titles and save names\n",
    "uID_to_gene_df = pd.read_csv('../Supplemental Tables - CSV/uID_protein_virus_mapping_HHV_expanded.csv')\n",
    "uID_to_gene_df = uID_to_gene_df.rename(columns={'UniProt ID':'Protein ID'})\n",
    "uID_list = list(uID_to_gene_df['Protein ID'])\n",
    "uID_to_gene_df['Virus Protein (ProteinID)'] = uID_to_gene_df['Virus'] + ' ' + uID_to_gene_df['Protein'] + ' (' + uID_to_gene_df['Protein ID'] + ')'\n",
    "uID_to_gene_df['Protein_Virus_ProteinID'] = uID_to_gene_df['Protein'] + '_' + uID_to_gene_df['Virus'] + '_' + uID_to_gene_df['Protein ID']\n",
    "gene_uID_list1 = list(uID_to_gene_df['Virus Protein (ProteinID)'])\n",
    "gene_uID_list_dict1 = dict(zip(uID_list, gene_uID_list1))\n",
    "gene_uID_list2 = list(uID_to_gene_df['Protein_Virus_ProteinID'])\n",
    "gene_uID_list_dict2 = dict(zip(uID_list, gene_uID_list2))\n",
    "\n",
    "###############################\n",
    "\n",
    "df = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_ActivD2-ReprD5_detailed.csv')\n",
    "indivDir_path = '../Individual Figure Panels/Figure 2'\n",
    "save_root = '_tileScores_EBNA.png'\n",
    "HHVprot = pd.read_csv('../Supplemental Tables - CSV/HHV-tiling/HHV_full-length-proteins_uID_sequence.csv')\n",
    "HHVprot = HHVprot.rename(columns={'Cluster ID':'Entry'})\n",
    "HHVprot = HHVprot[['Entry', 'Representative Sequence']]\n",
    "HHVprot = HHVprot.set_index('Entry')\n",
    "\n",
    "if os.path.isdir(indivDir_path) == False:\n",
    "    print('Making directory called %s to store plots' % indivDir_path)\n",
    "    os.mkdir(indivDir_path)\n",
    "\n",
    "# compute minimum and maximum log2(ON:OFF) for plots\n",
    "cutoffA_HHV = 1.0846347878236864\n",
    "cutoffR_HHV = 0.6987727639587654\n",
    "limsA_HHV = (-9.044003705, 11.45554289)\n",
    "limsR_HHV = (-3.76771998, 7.7418403210000015)\n",
    "minScoreA = limsA_HHV[0]\n",
    "maxScoreA = limsA_HHV[1]\n",
    "minScoreR = limsR_HHV[0]\n",
    "maxScoreR = limsR_HHV[1]\n",
    "\n",
    "plotlist = ['P52534', 'F5HF23', 'P52523']\n",
    "df = df[(df['Protein ID'].isin(plotlist))]\n",
    "HHVprot = HHVprot[HHVprot.index.isin(plotlist)] # EBNA2 type 1 and type 2 UniProt IDs\n",
    "\n",
    "# extract domains and make tiling plots\n",
    "domDF_HHV = extractDomains(df, HHVprot, 'herpesvirus', indivDir_path, save_root)\n",
    "domDF_HHV = pd.merge(domDF_HHV, uID_to_gene_df, on='Protein ID', how='left')\n",
    "domDF_HHV = domDF_HHV.drop_duplicates(subset=['Protein ID', 'Max Score', 'Effect'])\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
